---
title: "BioHyV BV2 Manuscript"
author: "Greg Cary"
date: "`r Sys.Date()`"
output: 
  html_document: 
    fig_width: 5
    fig_height: 4
  html_notebook:
    code_folding: hide
    theme: readable
    # css: style.css
    toc: true
    toc_float: true
---

# Setup
## pakages
```{r setup}
syn.client <- reticulate::import('synapseclient')
syn.utils <- reticulate::import('synapseutils')
syn <- syn.client$Synapse()

library(callr)
library(clusterProfiler)
library(furrr)
library(tidyverse)

theme_set(theme_bw())
```

Specify the function for comparing GSEA results between studies.

```{r gsea_comp_function}
gsea_comp <- function( enr_tbl, GO_col,
                       study1_nes, study1_sig,
                       study2_nes, study2_sig, sig_thresh = 0.05){

  terms <- enr_tbl %>%
    select(GO = all_of(GO_col),
           NES1 = all_of(study1_nes),
           PADJ1 = all_of(study1_sig),
           NES2 = all_of(study2_nes),
           PADJ2 = all_of(study2_sig)) %>%
    distinct()

  x <- tibble(
    a = terms %>%
      filter( NES1 > 0, PADJ1 <= sig_thresh, NES2 > 0, PADJ2 <= sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    b = terms %>%
      filter( PADJ1 > sig_thresh, NES2 > 0, PADJ2 <= sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    c = terms %>%
      filter( NES1 < 0, PADJ1 <= sig_thresh, NES2 > 0, PADJ2 <= sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    d = terms %>%
      filter( NES1 > 0, PADJ1 <= sig_thresh, PADJ2 > sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    e = terms %>%
      filter( PADJ1 > sig_thresh, PADJ2 > sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    f = terms %>%
      filter( NES1 < 0, PADJ1 <= sig_thresh, PADJ2 > sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    g = terms %>%
      filter( NES1 > 0, PADJ1 <= sig_thresh, NES2 < 0, PADJ2 <= sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    h = terms %>%
      filter( PADJ1 > sig_thresh, NES2 < 0, PADJ2 <= sig_thresh) %>%
      pull(GO) %>% unique() %>% length(),
    i = terms %>%
      filter( NES1 < 0, PADJ1 <= sig_thresh, NES2 < 0, PADJ2 <= sig_thresh) %>%
      pull(GO) %>% unique() %>% length()
  )

  ppv = x$a/sum(x$a,x$b,x$c)
    if(is.na(ppv)){ppv = 0}
  npv = x$i/sum(x$g,x$h,x$i)
    if(is.na(npv)){npv = 0}
  ppv_c = sum(x$a, x$d, x$g)/sum(x)
    if(is.na(ppv_c)){ppv_c = 0}
  npv_c = sum(x$c, x$f, x$i)/sum(x)
    if(is.na(npv_c)){npv_c = 0}
  gain = ((ppv - ppv_c)+(npv - npv_c))/2

  mtx = rbind(c(x$a, x$b,x$c), c(x$d, x$e,x$f), c(x$g,x$h,x$i))

  tau = try(cor.test(terms$NES1, terms$NES2, meethod = 'kendall'), silent = T)
  if(class(tau) == 'try-error'){ tau = tibble(NA)
  } else { tau = broom::tidy(tau) }

  res = tibble(
    terms = list(terms), olap = list(mtx),
    ppv, ppv_c, npv, npv_c, gain,
    tau
  )
  
  return(res)
  
}
```

## data
Omics table
```{r}
syn$login()

scores <- syn$tableQuery('select * from syn25575156') %>% 
  .$filepath %>% read_csv( col_types = cols() ) 

omics <- syn$tableQuery('SELECT * FROM syn22758536') %>% 
  .$filepath %>% read_csv( col_types = cols() ) 

omics.gsea <- syn$get('syn52749189') %>% .$path %>% readRDS()
```

Biodomain definitions
```{r}
syn$login()

# biodomains
biodom <- readRDS(syn$get('syn25428992') %>% .$path)
mm.biodom <- readRDS(syn$get('syn26592124') %>% .$path)

# domain labels
dom.cols <- read_csv(syn$get('syn26856828') %>% .$path)

biodom <- full_join( biodom, dom.cols,by = c('Biodomain'='domain') ) %>%
  mutate(Biodomain = case_when(Biodomain == 'none' ~ NA_character_, T ~ Biodomain),
         Subdomain = if_else(is.na(Subdomain), 'none',Subdomain),
         sd = str_c(abbr, ' | ', Subdomain)
         )

mm.biodom <- full_join( mm.biodom, dom.cols,by = c('Biodomain'='domain') ) %>%
  mutate(Biodomain = case_when(Biodomain == 'none' ~ NA_character_, T ~ Biodomain),
         Subdomain = if_else(is.na(Subdomain), 'none',Subdomain),
         sd = str_c(abbr, ' | ', Subdomain)
         )
```

BD term lists
```{r}
##
# HUMAN
##

# curate biodomain and sub-domain gene lists
bd.terms <- tibble( set = unique(biodom$Biodomain) ) %>% 
  mutate( terms = map(set, ~ biodom %>% filter(Biodomain == .x) %>% pull(GO_ID)),
          genes = map(set, ~ biodom %>% filter(Biodomain == .x) %>% pull(symbol)) )

sd.terms <- biodom %>% 
  mutate( Subdomain = if_else(is.na(Subdomain), 'none', Subdomain) ) %>% 
  select(Biodomain, abbr, Subdomain, subdomain_idx) %>% distinct() %>% 
  rowwise() %>% 
  mutate( 
    set = paste0(abbr,'_',Subdomain),
    bd = Biodomain, sdi = subdomain_idx,
    terms = biodom %>% filter(Biodomain == bd, subdomain_idx == sdi) %>% pull(GO_ID) %>% list(),
    genes = biodom %>% filter(Biodomain == bd, subdomain_idx == sdi) %>% pull(symbol) %>% list()
    ) 

# combine
bd.term.list = bind_rows(bd.terms, sd.terms %>% select(set, terms, genes)) %>% 
  filter(set != 'NA_none')

rm(bd.terms, sd.terms)

##
# MOUSE
##

# curate biodomain and sub-domain gene lists
bd.terms <- tibble( set = unique(mm.biodom$Biodomain) ) %>% 
  mutate( terms = map(set, ~ mm.biodom %>% filter(Biodomain == .x) %>% pull(GO_ID)),
          genes = map(set, ~ mm.biodom %>% filter(Biodomain == .x) %>% pull(symbol)) )

sd.terms <- mm.biodom %>% 
  mutate( Subdomain = if_else(is.na(Subdomain), 'none', Subdomain) ) %>% 
  select(Biodomain, abbr, Subdomain, subdomain_idx) %>% distinct() %>% 
  rowwise() %>% 
  mutate( 
    set = paste0(abbr,'_',Subdomain),
    bd = Biodomain, sdi = subdomain_idx,
    terms = mm.biodom %>% filter(Biodomain == bd, subdomain_idx == sdi) %>% pull(GO_ID) %>% list(),
    genes = mm.biodom %>% filter(Biodomain == bd, subdomain_idx == sdi) %>% pull(symbol) %>% list()
    ) 

# combine
mm.bd.term.list = bind_rows(bd.terms, sd.terms %>% select(set, terms, genes)) %>% 
  filter(set != 'NA_none')

rm(bd.terms, sd.terms)
```

biodom gene lists
```{r}
# curate biodomain and sub-domain gene lists
bd.genes <- tibble( set = unique(biodom$Biodomain) ) %>%
  mutate( genes = map(set, ~ biodom %>% filter(Biodomain == .x) %>% pull(symbol) %>% unlist) )

sd.genes <- biodom %>%
  mutate( Subdomain = if_else(is.na(Subdomain), 'none', Subdomain) ) %>%
  select(Biodomain, abbr, Subdomain, subdomain_idx) %>% distinct() %>%
  rowwise() %>%
  mutate(
    set = paste0(abbr,'_',Subdomain),
    bd = Biodomain, sdi = subdomain_idx,
    genes = biodom %>% filter(Biodomain == bd, subdomain_idx == sdi) %>% pull(symbol) %>% unlist %>% list()
    )

# combine
bd.gene.lists = bind_rows(bd.genes, sd.genes %>% select(set, genes)) %>%
  filter(set != 'NA_none')

rm(bd.genes, sd.genes)

# join with expression data
bd.genes <- omics %>% 
  select(symbol = GName, ENSG, RNA_TE, Pro_TE) %>% 
  filter(!is.na(symbol)) %>% distinct() %>% 
  left_join(bd.gene.lists %>% unnest(genes) %>% rename(symbol = genes),.) %>% 
  filter(!is.na(Pro_TE))

```

AMP-AD nominated target list
```{r}
agora <- syn$get('syn12540368') %>% .$path %>% read_csv(., quote = '"', col_types = cols())
```

SEA-AD single cell data
```{r}
syn$login()
# seaad <- syn$get('syn51658024') %>% .$path %>% read_csv(.,guess_max = 5e4)
# seaad <- syn$get('syn51658025') %>% .$path %>% read_csv(.,guess_max = 5e4)

# effect size table syn63885303
# cluster order and colors syn63885282
seaad.cps <- syn$get('syn63885303') %>% .$path %>% read_csv() %>% 
  rename(taxonomy = `Taxonomy Level`, 
         all.cps = `Effect size across all of pseudoprogression`,
         early.cps = `Effect size across early pseudoprogression`,
         late.cps = `Effect size across late pseudoprogression`,
         mean.expr = `Mean expression (natural log UMIs per 10k plus 1)`)
```

BioHyV targets
```{r}
syn$login()

biohyv_targets = read_tsv( syn$get('syn65987195') %>% .$path, show_col_types = F) %>% 
  mutate(
    hyp_area = case_when(grepl('immune',hypothesis) ~ 'Immune Response targets',
                         grepl('mito', hypothesis) ~ 'Mitochondrial Metabolism targets')) %>% 
  select(target = targets, hyp_area) %>% 
  group_by(target) %>% 
  summarise(hyp_area = paste0(unique(hyp_area), collapse = ' | ')) %>% 
  distinct()
```

# Fig 1 - Psen2 vs WT BV2 microglia

Read in DE analysis results
```{r}
psen.de <- read_tsv(here::here('manuscript','results', 'de_prot_results.tsv')) %>% 
  filter(tg == 'siRNA control', cell.2 == 'psen2_kd') %>% 
  mutate(n.pep = if_else(is.na(n.pep), 1, n.pep), 
         p.adj = if_else(is.na(p.adj), 1, p.adj))
```

## volcano
Cell line comparison -- Psen2 KD vs WT (scramble)
```{r fig.width=6, fig.height=4}
tmp <- psen.de

p <- ggplot(tmp, aes(diff, -log10(p.adj)))+ 
  geom_point(alpha = .5, aes(size = n.pep, color = p.adj <= 0.05 )) +
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(diff)), p.adj ), p.adj <= 0.05 & n.pep > 4),
    aes(label = gene),
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    size = 2, alpha = .8, min.segment.length = 0, max.overlaps = 25,
    max.time = 1, max.iter = 1e5)+
  labs(
    x = 'logFC',
    subtitle = str_c( unique(tmp$cell.2), ' vs ',  unique(tmp$cell.1) )
  )+
  scale_color_manual('significant', values = c('grey', 'black'))+
  scale_size_continuous('N peptides', breaks = c(1,10, 100, 1000, 2000))+
  theme(
    legend.position = 'bottom'
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','psen_volcano.pdf'),
  plot = p,
  device = cairo_pdf,
  width = 6, height = 4
  )
```

## domains

```{r}
tmp <- psen.de %>% filter(tg == 'siRNA control') %>%
  mutate(gene = str_split_fixed(gene, ';', 2) %>% .[,1])

gl <- tmp %>% arrange(desc(diff)) %>% pull(diff, name = gene)

psen.gsea <- clusterProfiler::gseGO(geneList = gl, ont = 'ALL', org.Mm.eg.db::org.Mm.eg.db, 
                                    keyType = 'SYMBOL', eps = 0, pvalueCutoff = 1)
```

```{r fig.width=6, fig.height=7, warning=F}
sig = 0.05

tmp <- psen.gsea@result %>% 
  left_join(., biodom %>% select(ID = GO_ID, Biodomain:Subdomain, abbr:sd)) %>% 
  mutate(
    Biodomain = if_else(is.na(Biodomain), 'no domain', Biodomain),
    sd = if_else(is.na(sd), 'none', sd)
  ) %>% 
  mutate(
    n_sig_bd = length(unique(ID[p.adjust <= sig])),
    med_nes_bd = median(NES[p.adjust <= sig]),
    .by = c(Biodomain)
    ) %>% 
  mutate(
    n_sig_sd = length(unique(ID[p.adjust <= sig])),
    med_nes_sd = median(NES[p.adjust <= sig]),
    .by = c(sd)
    ) %>% 
  mutate(
    across(contains('med_nes'), ~if_else(is.na(.x), 0, .x) ),
    Biodomain = fct_reorder(Biodomain, n_sig_bd),
    sd = fct_reorder(sd, med_nes_sd)
  ) 
    
p <- ggplot(tmp, aes(NES, Biodomain))+
  # facet_grid(rows = vars(Biodomain), scales = 'free', space = 'free', switch = 'y')+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_jitter(
    data = subset(tmp, p.adjust > sig),
    color = 'grey80', alpha = .5, size = .2)+
  geom_jitter(
    data = subset(tmp, p.adjust <= sig),
    shape = 21, color = 'grey30', alpha = .5,
    aes(fill = color, size = -log10(p.adjust)))+
  geom_violin(
    data = subset(tmp, p.adjust <= sig & NES > 0),
    aes(fill = color), color = 'grey30', lwd = .3,
    alpha = .1, scale = 'width', draw_quantiles = .5)+  
  geom_violin(
    data = subset(tmp, p.adjust <= sig & NES < 0),
    aes(fill = color), color = 'grey30', lwd = .3,
    alpha = .1, scale = 'width', draw_quantiles = .5)+  
  scale_y_discrete(position = 'right', label = ~str_remove_all(.x, '^[A-Za-z]{2} \\| ') %>% str_wrap(., ))+
  scale_shape_manual(values = c(20,21))+
  scale_fill_identity()+
  scale_color_identity()+
  labs(y = '', subtitle = str_c('terms enriched at p.adj \u2264 ', sig))+
  theme(legend.position = 'bottom',
        strip.text.y.left = element_text(angle = 0),
        plot.title.position = 'plot'
          )+
  ggtitle('Psen2 KD vs Scramble')
  
print(p)
```

```{r}
ggsave(
  here::here('manuscript','psen_vs_wt_biodomains.pdf'),
  plot = p,
  device = cairo_pdf,
  width = 6, height = 7
  )
```

```{r fig.width=7, fig.height=7, warning=F}
sig = 0.05

tmp <- psen.gsea@result %>% 
  filter(p.adjust <= sig) %>% 
  left_join(., biodom %>% select(ID = GO_ID, Biodomain:Subdomain, abbr:sd)) %>% 
  mutate(
    Biodomain = if_else(is.na(Biodomain), 'no domain', Biodomain),
    sd = if_else(is.na(sd), 'none', sd)
  ) %>% 
  mutate(
    n_sig_bd = length(unique(ID[p.adjust <= sig])),
    med_nes_bd = median(NES[p.adjust <= sig]),
    .by = c(Biodomain)
    ) %>% 
  mutate(
    n_sig_sd = length(unique(ID[p.adjust <= sig])),
    med_nes_sd = median(NES[p.adjust <= sig]),
    .by = c(sd)
    ) %>% 
  mutate(
    across(contains('med_nes'), ~if_else(is.na(.x), 0, .x) ),
    Biodomain = fct_reorder(Biodomain, n_sig_bd),
    sd = fct_reorder(sd, med_nes_sd)
  ) %>% 
  arrange(desc(Biodomain))

ord = tmp %>% select(Biodomain, n_sig_bd) %>% distinct() %>% arrange(desc(n_sig_bd)) %>% pull(Biodomain)

tmp <- tmp %>% mutate(Biodomain = factor(Biodomain, levels = ord))
    
p <- ggplot(tmp, aes(NES, sd))+
  facet_grid(rows = vars(Biodomain), scales = 'free', space = 'free', switch = 'y')+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_jitter(
    data = subset(tmp, p.adjust > sig),
    color = 'grey80', alpha = .5, size = .2)+
  geom_jitter(
    data = subset(tmp, p.adjust <= sig),
    shape = 21, color = 'grey30', alpha = .5,
    aes(fill = color, size = -log10(p.adjust)))+
  geom_violin(
    data = subset(tmp, p.adjust <= sig & NES > 0),
    aes(fill = color), color = 'grey30', lwd = .3,
    alpha = .1, scale = 'width', draw_quantiles = .5)+  
  geom_violin(
    data = subset(tmp, p.adjust <= sig & NES < 0),
    aes(fill = color), color = 'grey30', lwd = .3,
    alpha = .1, scale = 'width', draw_quantiles = .5)+  
  scale_y_discrete(position = 'right', 
                   label = ~str_remove_all(.x, '^[A-Za-z]{2} \\| ') %>% 
                     str_wrap(., width = 40))+
  scale_shape_manual(values = c(20,21))+
  scale_fill_identity()+
  scale_color_identity()+
  labs(y = '', subtitle = str_c('terms enriched at p.adj \u2264 ', sig))+
  theme(legend.position = 'top',
        strip.text.y.left = element_text(angle = 0, size = 8),
        plot.title.position = 'plot'
          )+
  ggtitle('Psen2 KD vs Scramble')
  
print(p)
```

```{r}
ggsave(
  here::here('manuscript','psen_vs_wt_subdomains.pdf'),
  plot = p,
  device = cairo_pdf,
  width = 7, height = 7
  )
```


```{r warning=FALSE, message=FALSE}
# plan('multisession', workers = parallel::detectCores() - 2)
sig = 0.05

oe <- omics.gsea %>% filter(score == 'Pro_TE') %>% select(tg = score, anno) %>% unnest(anno) %>% 
  select(term = GO_ID, Description, amp_nes = NES, amp_padj = p.adjust) %>% distinct()
  # prot.enr@result %>% select(term = ID, amp.nes = NES, amp.padj = p.adjust) %>% distinct()

bv2.v.tad <- psen.gsea@result %>% 
  select(term = ID, Description, cell_nes = NES, cell_padj = p.adjust) %>% 
  distinct() %>% 
  full_join(., oe,by = c('term','Description')) %>% 
  mutate(
    across(contains('padj'), ~ if_else(is.na(.x), 1, .x)),
    across(contains('nes'), ~ if_else(is.na(.x), 0, .x)),
    amp_nes = if_else(amp_padj > 0.05, 0, amp_nes),
    cell_nes = if_else(cell_padj > 0.05, 0, cell_nes)
  ) %>% 
  distinct() %>% 
  mutate(tg = 'psen_vs_wt') %>% 
  nest(data = c(term, Description, cell_nes, cell_padj, amp_nes, amp_padj), .by = tg)

comp = crossing( bv2.v.tad, set = bd.term.list$set ) %>% 
  mutate(
    data = map2(data, set, ~.x %>% filter(term %in% bd.term.list$terms[[which(bd.term.list$set == .y)]])),
    n.terms = map_dbl(data, ~nrow(.x)),
    n.sig.both = map_dbl(data, ~.x %>% filter(cell_padj <= 0.05 & amp_padj <= 0.05) %>% nrow()),
    jaccard = n.sig.both/n.terms
  ) %>% 
  filter(n.terms > 2, !is.na(tg)) %>% 
  mutate(
    cor = map(data, ~ cor.test(.x[['amp_nes']], .x[['cell_nes']], method = 'kendall')),
    correlation = map_dbl(cor, ~broom::tidy(.x) %>% pull(estimate)),
    p.value = map_dbl(cor, ~broom::tidy(.x) %>% pull(p.value))
  ) %>% 
  mutate( padj = p.adjust(p.value, method = 'BH'), .by = tg ) %>% 
  mutate( significant = padj < 0.05 )

```

```{r fig.width=4, fig.height=5}
tmp <- comp %>%
  filter(!grepl('_', set)
         # , target != 'siRNA control'
         ) %>% 
  group_by(tg) %>% 
  mutate(#any_sig = any(n.sig.both > 0),
         any_sig = any(significant),
         set = factor(set, 
                      levels = unique(comp$set) %>% 
                        str_subset(., 'All Terms|No Biodomain',negate = T) %>% 
                        c('All Terms','No Biodomain',.))) %>% 
  filter( any_sig ) %>% 
  mutate(cell = if_else(tg == 'psen_vs_wt', 'psen2_kd vs scramble', NA))
  # mutate(cell = if_else(target == 'siRNA control', 'psen2 vs scramble', cell),
  #        target = fct_relevel(target, c('siRNA control')))

p <- ggplot(tmp, aes(cell, set)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, significant),
             ,color="black", shape=0, size= 7, stroke = .75) +
  # facet_grid(rows = vars(target), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 8),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'bottom', drop = T) +
  scale_y_discrete(position = 'left', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "right",
    legend.box = 'vertical'
  )

print(p)
```

```{r fig.width=7, fig.height=5}
t = 'psen_vs_wt'
# c = 'psen2_kd'
d = 'Immune Response'

tmp <- comp %>% filter(tg == t, grepl(d, set)) %>% unnest(data) %>% 
  left_join(., biodom %>% select(term = GO_ID, Biodomain, Subdomain, subdomain_idx, abbr:color) ) %>% 
  filter(Biodomain == d) %>% 
  distinct() 

p1 <- ggplot(tmp, aes(cell_nes, amp_nes))+
  geom_vline(xintercept = 0, lwd = .2)+
  geom_hline(yintercept = 0, lwd = .2)+
  geom_point( aes(fill = color), shape = 21, color = 'grey20', stroke = .5, alpha = .5, size = 2)+
  geom_smooth( method = 'lm', formula = y ~ x - 1,  lwd = .5, lty = 2)+
  facet_grid(rows = vars(Biodomain)
             # , cols = vars(cell)
             )+
  labs(y = 'AMP-AD\nNES', x = 'BV2 cell\nNES')+
  scale_fill_identity()
  # +ggtitle(t)

cont.table <- comp %>% 
  filter(tg == t, grepl(d, set)) %>%
  select(set, tg, data, correlation, p.value) %>%
  mutate(
    c = map(data, ~ .x %>% mutate(
      amp = case_when(
        amp_padj > 0.05 ~ 'NonSig',
        amp_padj <= 0.05 & amp_nes > 0 ~ 'Up',
        amp_padj <= 0.05 & amp_nes < 0 ~ 'Dn'),
      cell = case_when(
        cell_padj > 0.05 ~ 'NonSig',
        cell_padj <= 0.05 & cell_nes > 0 ~ 'Up',
        cell_padj <= 0.05 & cell_nes < 0 ~ 'Dn') )),
    d = crossing(
      amp.s = c('Up', 'Dn', 'NonSig'),
      cell.s = c('Up', 'Dn', 'NonSig') ) %>% list(),
    d = map2(d, c, 
             ~ .x %>% rowwise() %>% mutate(
                 n = .y %>% filter(amp == amp.s, cell == cell.s) %>% nrow()
               ))) %>% 
  unnest(d)

p2 = ggplot(cont.table, aes(cell.s, amp.s, fill = n)) +
  geom_tile()+ 
  geom_text(aes(label = n))+
  scale_fill_gradient(low = "white", high = "blue", guide = 'none') +
  facet_grid(rows = vars(set))+ #, cols = vars('psen2_kd vs scramble')
  labs(x = 'BV2 cell\n# terms', 
       y = 'AMP-AD\n# terms')+
  theme_bw()

p3 = cowplot::plot_grid(p1,p2, ncol = 1, align = 'v', axis = 'lr' )
p4 = cowplot::plot_grid(p, p3, ncol = 2, align = 'h', axis='t',labels = c('C','D'))
print(p4)
```

```{r}
ggsave(
  here::here('manuscript','psen_vs_wt_domain_corr.pdf'),
  plot = p4,
  device = cairo_pdf,
  width = 7, height = 5
  )
```

## microglial modules & subtypes

```{r}
cell.subsets <- readRDS(here::here('data', 'cellSubsets_DAM_etc.rds'))
```

```{r}
gl = psen.de %>% 
  filter(tg == 'siRNA control'
         # , n.pep > 4
         ) %>%
  arrange(desc(diff)) %>% 
  pull(diff, name = gene)

# gl = de.results %>% 
#   filter(tg == 'siRNA control', n.pep > 1) %>%
#   mutate(signP = -log10(p.adj)*sign(diff)) %>%
#   filter(is.finite(signP), !is.na(signP)) %>%
#   arrange(desc(signP)) %>% pull(signP, name = gene)

# gl = log.fc %>%
#   left_join(., n.pep, by = c('gene'='Genes')) %>% 
#   mutate(gene = str_split_fixed(gene, ';', 2) %>% .[,1]) %>% 
#   filter(!is.na(`siRNA control.psen2_kd`)
#          # , !is.na(n.pep), n.pep > 1
#          ) %>%
#   arrange(desc(`siRNA control.psen2_kd`)) %>%
#   pull(`siRNA control.psen2_kd`, name = gene)

ct.gsea = fgsea::fgseaMultilevel(
  cell.subsets %>% pull(mm.gene, name = set),
  gl, eps = 0 )
```

```{r fig.width=7, fig.height=5}
tmp <- ct.gsea %>% 
  # filter(grepl(';|amit|levey', pathway)) %>%
  mutate(
    stat = -log10(pval)*sign(NES),
    grp = case_when(
      # grepl(';|daa|cluster',pathway)==F ~ 'subtypes',
      grepl('amit',pathway)==T ~ 'Keren-Shaul .. Amit\nmicroglia subtypes',
      grepl('colona',pathway)==T ~ 'Zhou .. Colonna\nmicroglia subtypes',
      grepl('levey',pathway)==T ~ 'Rangaraju .. Levey\nmicroglia subtypes',
      grepl('daa',pathway)==T ~ 'Schwartz astrocyte subtypes',
      grepl('cluster',pathway)==T ~ 'Jayadev microglial subtypes',
      grepl(':',pathway)==T ~ 'Lloyd .. Howden\nproteomic modules'
    ),
    # pathway = str_split_fixed(pathway, '; ', 2) %>% .[,2],
    # pathway = str_remove_all(pathway, '^[a-z60]+; '),
    pathway = fct_reorder(pathway, NES),
    sig = padj <= .05) %>% 
  filter(!grepl('astrocyte|Jayadev', grp)) %>% 
  mutate(grp = fct_relevel(grp, c('Lloyd .. Howden\nproteomic modules', 
                                  'Rangaraju .. Levey\nmicroglia subtypes', 
                                  'Keren-Shaul .. Amit\nmicroglia subtypes',
                                  'Zhou .. Colonna\nmicroglia subtypes'
                                  )))

p <- ggplot(data = tmp, #subset(tmp, !is.na(grp) ), 
       aes(NES, pathway))+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_point(aes(size = size, fill = -log10(padj), shape = sig), alpha = .7)+
  facet_grid(rows = vars(grp), scales = 'free_y', space = 'free', switch = 'y')+
  viridis::scale_fill_viridis()+
  scale_y_discrete(position = 'right')+
  scale_shape_manual('adj p â‰¤ 0.05', values = c(21, 24))+
  labs(y = '')+
  theme(
    strip.text.y.left = element_text(angle = 0),
    legend.position = 'left'
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','psen_uglia_module_subtypes.pdf'),
  plot = p,
  device = cairo_pdf,
  width = 7, height = 5
  )
```

# Fig 2 - Target selection and KD validation

## nominated tg DE in AD brain

```{r fig.width=4.5, fig.height=5}
tmp1 <- omics %>% 
  ungroup() %>% 
  filter(GName %in% biohyv_targets$target) %>% 
  left_join(., biohyv_targets, by = c('GName'='target')) %>% 
  mutate(
    agora = GName %in% agora$hgnc_symbol[agora$Source == 'AMP-AD'],
    label_color = ifelse(agora, 
                         glue::glue("<span style='color:{'darkred'}'>{GName}</span>"), 
                         glue::glue("<span style='color:{'black'}'>{GName}</span>")))

# d1 <- tmp1 %>% 
#   select(gene = GName, RNA_TE, Pro_TE) %>% 
#   column_to_rownames(var = 'gene') %>% 
#   as.matrix() %>% 
#   dist()
# 
# hc <- hclust(d1, method = 'ward.D2')

tmp1 <- tmp1 %>% 
  mutate(hyp_area = str_split(hyp_area, ' \\| ')) %>% 
  unnest_longer(hyp_area) %>% 
  filter(hyp_area!='synapse') %>% 
  pivot_longer(c(RNA_TE, Pro_TE)) %>% 
  mutate(
    # GName = factor(GName, levels = tmp1$GName[hc$order]),
    hyp_area = str_replace_all(hyp_area, ' targets', '\ntargets'),
    sig = if_else(name == 'RNA_TE', RNA_Sig, Pro_Sig), 
    name = factor(name, c('RNA_TE','Pro_TE')),
    )

p <- ggplot(tmp1,aes(name, label_color))+
  geom_tile(aes(fill = value), color = 'grey20')+
  geom_point(data = subset(tmp1, sig == 'YES'), shape = 8, size = 1)+
  scale_fill_gradient2(
    'meta-analysis LogFC',
    low = 'purple',mid = 'grey20',high = 'green', na.value = 'white')+
  scale_y_discrete(limit = rev)+
  facet_wrap(~hyp_area, scales = 'free_y')+
  labs(x = '', y = '')+
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    # axis.text.y = element_text(color = tmp1$label_color),
    axis.text.y = ggtext::element_markdown(),
    legend.position = 'top', legend.justification = 0)

print(p)
```

```{r}
ggsave(
  here::here('manuscript','de_tg_meta-analysis.pdf'),
  plot = p,
  device = cairo_pdf,
  width = 4.5, height = 5
  )
```

## nominated tg in cell types

```{r fig.width=3, fig.height=5.5}
tmp <- seaad.cps %>% 
  filter(
    Gene %in% biohyv_targets$target,
    Population %in% c('Neuronal: Glutamatergic', 'Neuronal: GABAergic',
                      'Astrocyte', 'Microglia-PVM')
  ) %>% 
  mutate(Population = str_remove_all(Population,'Neuronal: |-PVM'),
         Population = fct_relevel(Population, c('Glutamatergic','GABAergic','Astrocyte','Microglia'))) %>%
  left_join(., biohyv_targets %>% 
              mutate(hyp_area = str_split(hyp_area, ' \\| ')) %>%
              unnest(hyp_area),
            by = c('Gene'='target')) %>% 
  filter(hyp_area != 'synapse')

p <- ggplot(tmp,aes(Population, Gene))+
  geom_tile(fill = 'white')+
  geom_point(shape = 21, alpha = .7, aes(size = mean.expr, fill = all.cps) )+
  scale_fill_gradient2('SEA-AD\nLogFC', midpoint = 0
                       , low = 'purple',mid = 'grey80',high = 'green', 
                       na.value = 'white')+
  scale_y_discrete(position = 'right')+
  facet_grid(rows = vars(hyp_area), scales = 'free', space = 'free', switch = 'y')+
  labs(x = '', y = '')+
  theme(legend.position = 'top', legend.justification = 0)+
  theme(
    legend.position = 'left',
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))

print(p)
```

```{r}
ggsave(
  here::here('manuscript','tg_seaad_cell_data.pdf'),
  plot = p,
  device = cairo_pdf,
  width = 3, height = 5.5
  )
```

## nominated tg in GEO

```{r fig.height=6, fig.width=4}
geo.expr <- readRDS(here::here('data','bv2_shsy5y_cell_expr_data_from_geo.rds'))

tmp1 <- geo.expr %>% 
  unnest(data) %>% 
  filter(gene %in% biohyv_targets$target) %>% 
  group_by(gene, cell, acc) %>%
  summarise(mean = mean(mn, na.rm = T)) %>%
  mutate(mean = if_else(!is.finite(mean), 0, mean)) %>%
  left_join(., biohyv_targets, by = c('gene'='target')) 

tmp2 <- tmp1 %>% pivot_wider(names_from = 'cell', values_from = 'mean')

# d1 <- tmp2 %>% 
#   column_to_rownames(var = 'gene') %>% 
#   as.matrix() %>% 
#   dist()
# 
# hc <- hclust(d1, method = 'ward.D2')

p <- tmp1 %>% 
  mutate(hyp_area = str_split(hyp_area, ' \\| ')) %>% 
  unnest_longer(hyp_area) %>% 
  filter(hyp_area != 'synapse') %>% 
  # mutate(gene = factor(gene, levels = tmp2$gene[hc$order])) %>% 
  # mutate(lbl = str_c(acc,': ', group)) %>% 
  mutate(lbl = acc) %>% 
  ggplot(aes(lbl, gene))+
  geom_tile(aes(fill = mean), color = 'grey20')+
  viridis::scale_fill_viridis('mean\nlog2\nexpr')+
  scale_y_discrete(position = 'right', limits = rev)+
  facet_grid(rows = vars(hyp_area), cols = vars(cell), 
             scales = 'free', space = 'free', switch = 'y')+
  labs(x = '', y = '')+
  theme(legend.position = 'right', legend.justification = .5,
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))

print(p)
```

```{r}
ggsave(
  here::here('manuscript','tg_geo_expr.pdf'),
  plot = p,
  device = cairo_pdf,
  width = 4, height = 6
  )
```

## knockdown validation

Read data from Synapse and format.  
```{r}
syn$login()

synid <- c('syn51442596', 'syn51442597')

# normal expression
tg = readxl::read_xlsx( syn$get(synid[1]) %>% .$path, range = 'A3:A30', col_names = 'target' )
normal_expr = readxl::read_xlsx( syn$get(synid[1]) %>% .$path, 
                       range = 'A3:D30',
                       col_names = c('target','rep1', 'rep2','rep3')
                       ) %>% 
      mutate(line = 'scramble'
             , value = 'avg rel pct GAPDH'
             , note = 'biol replicates, passage 4, 5, and 6'
             )

normal_expr = readxl::read_xlsx( syn$get(synid[1]) %>% .$path, 
                       range = 'E3:G30', 
                       col_names = c('rep1', 'rep2','rep3')) %>% 
      mutate(line = 'psen2_kd'
             , value = 'avg rel pct GAPDH'
             , note = 'biol replicates, passage 4, 5, and 6'
             ) %>% 
  bind_cols(target = tg$target, .) %>% 
  bind_rows(normal_expr, .) 

normal_expr = normal_expr %>% 
  left_join(., biohyv_targets , by = c('target')) %>% 
  mutate( hyp_area = if_else(target == 'Control', 'mito | immune', hyp_area))

# siRNA expression
tg = readxl::read_xlsx( syn$get(synid[2]) %>% .$path, range = 'A3:A30', col_names = 'target')
si_expr = readxl::read_xlsx( syn$get(synid[2]) %>% .$path, 
                       range = 'A3:D30',
                       col_names = c('target','rep1', 'rep2','rep3')
                       ) %>% 
      mutate(line = 'scramble',
             si = 'control siRNA',
             , value = 'avg rel pct GAPDH'
             , note = 'tech replicates'
             ) 

si_expr = readxl::read_xlsx( syn$get(synid[2]) %>% .$path, 
                       range = 'E3:G30', 
                       col_names = c('rep1', 'rep2','rep3')) %>% 
      mutate(line = 'psen2_kd',
             si = 'control siRNA',
             , value = 'avg rel pct GAPDH'
             , note = 'tech replicates'
             ) %>% 
  bind_cols(target = tg$target, .) %>% 
  bind_rows(si_expr, .) 

si_expr = readxl::read_xlsx( syn$get(synid[2]) %>% .$path, 
                       range = 'H3:J30', 
                       col_names = c('rep1', 'rep2','rep3')) %>% 
      mutate(line = 'scramble',
             si = 'target siRNA',
             , value = 'avg rel pct GAPDH'
             , note = 'tech replicates'
             ) %>% 
  bind_cols(target = tg$target, .) %>% 
  bind_rows(si_expr, .) 

si_expr = readxl::read_xlsx( syn$get(synid[2]) %>% .$path, 
                       range = 'K3:M30', 
                       col_names = c('rep1', 'rep2','rep3')) %>% 
      mutate(line = 'psen2_kd',
             si = 'target siRNA',
             , value = 'avg rel pct GAPDH'
             , note = 'tech replicates'
             ) %>% 
  bind_cols(target = tg$target, .) %>% 
  bind_rows(si_expr, .) 

si_expr = si_expr %>% 
  left_join(., biohyv_targets , by = c('target')) %>% 
  mutate( hyp_area = if_else(target == 'Control', 'mito | immune', hyp_area))
```

```{r}
vars <- si_expr %>% select(starts_with('rep')) %>% names()

si_expr <- si_expr %>%
  mutate( avg = map_dbl( 1:nrow(si_expr), 
                         ~ si_expr[.x,] %>% select(all_of(vars)) %>% t() %>% mean(na.rm = T)),
          sd = map_dbl( 1:nrow(si_expr), 
                         ~ si_expr[.x,] %>% select(all_of(vars)) %>% t() %>% sd(na.rm = T)),
          upper = avg + sd,
          lower = avg - sd,
          target = factor( target, levels = unique(si_expr$target) )
  ) 
```

```{r fig.width=10, fig.height=5}
x= si_expr %>%
    group_by(target) %>%
    mutate(fc = avg / avg[line == 'scramble' & si == 'control siRNA'],
           lfc = log2(fc),
           lower_fc = lower / lower[line == 'scramble' & si == 'control siRNA'],
           log_lower = log2(lower_fc),
           upper_fc = upper / upper[line == 'scramble' & si == 'control siRNA'],
           log_upper = log2(upper_fc),
           set = paste0(line, ', ', si),
           set = factor(set, levels = c('scramble, control siRNA','psen2_kd, control siRNA',
                                    'scramble, target siRNA','psen2_kd, target siRNA'))
           ) %>% 
  arrange(set)

ggplot(x, aes( target, fc, fill = set)) + 
  geom_hline(yintercept = 1, lty = 2, lwd = .5)+
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6, width = .8) +
  geom_errorbar(aes(ymin = lower_fc, ymax = upper_fc), 
                position = position_dodge(width =.9),
                lwd = .4, width = .4)+
  labs(x = '', y = 'Relative Expression')+
  theme_bw()+
  theme(legend.position = 'top') +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+
  viridis::scale_fill_viridis('', discrete = T, option = 'D')+
  ggbreak::scale_y_break(c(2.5,6), scales = 0.25)+
  ggbreak::scale_y_break(c(7.5,35), scales = 0.25)
```

```{r fig.width=10, fig.height=5}
y = x %>% 
  mutate(
    fc = if_else(si == 'target siRNA', NA_real_, fc),
    lower_fc = if_else(si == 'target siRNA', NA_real_, lower_fc),
    upper_fc = if_else(si == 'target siRNA', NA_real_, upper_fc)
  )

ggplot(y, aes( target, fc, fill = set)) + 
  geom_hline(yintercept = 1, lty = 2, lwd = .5)+
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6, width = .8) +
  theme(legend.position = 'top') +
  geom_errorbar(aes(ymin = lower_fc, ymax = upper_fc), 
                position = position_dodge(width =.9),
                lwd = .4, width = .4)+
  labs(x = '', y = 'Relative Expression')+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+
  viridis::scale_fill_viridis('', discrete = T, option = 'D')+
  ggbreak::scale_y_break(c(2.5,6), scales = 0.25)+
  ggbreak::scale_y_break(c(7.5,35), scales = 0.25)
```

```{r fig.width=4, fig.height=7}
tmp <- si_expr %>%
  group_by(target) %>%
  mutate(
   set = paste0(line, ', ', si),
   set = factor(set, levels = c('scramble, control siRNA','psen2_kd, control siRNA',
                            'scramble, target siRNA','psen2_kd, target siRNA')),
   control.siRNA = avg[set == 'psen2_kd, control siRNA'] / avg[set == 'scramble, control siRNA'],
   siRNA.WT = avg[set == 'scramble, target siRNA'] / avg[set == 'scramble, control siRNA'],
   siRNA.psen2_kd = avg[set == 'psen2_kd, target siRNA'] / avg[set == 'psen2_kd, control siRNA'],
   across(contains('siRNA'), ~ log2(.x))
   ) %>% 
  pivot_longer(cols = c(control.siRNA, siRNA.WT, siRNA.psen2_kd),
               names_to = 'comp', values_to = 'lfc') %>% 
  mutate(
    cont = case_when(comp == 'control.siRNA' ~ 'psen2_kd\nvs\nscramble', 
                     comp %in% c('siRNA.WT','siRNA.psen2_kd') ~ 'siRNA\nvs\ncontrol'),
    avg_kd = mean(lfc[cont == 'siRNA\nvs\ncontrol']), 
    avg_fc = 2^avg_kd,
    avg_pct = if_else( avg_kd < 0, (1-avg_fc)*100*sign(avg_kd) , (avg_fc-1)*100*sign(avg_kd) ),
    lbl = str_c(target, '   [',signif(avg_pct,2),'%]'),
    hyp_area = str_split(hyp_area, ' \\| ')
    ) %>% 
  unnest_longer(hyp_area) %>% 
  filter(hyp_area != 'synapse') %>% 
  ungroup() %>% 
  mutate(lbl = fct_reorder(lbl, desc(avg_pct))) %>% 
  arrange(lbl) 

p <- ggplot(tmp, aes(comp, lbl) )+
  geom_tile(aes(fill = lfc))+
  scale_fill_gradient2(
    'log2 Fold Change',
    low = 'purple',mid = 'grey20',high = 'green', na.value = 'white')+
  facet_grid(cols = vars(cont), rows = vars(hyp_area), 
             scales = 'free', space = 'free', switch = 'y')+
  scale_y_discrete(position = 'right')+
  labs(x= '', y = '')+
  theme(
    legend.position = 'bottom',
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))

print(p)
```

```{r}
ggsave(
  here::here('manuscript','knockdown_validation.pdf'),
  plot = p,
  device = cairo_pdf,
  width=4, height=7
  )
```

# Fig 3 - Cellular phenotypes

Cell phenotype data
```{r}
load(here::here('manuscript','data','bv2_assay_data.rdata'))
```


## phagocytosis

```{r fig.width=10, fig.height=10 }
# plot raw data
x = phagocytosis_lmm %>% 
  # select(-value) %>% 
  pivot_longer(., starts_with('rep')) %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ),
          batch_num = as.factor(as.numeric(factor(batch))))

p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
  # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .6, show.legend = F)+
  geom_point( aes(fill = BV2.cell.line, shape = batch_num), 
              stroke =.5, alpha = .8, color = 'grey30',
              position = position_jitterdodge(), show.legend = T )+
  scale_y_discrete(limits = rev) +
  scale_x_continuous(trans = 'log10')+
  scale_shape_manual('batch', values = c(21,22,23,24,25))+
  guides(fill = F)+
  # facet_wrap(~hyp_area, scale = 'free_y')+
  theme(legend.position = 'bottom') +
  labs(y = 'siRNA target', x = '% positive cells\n ')

x = phagocytosis_lmm %>% 
  select(siRNA, BV2.cell.line, lfc = estimate, 
         log_lower = conf.low, log_upper = conf.high, p.value) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ) )

p2 <- ggplot(x, aes( lfc, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'right') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = log_lower, xmax = log_upper), 
                position = position_dodge(width =.9),
                lwd = .3, width = .4)+
  labs(y = '', 
       # x = '% positive cells\nlog2 fold change, siRNA vs control'
       x = 'linear mixed model effect\nsiRNA vs control'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow =1 , rel_widths = c(.75,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```


```{r}
ggsave(
  here::here('manuscript','phagocytosis_assay_raw_processed.pdf'),
  plot = p,
  device = cairo_pdf,
  width=10, height=10
  )
```

## NFkB

```{r fig.width=10, fig.height=16 }
# plot raw data
x = nfkb_lmm %>% 
  # select(-value) %>% 
  pivot_longer(., starts_with('rep')) %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ),
          batch_num = as.factor(as.numeric(factor(batch))))

p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
  # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .6, show.legend = F)+
  geom_point( aes(fill = BV2.cell.line, shape = batch_num), 
              stroke =.5, alpha = .8, color = 'grey30',
              position = position_jitterdodge(), show.legend = T )+
  scale_y_discrete(limits = rev) +
  scale_x_continuous(trans = 'log10')+
  scale_shape_manual('batch', values = c(21,22,23,24,25))+
  guides(fill = F)+
  # facet_grid(rows = vars(str_c('LPS @ ',LPS.dose)), scales = 'free_x', space = 'free_x')+
  facet_wrap( ~ str_c('LPS @ ',LPS.dose), scales = 'free_x', strip.position = "right", nrow = 2)+
  theme(legend.position = 'bottom', legend.box = 'vertical') +
  labs(y = 'siRNA target',  x = 'NFkB reporter luminescence, log10\n')

x = nfkb_lmm %>% 
  select(siRNA, BV2.cell.line, LPS.dose, tx, lfc = estimate, 
         log_lower = conf.low, log_upper = conf.high, p.value) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ) )

p2 <- ggplot(x, aes( lfc, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'right') +
  facet_grid(rows = vars(str_c('LPS @ ',LPS.dose)))+
  geom_errorbar(aes(xmin = log_lower, xmax = log_upper), 
                position = position_dodge(width =.9),
                lwd = .3, width = .4)+
  labs(y = '', 
       # x = 'NFkB reporter fold change\nlog2 fold change, siRNA vs control'
       x = 'linear mixed model effect\nsiRNA vs control'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow =1 , rel_widths = c(.75,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','nfkb_assay_raw_processed.pdf'),
  plot = p,
  device = cairo_pdf,
  width=10, height=20
  )
```

## Mitotracker

```{r fig.width=10, fig.height=12 }
# plot raw data
x = mitotracker_lmm %>% 
  select(-value) %>%
  pivot_longer(., starts_with('rep')) %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ),
          batch_num = as.factor(as.numeric(factor(batch))))

p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
  # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .6, show.legend = F)+
  geom_point( aes(fill = BV2.cell.line, shape = batch_num), 
              stroke =.5, alpha = .8, color = 'grey30',
              position = position_jitterdodge(), show.legend = T )+
  scale_y_discrete(limits = rev) +
  scale_x_continuous(trans = 'log10')+
  scale_shape_manual('batch', values = c(21,22,23,24,25, 3,4))+
  guides(fill = F)+
  theme(legend.position = 'bottom', legend.box = 'vertical') +
  labs(y = 'siRNA target', x = 'MitoTracker avg intensity / Cell Number\n')

x = mitotracker_lmm %>% 
  select(siRNA, BV2.cell.line, lfc = estimate, 
         log_lower = conf.low, log_upper = conf.high, p.value) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ) )

p2 <- ggplot(x, aes( lfc, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'right') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = log_lower, xmax = log_upper), 
                position = position_dodge(width =.9),
                lwd = .3, width = .4)+
  labs(y = '', 
       # x = 'normalized intensity\nlog2 fold change, siRNA vs control'
       x = 'linear mixed model effect\nsiRNA vs control'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow =1 , rel_widths = c(.75,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','mitotracker_assay_raw_processed.pdf'),
  plot = p,
  device = cairo_pdf,
  width=10, height=12
  )
```

## AlamarBlue

```{r fig.width=10, fig.height=10 }
# plot raw data
x = ablue_lmm %>% 
  select(-value) %>%
  pivot_longer(., starts_with('rep')) %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ),
          batch_num = as.factor(as.numeric(factor(batch))))

p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
  # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .6, show.legend = F)+
  geom_point( aes(fill = BV2.cell.line, shape = batch_num), 
              stroke =.5, alpha = .8, color = 'grey30',
              position = position_jitterdodge(), show.legend = T )+
  scale_y_discrete(limits = rev) +
  scale_x_continuous(trans = 'log10')+
  scale_shape_manual('batch', values = c(21,22,23,24,25))+
  guides(fill = F)+
  # facet_wrap(~hyp_area, scale = 'free_y')+
  theme(legend.position = 'bottom') +
  labs(y = 'siRNA target', x = 'alamarBlue fluorescence signal (560Ex/590Em)')

x = ablue_lmm %>% 
  select(siRNA, BV2.cell.line, lfc = estimate, 
         log_lower = conf.low, log_upper = conf.high, p.value) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(norm_ph$siRNA) ) )

p2 <- ggplot(x, aes( lfc, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'right') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = log_lower, xmax = log_upper), 
                position = position_dodge(width =.9),
                lwd = .3, width = .4)+
  labs(y = '', 
       # x = '% positive cells\nlog2 fold change, siRNA vs control'
       x = 'linear mixed model effect\nsiRNA vs control'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow =1 , rel_widths = c(.75,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','alamarblue_assay_raw_processed.pdf'),
  plot = p,
  device = cairo_pdf,
  width=10, height=10
  )
```

## combined
```{r fig.width=9, fig.height=8}
tmp <- biohyv.lmm %>% 
  filter(pheno != 'caspase') %>%
  left_join(., biohyv_targets) %>% 
  mutate(hyp_area = str_split(hyp_area, ' \\| ')) %>% 
  unnest(cols = c(hyp_area)) %>% 
  mutate(
    target = str_to_sentence(target),
    line = factor(line, c('scramble','psen2_kd')),
    pheno = case_when(pheno == 'nfkb_LPS' ~ 'NFkB +LPS',
                      pheno == 'nfkb_none' ~ 'NFkB -LPS',
                      pheno == 'alamar_blue' ~ 'alamarBlue',
                      pheno == 'mitotracker' ~ 'MitoTracker',
                      T ~ pheno),
    pheno = factor(pheno, c('alamarBlue','MitoTracker',
                            'NFkB -LPS', 'NFkB +LPS', 'caspase', 'phagocytosis') )
  ) 

p <- ggplot(tmp, aes( line,target )) + 
  geom_point( aes(fill = lfc, size = -log10(p.adj.BH), shape = hit_class
                  , color = hit_class))+ #, alpha = hit_class
  facet_grid(cols = vars(pheno), rows = vars(hyp_area), scales = 'free', space = 'free')+
  theme( axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1) )+
  guides( alpha = 'none',  color = 'none', 
          size = guide_legend( override.aes=list(pch = 22)) ) +
  scale_y_discrete(limits = rev)+ 
  scale_size_continuous('-log10(p-value)',guide = 'none', range = c(1,5))+ 
  scale_color_manual(values = c('grey20','grey50','grey20'))+
  scale_alpha_manual(values = c(1,1,1))+
  scale_shape_manual('Hit Class', values = c(24,22,25), 
                     breaks = c('Positive Hit','Non-Hit','Negative Hit') )+
  scale_fill_gradient2('Effect Size\nsiRNA vs Control', 
                       low = 'purple', mid = 'white', high = 'green', midpoint = 0)+
  labs(x = '', y = '' 
       # , title = 'BV2 siRNA phenotypes'
       # , subtitle = 'significance assessed by Wilcoxon rank sum test'
  )
print(p)
```

```{r}
ggsave(
  here::here('manuscript','cellular_assays_combined.pdf'),
  plot = p,
  device = cairo_pdf,
  # width=12, height=6
  width=9, height=8
  )
```

# Fig 4 - Proteomics data

Proteomics data
```{r}
syn$login()

# proteomics data ---------------------------------------------------------

# metadata synID
synid <- 'syn65986467'

# metadata
m.files = readxl::read_xlsx(syn$get(synid) %>% .$path,sheet = 2) %>% 
  mutate(sample = str_remove(SampleName, '-[1-9]$'),
         rep = str_extract(SampleName, '[1-9]$'),
         sample = if_else(sample == 'KD-sicon', 'KD-siCon', sample)
  ) %>% 
  mutate(name = str_remove(Name, '^ranjita_dia__') %>% 
           str_remove(., '.d.rar$'))

m.samples = 
  bind_rows(
    readxl::read_xlsx(
      syn$get(synid) %>% .$path,
      sheet = 1,
      range = 'L2:N32', 
      col_names = c('sample', 'target','passage_n')) %>% 
      mutate(cell = 'scramble'),
    readxl::read_xlsx(
      syn$get(synid) %>% .$path,
      sheet = 1,
      range = 'P2:R32', 
      col_names = c('sample', 'target','passage_n')) %>% 
      mutate(cell = 'psen2_kd')
  ) 

raw.m = left_join(m.files, m.samples, by = 'sample') 

# DIA-NN normalized data
raw.d <- read_tsv( syn$get('syn65986474') %>% .$path )

n.pep <- read_tsv( syn$get('syn65986477') %>% .$path ) %>% 
  summarise(n.pep = length(unique(Precursor.Id)), .by = c(Genes)) 


# align data & metadata  --------------------------------------------------

d <- raw.d %>% 
  select(-c( Knockdown_0_Scrambled_1, KD_SCR_Gene,Group) ) %>% 
  column_to_rownames(var = 'Sample.Name') %>% 
  mutate(across(everything(), ~ if_else( .x == 0, NA_real_, .x))) %>% 
  as.matrix() %>% t()

dups <- raw.m$Name[which(duplicated(raw.m$name))]

m <- raw.m[match( colnames(d), raw.m$name ),] 

# # filtering & QC ----------------------------------------------------------
# 
# # samples to remove
# gis_n <- which(m$SampleName == 'GIS')
# outlier_n <- which(m$n_prot < 6100)
# 
# # data table
# data <- bind_cols(m, t(d)) %>% slice(-c(gis_n,outlier_n))
# 
# # number of samples per group where protein is present
# n.na = data %>% group_by(target, cell) %>% 
#   summarise( across(Tap1:(ncol(data)-2), ~ sum( !is.na(.x) )) )
# 
# # which proteins are ID'd in at least 2 samples in at least 2 groups 
# idx = map_lgl(3:ncol(n.na), ~ n.na[,.x] %>% .[.>=2] %>% length() >= 2)
# 
# # data table
# data <- bind_cols(m, t(d[idx,])) %>% slice(-c(gis_n,outlier_n))
```

Update metadata
```{r}
s.count <- map_dbl(1:ncol(d),  ~ sum(!is.na( d[,.x] )))
p.count <- map_dbl(1:nrow(d),  ~ sum(!is.na( d[.x,] )))
s.med <- map_dbl(1:ncol(d), ~ median(d[,.x], na.rm = T))

m <- bind_cols(m, n_prot = s.count, med_expr = s.med) 
```

filter -- remove low quality samples, remove proteins not 
```{r}
# samples to remove
gis_n <- which(m$SampleName == 'GIS')
outlier_n <- which(m$n_prot < 6100)

# data table
data <- bind_cols(m, t(d)) %>% 
  slice(-c(gis_n,outlier_n))

# number of samples per group where protein is present
n.na = data %>% group_by(target, cell) %>% 
  summarise( across(Tap1:(ncol(data)-2), ~ sum( !is.na(.x) )) )

# which proteins are ID'd in at least 2 samples in at least 2 groups 
idx = map_lgl(3:ncol(n.na), ~ n.na[,.x] %>% .[.>=2] %>% length() >= 2)

# data table
data <- bind_cols(m, t(d[idx,])) %>% 
  slice(-c(gis_n,outlier_n))
```

## sample count tables
```{r}
m.tg <- data %>% 
  filter(!is.na(target), 
         str_to_upper(target) %in% biohyv_targets$target[grepl('Mito', biohyv_targets$hyp_area)]
         ) %>% 
  group_by(target, cell) %>% 
  summarise(n = length(unique(SampleName))) %>% 
  pivot_wider(names_from = 'cell', values_from = 'n')

m.tg %>% 
  kableExtra::kable("html", caption = "Mito Hypothesis Targets") %>%
  kableExtra::kable_styling(bootstrap_options = c("striped","condensed"), #,"hover"
                            full_width = FALSE)
```

```{r}
i.tg <- data %>% 
  filter(!is.na(target), 
         str_to_upper(target) %in% biohyv_targets$target[grepl('Immune', biohyv_targets$hyp_area)]
         ) %>% 
  group_by(target, cell) %>% 
  summarise(n = length(unique(SampleName))) %>% 
  pivot_wider(names_from = 'cell', values_from = 'n')
  
i.tg %>% 
  kableExtra::kable("html", caption = "Immune Hypothesis Targets") %>%
  kableExtra::kable_styling(bootstrap_options = c("striped","condensed"), #,"hover"
                            full_width = FALSE)
```

```{r}
c.tg <- data %>% 
  filter(!is.na(target), 
         target == 'siRNA control'
         ) %>% 
  group_by(target, cell) %>% 
  summarise(n = length(unique(SampleName))) %>% 
  pivot_wider(names_from = 'cell', values_from = 'n')
  
c.tg %>% 
  kableExtra::kable("html", caption = "Control Samples") %>%
  kableExtra::kable_styling(bootstrap_options = c("striped","condensed"), #,"hover"
                            full_width = FALSE)
```

## QC

```{r fig.width=10, fig.height=3}
p <- cowplot::plot_grid(
  ggplot(m, aes( log2(med_expr), n_prot ))+
    geom_hline(yintercept = 6100, lty = 2, color = 'red')+
    geom_point(alpha = .3)+
    labs(y = '# protein ID per sample',
         x = 'median log2 intensity')+
    scale_y_log10(),
  ggplot(tibble(prot_per_sample=s.count),aes(prot_per_sample))+
    geom_histogram(color = 'grey25',fill = 'grey75')+
    geom_vline(xintercept = 6100, lty = 2, color = 'red')+
    scale_x_log10()+scale_y_log10()+
    # theme(axis.text.y = element_blank())+
    labs(x='')+
    coord_flip(),
  ggplot(tibble(sample_per_prot=p.count),aes(sample_per_prot))+
    geom_histogram(color = 'grey25',fill = 'grey75')+
    labs(x = '# samples per protein')+theme_bw()+
    coord_flip(),
  nrow = 1, align = 'hv', axis = 'l',labels = 'AUTO'
)
print(p)
```

```{r}
ggsave(
  here::here('manuscript','proteomics_qc_sample_prot_missingness.pdf'),
  plot = p,
  device = cairo_pdf,
  width=10, height=3
  )
```

pca
```{r}
d.mat <- data %>% select(-c(Name:med_expr)) %>% t() %>% as.matrix()
colnames(d.mat) <- data$name

log.d.mat <- log2(d.mat+1)

prot_sd <- map_dbl(1:nrow(log.d.mat), 
                   ~ sd(log.d.mat[.x,], na.rm = T))

max.var.rows <- which(
  prot_sd > 
    sort(prot_sd, decreasing = T) %>% .[500]
)

no.na.rows <- which(
  rowSums(is.na(log.d.mat)) == 0
)

pca <- log.d.mat[no.na.rows,] %>% t() %>% prcomp()

eigenvalues <- broom::tidy(pca, matrix = 'eigenvalues') 

pc.scores <- broom::tidy(pca, matrix = 'scores') %>% 
  rename(name = row) %>% 
  left_join(., data %>% select(Name:med_expr) , by = 'name') %>% 
  pivot_wider(names_from = PC, 
              values_from = value, 
              names_prefix = 'PC')
```

```{r fig.width=8, fig.height=5}
p1 <- eigenvalues %>% 
  filter(PC <= 10) %>%
  ggplot(aes(y = factor(PC) )) +
  geom_col(aes(x = percent*100)) +
  geom_line(aes(x = cumulative*100, group = 1)) +
  geom_point(aes(x = cumulative*100)) +
  scale_y_discrete(limits = rev)+
  labs(y= "Principal Component",
       x = "% variance explained"
       # ,subtitle = str_c( 'First 10 PCs out of ', nrow(eigenvalues) ) 
  )

p2 <- ggplot( pc.scores, aes(x = PC1, y = PC2)) + theme_bw()+
  geom_vline(xintercept = 0, lty = 2, lwd = .1)+
  geom_hline(yintercept = 0, lty = 2, lwd = .1)+
  stat_ellipse(geom = "polygon", alpha = 0.2, aes(fill = cell), level = 0.99) +  # Add ellipses!
  geom_point(aes(shape = cell
                 # , color = RunOrder
                 ), color = 'grey30', size = 3, alpha = .7)+ #factor(rep)
  ggrepel::geom_label_repel(
    data = subset(pc.scores, cell == 'scramble' & PC1 > 0 | cell == 'psen2_kd' & PC1 < 0),
    aes(label = str_c(target, '_', rep)), size = 2.5, min.segment.length = 0, alpha = .8)+
  labs(
    x = str_c(
      'PC1\n', 100* eigenvalues$percent[eigenvalues$PC == 1] %>% 
        signif(digits = 3), ' % variance explained'),
    y = str_c(
      'PC2\n', 100* eigenvalues$percent[eigenvalues$PC == 2] %>% 
        signif(digits = 3), ' % variance explained')
  )

p <- cowplot::plot_grid(p1,p2, nrow =1 , rel_widths = c(.4,1), align = 'h', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','proteomics_pca.pdf'),
  plot = p,
  device = cairo_pdf,
  width=8, height=5
  )
```

```{r}
m %>% 
  mutate(ord = min(RunOrder), .by = target) %>% 
  mutate(target = fct_reorder(target, ord)) %>% 
  ggplot(., aes(RunOrder, target))+geom_point()+scale_y_discrete(limits = rev)
```

```{r fig.width=15, fig.height=5}
p3 <- ggplot( pc.scores, aes(x = PC3, y = PC5)) + theme_bw()+
  geom_vline(xintercept = 0, lty = 2, lwd = .1)+
  geom_hline(yintercept = 0, lty = 2, lwd = .1)+
  stat_ellipse(geom = "polygon", alpha = 0.2, aes(fill = target), level = 0.9) +  # Add ellipses!
  geom_point(aes(shape = cell), color = 'grey30', size = 3, alpha = .7)+ #factor(rep) , color = RunOrder
  ggrepel::geom_label_repel(
    # data = subset(pc.scores, cell == 'scramble' & PC1 > 0 | cell == 'psen2_kd' & PC1 < 0),
    aes(label = str_c(target, '_', rep)), size = 2.5, min.segment.length = 0, alpha = .7)+
  labs(
    x = str_c(
      'PC3\n', 100* eigenvalues$percent[eigenvalues$PC == 3] %>% 
        signif(digits = 3), ' % variance explained'),
    y = str_c(
      'PC5\n', 100* eigenvalues$percent[eigenvalues$PC == 5] %>% 
        signif(digits = 3), ' % variance explained')
  )

# p <- cowplot::plot_grid(p1,p2, nrow =1 , rel_widths = c(.35,1), align = 'h', labels = 'AUTO')
# print(p3)

p4 <- cowplot::plot_grid(p1,p2,p3, nrow =1 , rel_widths = c(.3,.85,1), 
                        align = 'h', labels = 'AUTO')
# print(p)
```



```{r}
ggsave(
  here::here('manuscript','proteomics_pca_all.pdf'),
  plot = p4,
  device = cairo_pdf,
  width=15, height=5
  )
```

## DE

Read in DE analysis results
```{r}
de.results <- read_tsv(here::here('manuscript','results','de_prot_results.tsv'))
```


```{r}
de.results <- readRDS(here::here('manuscript','results',"de_prot_results.rds")) %>% 
  unnest(tukey) %>% 
  filter(!is.na(comp)) %>% 
  mutate(
    diff.og = diff,
    tg.1 = str_split_fixed(comp, ':|-',4)[,1],
    cell.1 = str_split_fixed(comp, ':|-',4)[,2],
    tg.2 = str_split_fixed(comp, ':|-',4)[,3],
    cell.2 = str_split_fixed(comp, ':|-',4)[,4]
    ) %>% 
  mutate(tg = if_else(grepl('siRNA',tg.1), tg.2,tg.1),
         diff = if_else(grepl('siRNA',tg.1), -1*diff, diff)) %>% 
  relocate(tg, cell.2, gene, diff, p.adj, comp ) %>% 
  left_join(., n.pep, by = c('gene' = 'Genes'))
```

Number of DE proteins at different alpha levels
```{r fig.width=5, fig.height=5}
tmp = de.results %>% 
  filter(
    !is.na(diff), 
    (cell.1 == cell.2 & (grepl('siRNA', tg.1)|grepl('siRNA',tg.2))) 
    | (grepl('siRNA', tg.1) & grepl('siRNA',tg.2)) 
    ) %>% 
  mutate(sample = str_c(tg,'.',cell.2))

n.deg <- crossing( 
  expt = tmp$sample, 
  alpha = c(.05, .01, 1e-3)) %>% 
  mutate(
    up = map2_dbl(
      .x = expt, .y = alpha,
      ~ tmp %>% filter(sample == .x, p.adj <= .y, diff > 0) %>% nrow()
      ),
    down = map2_dbl(
      .x = expt, .y = alpha,
      ~ tmp %>% filter(sample == .x, p.adj <= .y, diff < 0) %>% nrow()
      )
  )

# plot
p <- tibble(expt = 'siRNA control.scramble', up = 0, down = 0, alpha = .05) %>% 
  bind_rows(., n.deg) %>% 
  filter(alpha == .05) %>% 
  mutate(total.n = up+down, 
         tg = str_split_fixed(expt, '\\.', 2) %>% .[,1],
         cell = str_split_fixed(expt, '\\.', 2) %>% .[,2]) %>% 
  pivot_longer(cols = c(up, down)) %>% 
  mutate(value = case_when(grepl('down',name) ~ -1*value, T ~ value),
         tg = fct_reorder(tg, total.n),
         tg = fct_relevel(tg, 'siRNA control' )) %>% 
  ggplot(., aes(value, tg,  group = rev(alpha) ))+ 
  geom_bar(stat = 'identity', position = 'dodge', alpha = .8, width = .7, 
           aes(fill = cell, group = cell ))+
  # viridis::scale_fill_viridis('p.adj', direction = 1, discrete = T)+
  scale_fill_discrete('cell line')+
  scale_x_continuous('# differentially expressed proteins\nadjusted p-value \u2264 0.05',
                     labels = ~abs(.x))+
  labs(y = '')+
  facet_grid(cols = vars(name), scales = 'free')

print(p)
```

```{r}
ggsave(
  here::here('manuscript','n_dep.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=5
  )
```

```{r}
tmp = de.results %>% 
  filter(
    !is.na(diff), 
    (cell.1 == cell.2 & (grepl('siRNA', tg.1)|grepl('siRNA',tg.2))) 
    | (grepl('siRNA', tg.1) & grepl('siRNA',tg.2)) 
    , p.adj <= 0.05
    ) %>% 
  mutate(sample = str_c(tg,'.',cell.2))

orth <- gprofiler2::gorth(unique(tmp$gene), 'mmusculus','hsapiens' )

tmp <- left_join(tmp, orth %>% select(gene = input, hs.gene = ortholog_name), na_matches = 'never') %>% 
  nest( dep = -c(tg, cell.2) ) %>% 
  mutate( n = map_dbl(dep, ~nrow(.x)),
          tad = map(dep, ~scores %>% filter(GeneName %in% .x$hs.gene[!is.na(.x$hs.gene)]) %>% 
                      mutate(set = 'BioHyV KD DEP')),
          dist.plot = map(tad, ~ scores %>% mutate(set = 'All scored') %>% bind_rows(., .x) %>% 
                            ggplot(., aes(Overall, fill = set))+geom_density(alpha = .3)+guides(fill = 'none'))) %>% 
  arrange(desc(n))
```

```{r fig.width=12, fig.height=12}
# i = 4; tmp$dist.plot[[i]]+ggtitle(str_c(tmp$tg[i],' ', tmp$cell.2[i]))
cowplot::plot_grid(plotlist = map(1:59, ~ tmp$dist.plot[[.x]]+
                                    ggtitle( str_c(tmp$tg[.x],' ', tmp$cell.2[.x]) )))
```

example:
```{r fig.width=15, fig.height=5}
t = 'Pdhb'

tmp <- de.results %>% filter( tg == t ) %>% 
    mutate(n.pep = if_else(is.na(n.pep), 1, n.pep))

p1 = ggplot(tmp, aes(diff, -log10(p.adj)))+ 
    geom_vline(xintercept = 0, lwd = .1, lty = 2)+
    geom_point(alpha = .5, aes(size = n.pep, color = p.adj <= 0.05 & !is.na(p.adj) )) +
    facet_wrap(~cell.2)+
    ggrepel::geom_label_repel(
        data = subset(tmp, p.adj <= sig & gene != t),
        aes(label = gene),
        segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
        size = 2, alpha = .7, min.segment.length = 0, max.overlaps = 10,
        max.time = 1, max.iter = 1e5)+
    ggrepel::geom_label_repel( 
        data = subset(tmp, gene == t),
        aes(label = gene), 
        segment.curvature = -0.1,segment.ncp = 3,segment.angle = 20,
        size = 2, alpha = .7, min.segment.length = 0, max.overlaps = 40, 
        max.time = 1, max.iter = 1e5, fill = 'red')+
    labs(
        subtitle = unique(tmp$tg),
        x = 'logFC'
    )+
    scale_color_manual('significant', values = c('grey', 'black'))+
    scale_size_continuous('N peptides', breaks = c(1,10, 100, 1000, 2000))+
    theme( legend.position = 'bottom' )

tmp2 <- tmp %>% 
    select(gene, cell.2, diff, p.adj, n.pep) %>% 
    filter(p.adj <= sig) %>% 
    pivot_wider(id_cols = c(gene,n.pep), 
                names_from = cell.2, 
                values_from = c(diff, p.adj)
                ,values_fill = 0) %>% 
    rowwise() %>% 
    mutate( mn = mean(-log10(p.adj_scramble), -log10(p.adj_psen2_kd)),
            mn = if_else(p.adj_scramble != 0 & p.adj_psen2_kd != 0, mn, 0))

p2 = ggplot(tmp2, 
            aes(diff_scramble, diff_psen2_kd))+
    geom_vline(xintercept = 0, lwd = .05)+
    geom_hline(yintercept = 0, lwd = .05)+
    geom_smooth(data = subset(tmp2, p.adj_scramble != 0 & p.adj_psen2_kd != 0),
                method = 'lm', lwd = .3, alpha = .5)+
    geom_point(aes(size = n.pep), alpha = .3)+
    ggrepel::geom_label_repel(
        data = subset(tmp2, mn != 0),
        aes(label = gene),
        segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
        size = 2, alpha = .7, min.segment.length = 0, max.overlaps = 10,
        max.time = 1, max.iter = 1e5)+
    scale_size_continuous('N peptides', breaks = c(1,10, 100, 1000, 2000))+
    labs(x = 'logFC scramble', y = 'logFC Psen2 KD')

p = cowplot::plot_grid(p1,p2, nrow = 1, align = 'hv', axis = 'tb', rel_widths = c(1, .7))
print(p)
```

```{r}
ggsave(
  here::here('manuscript','pdhb_volcano.pdf'),
  plot = p,
  device = cairo_pdf,
  width=15, height=5
  )
```

```{r fig.width=5, fig.height=8}
tmp <- de.results %>% 
    mutate(g = str_split(gene,';')) %>% 
    unnest_longer(g) %>% 
    filter(tg == g) 

missing <- setdiff(de.results$tg, tmp$tg) %>% str_subset(.,"siRNA control", negate=T)

tmp <- tibble(tg = missing, diff = rep(0, length(missing)), p.adj = rep(1, length(missing))) %>% 
  crossing(., cell.2 = c('psen2_kd','scramble')) %>% 
  bind_rows(tmp, .) %>% 
  mutate(mn_lfc = mean(diff, na.rm =T), .by = tg) %>% 
  mutate(
    cell.2 = fct_relevel(cell.2, 'scramble'),
    tg = fct_reorder(tg, mn_lfc)
    )

p <- ggplot(tmp, aes(diff, tg, group = gene))+
    geom_bar(stat = 'identity', position = 'dodge', 
             aes(fill = p.adj <= 0.05),
             alpha = .6, color = 'black')+
    scale_fill_manual('significant', values = c('grey30','red'))+
    scale_y_discrete(limits = rev)+
    facet_grid(cols = vars(cell.2))+
    labs(x = 'target protein log2 Fold Change\nsiRNA vs Control', y = '')+
    theme(legend.position = 'top')

print(p)

ggsave(
  here::here('manuscript','target_sig_de.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=8
  )
```

## Module corr

```{r}
syn$login()
path = syn$get('syn59500204') %>% .$path 
load(path)

disease.modules <- unique(pmod.AD.Control$Module) %>%
    str_subset(pattern = 'M42_|M11_|M24_|M7_|M25_|M3_|M5_|M8_|M1_|M20_|M2_|M29_')
```

```{r}
tmp <- de.results %>% 
  group_by(gene, n.pep) %>% 
  summarise(mn = median(abs(diff), na.rm = T),
            sd = sd(abs(diff), na.rm = T),
            max = max(abs(diff), na.rm =T))

ggplot(tmp, aes(log10(n.pep), max))+
  geom_vline(xintercept = log10(4), color = 'red', lty = 2)+
  geom_point()+
  geom_smooth(method='lm')+
  labs(x = 'n peptides, log10', y = 'max logFC', subtitle = 'red line = 4 peptides')
```

```{r}
# model_vs_ampad <- log.fc %>%
#   pivot_longer(-gene,names_to = 'sample', values_to = 'log2.fc') %>%
#   mutate(target = str_split_fixed(sample, '\\.',2)[,1],
#          cell = str_split_fixed(sample, '\\.',2)[,2]) %>%
#   left_join(., n.pep, by = c('gene'='Genes')) %>%
#   # filter(n.pep > 3) %>%
#   inner_join(pmod.AD.Control,., by = c('Symbol'='gene'), multiple = 'all') %>%
#   group_by(Module, target, cell) %>%
#   nest(data = c(Symbol, AD.Control, log2.fc, n.pep))

model_vs_ampad <- de.results %>%
  filter(
    !is.na(diff), 
    (cell.1 == cell.2 & (grepl('siRNA', tg.1)|grepl('siRNA',tg.2))) 
    | (grepl('siRNA', tg.1) & grepl('siRNA',tg.2)) 
    ) %>% 
  # mutate(sample = str_c(tg,'.',cell.2)) %>% 
  # filter(n.pep > 4) %>%
  select(target = tg, cell = cell.2, gene, log2.fc = diff, n.pep) %>%
  inner_join(pmod.AD.Control,., by = c('Symbol'='gene'), multiple = 'all') %>%
  group_by(Module, target, cell) %>%
  nest(data = c(Symbol, AD.Control, log2.fc, n.pep))

cor.df <- model_vs_ampad %>% 
  mutate(
    cor_test = map(data, ~ cor.test(.x[["log2.fc"]], .x[["AD.Control"]], method = "pearson")),
    correlation = map_dbl(cor_test, "estimate"),
    p_value = map_dbl(cor_test, "p.value")
    ) %>%
  ungroup() %>%
  dplyr::select(-cor_test) %>% 
  mutate(significant = p_value < 0.05,
         mod.n = str_split_fixed(Module, '_', 2) %>% .[,1] %>% str_remove_all('M') %>% as.numeric,
         Module = fct_reorder(Module, mod.n)) %>% 
  left_join(., biohyv_targets %>% 
              mutate(target = str_to_sentence(target),
                     hyp_area = str_split(hyp_area , ' \\| ')) %>% 
              unnest(hyp_area)) %>% 
  filter(grepl('synapse',hyp_area)==F)

dz.cor.df <- cor.df %>% filter(Module %in% disease.modules) 
```

```{r fig.width = 16, fig.height=16}
tmp <- cor.df %>%
  ungroup() %>%
  mutate(mn = median(correlation[significant == T]), .by = target) %>%
  mutate(
    target = fct_relevel(target, 'siRNA control')
    # target = fct_reorder(target, mn)
    )

p <- ggplot(tmp, aes(Module, cell))+
  geom_tile(color = 'black', fill = 'white')+
  geom_point(aes(color = correlation, size = abs(correlation)))+
  geom_point(data = subset(tmp, significant), stroke = 1.2,shape = 0, size = 5)+
  # scale_x_discrete(position = 'top')+
  scale_x_discrete(
    position = 'top', 
    labels = function(x) { ifelse(x %in% disease.modules,paste0("**", x, "**"), x)}
    )+
  scale_y_discrete(limits = rev, position = 'right')+
  scale_size(guide = 'none')+
  scale_color_gradient2(name = "Correlation", low = "#85070C", high = "#164B6E", 
                         guide = guide_colorbar(ticks = FALSE) )+
  labs(x = NULL, y = NULL)+
  facet_grid(rows = vars(target), scales = 'free', space = 'free', switch = 'y')+ #
  theme(
    strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),
    strip.text.y.left = element_text(angle = 0,size = 12),
    axis.ticks = element_blank(),
    axis.text.x = ggtext::element_markdown(angle = 45, hjust = 0, vjust = 0, size = 12, face = 'plain'),
    axis.text.y = element_text(angle = 0, size = 12),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "right",
    plot.margin = margin(2,2,2,20)
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','module_prot_corr.pdf'),
  plot = p,
  device = cairo_pdf,
  width=16, height=16
  )
```

```{r fig.width = 8, fig.height=16}
tmp <- dz.cor.df %>%
  ungroup() %>%
  mutate(mn = median(correlation[significant == T]), .by = target) %>%
  mutate(
    target = fct_relevel(target, 'siRNA control')
    # target = fct_reorder(target, mn)
    )


p <- ggplot(tmp, aes(Module, cell))+
  geom_tile(color = 'black', fill = 'white')+
  geom_point(aes(color = correlation, size = abs(correlation)))+
  geom_point(data = subset(tmp, significant), stroke = 1.2,shape = 0, size = 5)+
  scale_x_discrete(position = 'top')+
  scale_y_discrete(limits = rev, position = 'right')+
  scale_size(guide = 'none')+
  scale_color_gradient2(name = "Correlation", low = "#85070C", high = "#164B6E", 
                         guide = guide_colorbar(ticks = FALSE) )+
  labs(x = NULL, y = NULL)+
  facet_grid(rows = vars(target), scales = 'free', space = 'free', switch = 'y')+
  theme(
    strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),
    strip.text.y.left = element_text(angle = 0,size = 12),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 0, vjust = 1, size = 12),
    axis.text.y = element_text(angle = 0, size = 12),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "right",
    plot.margin = margin(2,2,2,20)
  )
print(p)
```

```{r}
ggsave(
  here::here('manuscript','dz_module_prot_corr.pdf'),
  plot = p,
  device = cairo_pdf,
  width=8, height=16
  )
```

```{r fig.width=6, fig.height=6}
cor.df %>% filter(target != 'siRNA control') %>% 
  group_by(target) %>% 
  summarise(
    n = length((Module[which(significant)])),
    n.pos = length((Module[which(significant & correlation > 0)])),
    n.neg = length((Module[which(significant & correlation < 0)])),
    ) %>% 
  # filter(any(n>4)) %>% 
  ungroup() %>% 
  mutate(target = fct_reorder(target, n.neg)) %>% 
  pivot_longer(cols = c(n.pos, n.neg)) %>% 
  ggplot(aes(value, target, fill = name))+
  geom_bar(stat= 'identity', position = 'dodge')+
  labs(x = '# correlated Modules', y = '')
```

```{r fig.width=6, fig.height=6}
cor.df %>% filter(target != 'siRNA control') %>% 
  group_by(Module) %>% 
  summarise(
    n = length(unique(target[which(significant)])),
    n.pos = length(unique(target[which(significant & correlation > 0)])),
    n.neg = length(unique(target[which(significant & correlation < 0)])),
    ) %>% 
  # filter(any(n>4)) %>% 
  ungroup() %>% 
  mutate(Module = fct_reorder(Module, n.neg)) %>% 
  pivot_longer(cols = c(n.pos, n.neg)) %>% 
  # mutate(Module = fct_reorder(Module, n)) %>% 
  ggplot(aes(value, Module, fill = name))+
  geom_bar(stat= 'identity', position = 'dodge')+
  scale_y_discrete(
    labels = function(x) { ifelse(x %in% disease.modules,paste0("**", x, "**"), x)}
  )+
  labs(y = '', x = '# targets')+
  theme(
    axis.text.y = ggtext::element_markdown(face = 'plain'),
  )
```

```{r fig.width=6, fig.height=3}
cor.df %>% filter(target != 'siRNA control') %>% 
    group_by(Module, hyp_area) %>% summarise(n = length(unique(target[which(significant)]))) %>% 
    filter(any(n>3)) %>%
    ungroup() %>% mutate(Module = fct_reorder(Module, n)) %>% 
    ggplot(aes(n, Module, fill = hyp_area))+geom_bar(stat= 'identity', position = 'dodge')
```

```{r fig.width = 15, fig.height=5}
tmp <- cor.df %>% 
  filter(
    grepl('M2_|M21_|M25_|M26_|M27_|M42_',Module)
  )

ggplot(tmp, aes(target,cell))+
  geom_tile(color = 'black', fill = 'white')+
  geom_point(aes(color = correlation, size = abs(correlation)))+
  geom_point(data = subset(tmp, significant), stroke = 1.2,shape = 0, size = 5)+
  scale_x_discrete(position = 'bottom')+
  scale_y_discrete(limits = rev, position = 'right')+
  scale_size(guide = 'none')+
  scale_color_gradient2(name = "Correlation", low = "#85070C", high = "#164B6E", 
                         guide = guide_colorbar(ticks = FALSE) )+
  labs(x = NULL, y = NULL)+
  facet_grid(rows = vars(Module), cols = vars(hyp_area), scales = 'free', space = 'free', switch = 'y')+
  theme(
    strip.text.x = element_text(angle = 0, size = 10,colour = c("black")),
    strip.text.y.left = element_text(angle = 0,size = 12),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 12),
    axis.text.y = element_text(angle = 0, size = 12),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "right",
    plot.margin = margin(2,2,2,30)
  )

```

```{r fig.width=6, fig.height=3}
# t = 'Ap2a2'
# m = 'M2_'

t = 'Adh5'
m = 'M2_'

tmp <- cor.df %>% 
  filter(target == t, 
         grepl(m,Module)) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(cor = cor.test(log2.fc, AD.Control) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, AD.Control))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(Symbol)),
    aes(label = Symbol), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell)+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$Module[grep(m,cor.df$Module)]))

```

## Domain corr

```{r}
# convert symbols to mm
if(!exists('ids')){ ids <- gprofiler2::gorth(unique(bd.genes$symbol), 
                                             source_organism = 'hsapiens', 
                                             target_organism = 'mmusculus') }

bd.genes.mm <- inner_join(bd.genes, ids %>% select(symbol = input, mm.symbol = ortholog_name))

# join up
model_vs_ampad <- de.results %>%
  filter(
    !is.na(diff), 
    (cell.1 == cell.2 & (grepl('siRNA', tg.1)|grepl('siRNA',tg.2))) 
    | (grepl('siRNA', tg.1) & grepl('siRNA',tg.2)) 
    ) %>% 
  # mutate(sample = str_c(tg,'.',cell.2)) %>% 
  # filter(n.pep > 4) %>%
  select(target = tg, cell = cell.2, prot = gene, log2.fc = diff, n.pep) %>%
  inner_join(bd.genes.mm %>% select(set, prot = mm.symbol, ad.ctrl = Pro_TE),., 
             by = c('prot'), multiple = 'all') %>% 
  filter(!is.na(set)) %>% 
  nest(data = c(prot, n.pep, ad.ctrl, log2.fc), .by = c(set, target, cell)) %>% 
  mutate(data = map(data, ~distinct(.x)),
         n = map_dbl(data, ~nrow(.x))) %>% 
  filter(n > 3)

cor.df <- model_vs_ampad %>% 
  mutate(
    cor_test = map(data, ~ cor.test(.x[["log2.fc"]], .x[["ad.ctrl"]], method = "pearson")),
    correlation = map_dbl(cor_test, "estimate"),
    p_value = map_dbl(cor_test, "p.value")
    ) %>%
  ungroup() %>%
  dplyr::select(-cor_test) %>% 
  mutate( padj = p.adjust(p_value, method = 'BH'), 
          .by = c(target, cell) ) %>% 
  mutate(significant = padj <= 0.05) %>% 
  left_join(., biohyv_targets %>% 
              mutate(target = str_to_sentence(target),
                     hyp_area = str_split(hyp_area , ' \\| ')) %>% 
              unnest(hyp_area)) %>% 
  filter(grepl('synapse',hyp_area)==F)
```

### biodomain

```{r fig.width = 8, fig.height=16}
tmp <- cor.df %>% filter(grepl('_',set)==F) %>% mutate(target = fct_relevel(target, 'siRNA control'))

p <- ggplot(data = tmp, aes(set, cell))+
  geom_tile(color = 'black', fill = 'white')+
  geom_point(aes(color = correlation, size = abs(correlation)))+
  geom_point(data = subset(tmp, significant), stroke = 1.2,shape = 0, size = 5)+
  scale_x_discrete(position = 'top')+
  scale_y_discrete(limits = rev, position = 'right')+
  scale_size(guide = 'none')+
  scale_color_gradient2(name = "Correlation", low = "#85070C", high = "#164B6E", 
                         guide = guide_colorbar(ticks = FALSE) )+
  labs(x = NULL, y = NULL)+
  facet_grid(rows = vars(target), scales = 'free', space = 'free', switch = 'y')+
  theme(
    strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),
    strip.text.y.left = element_text(angle = 0,size = 12),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 0, vjust = 1, size = 12),
    axis.text.y = element_text(angle = 0, size = 12),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "right",
    plot.margin = margin(2,2,2,20)
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','biodom_prot_corr.pdf'),
  plot = p,
  device = cairo_pdf,
  width=8, height=16
  )
```

```{r fig.width=6, fig.height=6}
p <- cor.df %>% filter(target != 'siRNA control', grepl('_',set)==F) %>% 
  group_by(target) %>% 
  summarise(
    n = length((set[which(significant)])),
    n.pos = length((set[which(significant & correlation > 0)])),
    n.neg = length((set[which(significant & correlation < 0)])),
    ) %>% 
  # filter(any(n>4)) %>% 
  ungroup() %>% 
  mutate(target = fct_reorder(target, n.neg)) %>% 
  pivot_longer(cols = c(n.pos, n.neg)) %>% 
  ggplot(aes(value, target, fill = name))+
  geom_bar(stat= 'identity', position = 'dodge')+
  labs(x = '# correlated Biodomains', y = '')

print(p)

ggsave(
  here::here('manuscript','n_biodom_corr.pdf'),
  plot = p,
  device = cairo_pdf,
  width=6, height=6
  )
```

```{r fig.width=6, fig.height=6}
p <- cor.df %>% filter(target != 'siRNA control', grepl('_',set)==F) %>% 
  group_by(set, hyp_area) %>% 
  summarise(
    n = length(unique(target[which(significant)])),
    n.pos = length(unique(target[which(significant & correlation > 0)])),
    n.neg = length(unique(target[which(significant & correlation < 0)])),
    ) %>% 
  # filter(any(n>4)) %>% 
  ungroup() %>% 
  mutate(set = fct_reorder(set, n.neg)) %>% 
  pivot_longer(cols = c(n.pos, n.neg)) %>% 
  # mutate(Module = fct_reorder(Module, n)) %>% 
  ggplot(aes(value, set, fill = name))+
  facet_grid(cols = vars(hyp_area))+
  geom_bar(stat= 'identity', position = 'dodge')+
  labs(y = '', x = '# targets')+
  theme(legend.position = 'top')

print(p)

ggsave(
  here::here('manuscript','n_tg_by_biodom_corr.pdf'),
  plot = p,
  device = cairo_pdf,
  width=6, height=6
  )
```

### select subdoms

```{r fig.width=8, fig.height=16}
tl <- cor.df %>% 
  filter(target!='siRNA control',
         # significant, 
         grepl('_',set),
         grepl('MM_|IR_',set),
         !grepl('none',set)
         # , correlation < 0
         ) %>% pull(target)
l <- cor.df %>% 
  filter(#target %in% tl, 
         target!='siRNA control',
         # significant, 
         grepl('MM_|IR_',set),
         grepl('_',set), 
         !grepl('none',set)
         # ,correlation < 0
         ) %>% pull(set) %>% unique()

tmp <- cor.df %>% 
    filter(set %in% l, target %in% tl) %>% 
    mutate(max_n = max(n), .by = set) %>% 
    mutate(set1 = str_remove_all(set, '^[A-Za-z]{2}_') %>% str_trunc(., 45) %>% str_c('[',max_n,'] ' , .)
           , set1 = fct_reorder(set1, max_n)
           , dom = str_split_fixed(set,'_',2) %>% .[,1]
           ) %>% 
  filter( (dom %in% c('Ep','OS'))==F )
    
p <- ggplot(data = tmp, aes( set1, cell ))+
  facet_grid(cols = vars(dom), rows = vars(target), scales = 'free', space = 'free', switch = 'y')+
  geom_tile(color = 'black', fill = 'white')+
  geom_point(aes(color = correlation, size = abs(correlation)))+
  geom_point(data = subset(tmp, significant), stroke = 1.2,shape = 0, size = 6)+
  scale_x_discrete(position = 'bottom')+
  scale_y_discrete(limits = rev, position = 'right')+
  scale_size()+#guide = 'none'
  scale_color_gradient2(name = "Correlation", low = "#85070C", high = "#164B6E", 
                         guide = guide_colorbar(ticks = FALSE) )+
  labs(x = NULL, y = NULL)+
  theme(
    # strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 12),
    axis.text.y = element_text(angle = 0, size = 12),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "bottom",
    plot.margin = margin(2,2,2,20)
  )
print(p)
```

```{r fig.width=15, fig.height=15}
tl <- c('Ap2a2', 'Dlat','Dld','Pdhb','Pdha1','Psmc3', 'Yy1',
        'Ano6','Ifih1', 'Capn1', 'Itgav','Itgb2','Gabpa','Tlr4', 'Traf3','Sqstm1')

l <- cor.df %>% 
  filter(target %in% tl, 
         significant, 
         grepl('_',set), 
         !grepl('none',set), 
         correlation < 0) %>% pull(set) %>% unique()

tmp <- cor.df %>% 
    filter(set %in% l, target %in% tl) %>% 
    mutate(max_n = max(n), .by = set) %>% 
    mutate(set1 = str_remove_all(set, '^[A-Za-z]{2}_') %>% str_trunc(., 45) %>% str_c('[',max_n,'] ' , .)
           , set1 = fct_reorder(set1, max_n)
           , dom = str_split_fixed(set,'_',2) %>% .[,1]
           ) %>% 
  filter( (dom %in% c('Ep','OS'))==F )
    
p <- ggplot(data = tmp, aes( cell, set1 ))+
  facet_grid(rows = vars(dom), cols = vars(target), scales = 'free', space = 'free', switch = 'y')+
  geom_tile(color = 'black', fill = 'white')+
  geom_point(aes(color = correlation, size = abs(correlation)))+
  geom_point(data = subset(tmp, significant), stroke = 1.2,shape = 0, size = 6)+
  scale_x_discrete(position = 'bottom')+
  scale_y_discrete(limits = rev, position = 'right')+
  scale_size()+#guide = 'none'
  scale_color_gradient2(name = "Correlation", low = "#85070C", high = "#164B6E", 
                         guide = guide_colorbar(ticks = FALSE) )+
  labs(x = NULL, y = NULL)+
  theme(
    # strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 12),
    axis.text.y = element_text(angle = 0, size = 12),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "bottom",
    plot.margin = margin(2,2,2,20)
  )
print(p)
```

```{r}
ggsave(
  here::here('manuscript','select_subdom_prot_corr.pdf'),
  plot = p,
  device = cairo_pdf,
  width=15, height=15
  )
```

```{r fig.width=12, fig.height=18}
integer_breaks <- function(x) {
  seq(floor(min(x)), ceiling(max(x)))
}

tmp <- cor.df %>% 
  filter(target != 'siRNA control', grepl('_',set)==T) %>% 
  mutate(abbr = str_split_fixed(set, '_',2) %>% .[,1]) %>% 
  group_by(target, cell, abbr) %>% 
  summarise(
    n = length((set[which(significant)])),
    n.pos = length((set[which(significant & correlation > 0)])),
    n.neg = length((set[which(significant & correlation < 0)])),
    ) %>% 
  # filter(any(n>4)) %>% 
  ungroup() %>% 
  # mutate(target = fct_reorder(target, rev(n.neg) )) %>% 
  pivot_longer(cols = c(n.pos, n.neg)) %>% 
  left_join(., dom.cols %>% select(domain, abbr)) %>% 
  left_join(., biohyv_targets %>% mutate(target = str_to_sentence(target))) %>% 
  mutate(hyp = if_else(grepl('\\|', hyp_area), 'Both', str_split_fixed(hyp_area, ' ', 2) %>% .[,1]))

p <- ggplot(tmp, aes(value, domain, fill = name))+
  # facet_grid(cols = vars(target))+ #, rows = vars(name)
  facet_wrap(~target+cell+hyp, ncol = 10)+ #, rows = vars(name)
  geom_bar(stat= 'identity', position = 'dodge', color = 'grey20', lwd = .3, width = .6)+
  scale_fill_discrete('correlation sign', labels = c('negative','positive'))+
  scale_y_discrete(limits = rev)+
  scale_x_continuous(breaks = seq(0,12,2))+
  labs(x = '# significantly correlated Subdomains', y = '')+
  theme(legend.position = 'top')

print(p)

ggsave(
  here::here('manuscript','n_subdom_corr.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=18
  )
```

```{r fig.width=6, fig.height=6}
p <- cor.df %>% filter(target != 'siRNA control', grepl('_',set)==T) %>% 
  mutate(abbr = str_split_fixed(set, '_',2) %>% .[,1]) %>% 
  group_by(abbr, hyp_area) %>% 
  summarise(
    n = length(unique(target[which(significant)])),
    n.pos = length(unique(target[which(significant & correlation > 0)])),
    n.neg = length(unique(target[which(significant & correlation < 0)])),
    ) %>% 
  # filter(any(n>4)) %>% 
  ungroup() %>% 
  mutate(set = fct_reorder(abbr, n.neg)) %>% 
  pivot_longer(cols = c(n.pos, n.neg)) %>% 
  # mutate(Module = fct_reorder(Module, n)) %>% 
  ggplot(aes(value, set, fill = name))+
  facet_grid(cols = vars(hyp_area))+
  geom_bar(stat= 'identity', position = 'dodge')+
  labs(y = '', x = '# targets')+
  theme(legend.position = 'top')

print(p)

# ggsave(
#   here::here('manuscript','n_tg_by_biodom_corr.pdf'),
#   plot = p,
#   device = cairo_pdf,
#   width=6, height=6
#   )
```

<!-- ```{r, fig.width=10, fig.height=18, echo=FALSE, message=FALSE, results='asis'} -->
<!-- s = biodom %>% select(Biodomain, abbr) %>% distinct() %>% filter(!is.na(Biodomain)) -->

<!-- for(i in 1:nrow(s)){ -->

<!--   tmp <- cor.df %>%  -->
<!--     filter(grepl(str_c(s$abbr[i],'_'), set)) %>%  -->
<!--     mutate(max_n = max(n), .by = set) %>%  -->
<!--     mutate(set1 = str_remove_all(set, '^[A-Za-z]{2}_') %>% str_trunc(., 30) %>% str_c(., '\n','N = ',max_n), -->
<!--            set1 = fct_reorder(set1, desc(n))) -->

<!--   cor.plot = ggplot(data = tmp, aes( set1, cell))+ -->
<!--     facet_grid(rows = vars(target), scales = 'free', space = 'free', switch = 'y')+ -->
<!--     geom_tile(color = 'black', fill = 'white')+ -->
<!--     geom_point(aes(color = correlation, size = abs(correlation)))+ -->
<!--     geom_point(data = subset(tmp, significant), stroke = 1.2,shape = 0, size = 6)+ -->
<!--     scale_x_discrete(position = 'top')+ -->
<!--     scale_y_discrete(limits = rev, position = 'right')+ -->
<!--     scale_size()+#guide = 'none' -->
<!--     scale_color_gradient2(name = "Correlation", low = "#85070C", high = "#164B6E",  -->
<!--                            guide = guide_colorbar(ticks = FALSE) )+ -->
<!--     labs(x = NULL, y = NULL)+ -->
<!--     theme( -->
<!--       # strip.text.x = element_text(angle = 90, size = 10,colour = c("black")), -->
<!--       strip.text.y.left = element_text(angle = 0,size = 12), -->
<!--       axis.ticks = element_blank(), -->
<!--       axis.text.x = element_text(angle = 45, hjust = 0, vjust = 0, size = 12), -->
<!--       axis.text.y = element_text(angle = 0, size = 12), -->
<!--       panel.background = element_blank(), -->
<!--       plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"), -->
<!--       plot.title.position = "plot", -->
<!--       panel.grid = element_blank(), -->
<!--       legend.position = "bottom", -->
<!--       plot.margin = margin(2,2,2,20) -->
<!--     ) -->

<!--   cat("  \n###",  s$Biodomain[i], "  \n") -->
<!--   print(cor.plot) -->
<!--   plot.new() -->
<!--   dev.off() -->
<!--   cat("  \n  \n")   -->
<!-- } -->

<!-- ``` -->

### Ap

```{r fig.width=7, fig.height=4}
t = 'Kat5'
m = 'Ap_inflammatory cell'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```



### AM

```{r fig.width=8, fig.height=4}
t = 'Sqstm1'
m = 'AM_amyloid-beta formation'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```

### IR

```{r fig.width=8, fig.height=4}
t = 'Pdhb'
m = 'IR_neuroinflam'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```

```{r fig.width=8, fig.height=4}
t = 'Psmc3'
m = 'IR_neuroinflam'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```

```{r fig.width=8, fig.height=4}
t = 'Tlr4'
m = 'IR_activation of innate'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```

### MM

```{r fig.width=8, fig.height=4}
t = 'Ap2a2'
m = 'Mitochondrial Metabolism'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```

```{r fig.width=8, fig.height=4}
t = 'Ap2a2'
m = 'MM_tricar'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```

```{r fig.width=8, fig.height=4}
t = 'Pdhb'
m = 'MM_electron trans'
  
tmp <- cor.df %>% 
  filter( 
    target == t, 
    grepl(m,set) ) %>% 
  unnest(data) %>% 
  group_by(cell) %>% 
  mutate(n.pep = if_else(is.na(n.pep),1,n.pep),
         cor = cor.test(log2.fc, ad.ctrl) %>% list(),
         r = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(estimate)),
         p = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(p.value)),
         n = map_dbl(cor, ~ .x %>% broom::tidy() %>% pull(parameter))+2
         )

ggplot(tmp, aes(log2.fc, ad.ctrl))+
  geom_point(alpha = .2, aes(size = n.pep))+
  geom_smooth(method = 'lm', alpha = .3, lwd = .2)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_hline(yintercept = 0, lwd = .1)+
  ggrepel::geom_label_repel(
    data = subset(arrange(tmp, desc(abs(log2.fc))), !duplicated(prot)),
    aes(label = prot), size = 2, alpha = .7, 
    segment.curvature = -0.1, segment.ncp = 3, segment.angle = 20,
    min.segment.length = 0, max.overlaps = 10)+
  facet_wrap(~cell, scales = 'free_x')+
  geom_label(size = 2.5,
           x = -Inf, y = Inf, hjust = 0, vjust = 1,
           aes( label = str_c( 'r = ', signif(r, digits = 3),
                               '\np = ', signif(p, digits = 3),
                               '\nn = ', n )
           ))+
  labs(y = 'AMP-AD logFC', x = 'BV2 cell logFC', 
       subtitle = str_c(t,'\n',cor.df$set[grep(m,cor.df$set)]))

```


## GSEA

GSEA results
```{r}
syn$login()

omics.gsea <- syn$get('syn52749189') %>% .$path %>% readRDS()
# score.gsea <- syn$get('syn52749187') %>% .$path %>% readRDS()

gsea <- readRDS(here::here('manuscript','results',"gsea_results.rds")) 
```

Annotate biodomains
```{r message = F}
gsea <- gsea %>% 
  mutate(
    anno = map(
      res,
      ~ .x@result %>% 
        left_join(., 
                  biodom %>% 
                    select(ID = GO_ID, Biodomain:Subdomain, abbr:sd))
      )) 
```

Number of significantly enriched GO terms
```{r fig.width=5, fig.height=5}
sig = 0.05

tmp <- gsea %>% 
  left_join(., biohyv_targets %>% rename('tg' = 'target') %>% 
              mutate(tg = str_to_sentence(tg)), by = 'tg') %>% 
  mutate(
    up = map_dbl(anno, ~ length(unique(.x$ID[.x$NES>0 & .x$p.adjust <= sig]))),
    down = map_dbl(anno, ~ length(unique(.x$ID[.x$NES<0 & .x$p.adjust <= sig])))
    ) %>% 
  filter( #n >= 1,
         name == 'all'
         # , hyp_area %in% c('Mitochondrial Metabolism targets','Immune Response targets', NA_character_)
         ) %>% 
  bind_rows(
    tibble(tg = 'siRNA control', cell.2 = 'scramble', up = 0, down = 0), .) %>% 
  mutate(total.n = up+down) %>% 
  pivot_longer(c(up, down), names_to = 'dir', values_to = 'n') %>% 
  mutate(n = case_when(grepl('down',dir) ~ -1*n, T ~ n),
         tg = fct_reorder(tg, total.n),
         tg = fct_relevel(tg, 'siRNA control' ))
  

# plot
p <- tmp%>% 
  ggplot(., aes(n, tg))+ 
  geom_bar(stat = 'identity', position = 'dodge', alpha = .8, width = .7, 
           aes(fill = cell.2, group = cell.2 ))+
  # viridis::scale_fill_viridis('p.adj', direction = 1, discrete = T)+
  scale_fill_discrete('cell line')+
  scale_x_continuous(str_c('# significant GSEA terms\nadjusted p-value \u2264 ',sig),
                     labels = ~abs(.x))+
  labs(y = '')+
  facet_grid(cols = vars(dir), scales = 'free')

print(p)
```

```{r}
ggsave(
  here::here('manuscript','n_sig_terms.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=5
  )
```

```{r fig.width=8.5, fig.height=4}
sig = 0.05

p <- gsea %>% 
  left_join(., biohyv_targets %>% rename('tg' = 'target') %>% 
              mutate(tg = str_to_sentence(tg)), by = 'tg') %>% 
  mutate( n = map_dbl(anno, ~ length(unique(.x$ID[.x$p.adjust <= sig 
                                                     # & .x$Biodomain != 'none'
                                                  ])))
          # hyp_area = str_split(hyp_area, ' \\| ')
          ) %>% 
  # unnest(hyp_area) %>% 
  pivot_wider(id_cols = c(tg,hyp_area,name), 
              values_from = n, names_from = cell.2) %>% 
  filter(!is.na(hyp_area)
         , grepl('syn', hyp_area)==F
         , name!='sig'
         ) %>% 
  ggplot(aes(scramble, psen2_kd)) +
  # geom_smooth(method = 'lm', lty = 1, lwd = .5) + 
  geom_abline(intercept = 0, slope = 1, lty = 2, lwd = .2)+
  geom_point(aes(color = hyp_area), size = 2) + scale_x_log10()+ scale_y_log10() +
  # theme_bw()+ coord_cartesian(xlim = c(0.5,2e3), ylim = c(0.5,1.5e3)) +
  scale_color_discrete('hypothesis area')+
  labs(x = '# enriched GO terms\nscramble cells', y= '# enriched GO terms\nPsen2 KD cells')+
  ggrepel::geom_text_repel(aes(label = tg), size = 3.5, min.segment.length = 0)+
  facet_wrap(~str_c('GO Terms, adj p \u2264 ',sig))

print(p)
```

```{r}
ggsave(
  here::here('manuscript','n_sig_terms_by_celltype.pdf'),
  plot = p,
  device = cairo_pdf,
  width=8.5, height=4
  )
```


### Biodomains  {.tabset}

```{r fig.width=7, fig.height=5, warning=F}
sig = 0.05

tmp <- gsea %>% 
  # select(-gl, -res) %>% 
  filter(tg == 'Pdhb', name == 'all', cell.2 == 'psen2_kd') %>% 
  unnest(anno) %>% 
  bind_rows(., 
            omics.gsea %>% filter(score == 'Pro_TE') %>% 
              select(tg = score, anno) %>% unnest(anno)) %>% 
  ungroup() %>% 
  filter(p.adjust <= sig) %>% 
  mutate(Biodomain = if_else(is.na(Biodomain), 'none', Biodomain),
         color = if_else(Biodomain=='none', NA_character_, color)) %>% 
  mutate( n_sig = length(unique(ID))
          , med_nes = median(NES)
          , .by = c(Biodomain) 
          ) %>% 
  mutate( Biodomain = fct_reorder(Biodomain, n_sig),
          tg = case_when(tg == 'Pdhb' ~ 'Pdhb siRNA vs Control\nPsen2 KD BV2 cells', 
                         tg == 'Pro_TE' ~ 'AMP-AD Proteomics\nmeta-analysis'),
          tg = fct_relevel(tg, 'Pdhb siRNA vs Control\nPsen2 KD BV2 cells' ))

ggplot(tmp, aes(NES, Biodomain, fill = color), color = 'grey50')+
  geom_vline(xintercept = 0, lwd =.1)+
  # geom_jitter(data = subset(tmp, p.adjust > sig),
  #             shape = 21, alpha = .3, stroke = .5, size = .5, color = 'grey85')+
  geom_jitter(data = subset(tmp, p.adjust <= sig),
              shape = 21, alpha = .3, stroke = .5, aes(size = -log10(p.adjust)))+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), alpha = .1, lwd = .2,
              scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), alpha = .1, lwd = .2,
              scale= 'width', draw_quantiles = .5)+
  facet_wrap(~tg)+
  scale_fill_identity()+scale_color_identity()+
  labs(y = '')+
  theme(legend.position = 'top')

```

```{r fig.width=6, fig.height=5, warning=F}
sig = 0.05

tmp <- gsea %>% 
  # select(-gl, -res) %>% 
  filter(tg %in% c('siRNA control'), cell.2 == 'psen2_kd', name == 'all') %>% 
  unnest(anno) %>% 
  bind_rows(., 
            omics.gsea %>% filter(score == 'Pro_TE') %>% 
              select(tg = score, anno) %>% unnest(anno)) %>% 
  ungroup() %>% 
  filter(p.adjust <= sig) %>% 
  mutate(Biodomain = if_else(is.na(Biodomain), 'none', Biodomain),
         color = if_else(Biodomain=='none', NA_character_, color)) %>% 
  mutate( n_sig = length(unique(ID))
          , med_nes = median(NES)
          , .by = c(Biodomain) 
          ) %>% 
  mutate( Biodomain = fct_reorder(Biodomain, n_sig),
          tg = case_when(tg == 'siRNA control' ~ 'psen2 kd vs scramble', 
                         tg == 'Pro_TE' ~ 'AMP-AD Proteomics\nmeta-analysis')) 

ggplot(tmp, aes(NES, Biodomain, fill = color), color = 'grey50')+
  geom_vline(xintercept = 0, lwd =.1)+
  # geom_jitter(data = subset(tmp, p.adjust > sig),
  #             shape = 21, alpha = .3, stroke = .5, size = .5, color = 'grey85')+
  geom_jitter(data = subset(tmp, p.adjust <= sig),
              shape = 21, alpha = .3, stroke = .5, aes(size = -log10(p.adjust)))+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), alpha = .1, lwd = .2,
              scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), alpha = .1, lwd = .2,
              scale= 'width', draw_quantiles = .5)+
  facet_wrap(~tg)+
  scale_fill_identity()+scale_color_identity()+
  labs(y = '')+
  theme(legend.position = 'top')

```

```{r fig.width=6, fig.height=5, echo=FALSE, warning=FALSE, message=FALSE, results='asis'}

sig = 0.05
targets = unique(gsea$tg) %>% str_subset('siRNA',negate = T)

# plots <- map( targets, ~{})

# plot in tabs
plot.new();dev.off()
for(i in 1:length(targets) ){
  
  tmp <- gsea %>% 
    # select(-gl, -res) %>% 
    filter(tg == targets[i], name == 'all') %>% 
    unnest(anno) %>% 
    mutate(Biodomain = if_else(is.na(Biodomain), 'none', Biodomain)) %>% 
    group_by(Biodomain) %>% 
    mutate( n_sig = length(unique(ID[p.adjust <= sig])) ) %>% 
    ungroup() %>% 
    mutate( Biodomain = fct_reorder(Biodomain, n_sig)) 
  
  p <- ggplot(tmp, aes(NES, Biodomain, fill = color))+ #, color = color
    geom_vline(xintercept = 0, lwd =.1)+
    geom_jitter(data = subset(tmp, p.adjust > sig), shape = 21,
                alpha = .3, stroke = .5, size = .5, color = 'grey85')+
    geom_jitter(data = subset(tmp, p.adjust <= sig), shape = 21,
                alpha = .3, stroke = .5, aes(size = -log10(p.adjust)))+
    geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), alpha = .1, lwd = .2,
                scale= 'width', draw_quantiles = .5)+
    geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), alpha = .1, lwd = .2,
                scale= 'width', draw_quantiles = .5)+
    facet_wrap(~tg+cell.2)+
    scale_fill_identity()+scale_color_identity()+
    labs(y = ''
         # , subtitle = 'Psen2 KD vs Scramble (siRNA control)'
         )+
    theme(legend.position = 'top')
  
  cat("  \n####",  targets[i], "  \n")
  
  plot(p)
  plot.new()
  dev.off()
  cat("  \n  \n")  
}

```

### Subdomains {.tabset}

```{r fig.width=6, fig.height=4}
sig = 0.05
tmp <- gsea %>% 
  select(-gl, -res) %>%
  filter(tg %in% c('siRNA control'), cell.2 == 'psen2_kd', name == 'all') %>% 
  unnest(anno) %>% 
  mutate(Subdomain = if_else(is.na(Subdomain), 'none',Subdomain),
          sd = str_c(abbr, ' | ', Subdomain)) %>% 
  group_by(sd) %>% 
  mutate( n_sig = length(unique(ID[ p.adjust <= sig ])) ) %>% 
  ungroup() %>% group_by(Biodomain) %>% 
  mutate( sd = fct_reorder(sd, n_sig) ) %>% 
  filter(Subdomain != 'none')

ggplot(tmp, aes(NES, sd, fill = color, color = color))+
  theme(legend.position = 'top')+
  labs(y = ''
       # , subtitle = 'Psen2 KD vs Scramble (siRNA control)'
       )+
  scale_fill_identity()+scale_color_identity()+
  geom_vline(xintercept = 0, lwd =.1)+
  # geom_jitter(data = subset(tmp, p.adjust > 0.05),
  #             alpha = .3, stroke = .5, size = .5, color = 'grey85')+
  geom_jitter(data = subset(tmp, p.adjust <= sig),
              alpha = .3, stroke = .5, aes(size = -log10(p.adjust)))+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), alpha = .1, scale= 'width', draw_quantiles = .5)+
  facet_wrap(~'Psen2 KD vs Scramble')

```


```{r fig.width=7, fig.height=8, echo=FALSE, warning=FALSE, message=FALSE, results='asis'}

sig = 0.05
targets = unique(gsea$tg) %>% str_subset('siRNA',negate = T)

# plots <- map( targets, ~{})

# plot in tabs
plot.new();dev.off()
for(i in 1:length(targets) ){
  
  tmp <- gsea %>% 
    select(-gl, -res) %>%
    filter(tg == targets[i], name == 'all') %>% 
    unnest(anno) %>% 
    mutate(Subdomain = if_else(is.na(Subdomain), 'none',Subdomain),
            sd = str_c(abbr, ' | ', Subdomain)) %>% 
    group_by(sd) %>% 
    mutate( n_sig = length(unique(ID[ p.adjust <= sig ])) ) %>% 
    ungroup() %>% group_by(Biodomain) %>% 
    mutate( sd = fct_reorder(sd, n_sig) ) %>% 
    filter(Subdomain != 'none')
  
  ns = map_dbl(seq(0.05,0.95,0.05), ~ tmp %>% filter(p.adjust <= .x) %>% pull(sd) %>% unique %>% length) 
  sig = seq(0.05,0.95,0.05)[which(ns > 30) %>% .[1]]
  
  p <- ggplot(tmp, aes(NES, sd, fill = color, color = color))+
    theme(legend.position = 'bottom')+
    labs(y = '', subtitle = str_c('p.adjust \u2264 ',sig))+
    scale_fill_identity()+scale_color_identity()+
    geom_vline(xintercept = 0, lwd =.1)+
    geom_jitter(data = subset(tmp, p.adjust <= sig),
                alpha = .3, stroke = .5, aes(size = -log10(p.adjust)))+
    geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), alpha = .1, scale= 'width', draw_quantiles = .5)+
    geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), alpha = .1, scale= 'width', draw_quantiles = .5)+
    facet_wrap(~tg+cell.2)
  
  cat("  \n####",  targets[i], "  \n")
  
  plot(p)
  plot.new()
  dev.off()
  cat("  \n  \n")
  
}

```

### Comp vs AMP-AD{.tabset}

```{r warning=FALSE, message=FALSE}
# plan('multisession', workers = parallel::detectCores() - 2)
sig = 0.05

oe <- omics.gsea %>% filter(score == 'Pro_TE') %>% select(tg = score, anno) %>% unnest(anno) %>% 
  select(term = GO_ID, Description, amp_nes = NES, amp_padj = p.adjust) %>% distinct()
  # prot.enr@result %>% select(term = ID, amp.nes = NES, amp.padj = p.adjust) %>% distinct()

bv2.v.tad <- gsea %>% 
  ungroup() %>% 
  filter(name == 'all') %>%
  select(tg, cell.2, anno) %>% 
  unnest(anno) %>% 
  select(tg, cell.2, term = ID, Description, cell_nes = NES, cell_padj = p.adjust) %>% 
  distinct() %>% 
  full_join(., oe,by = c('term','Description')) %>% 
  mutate(
    across(contains('padj'), ~ if_else(is.na(.x), 1, .x)),
    across(contains('nes'), ~ if_else(is.na(.x), 0, .x)),
    amp_nes = if_else(amp_padj > 0.05, 0, amp_nes),
    cell_nes = if_else(cell_padj > 0.05, 0, cell_nes)
  ) %>% 
  distinct() %>% 
  nest(data = c(term, Description, cell_nes, cell_padj, amp_nes, amp_padj), .by = c(tg, cell.2))
  
comp = crossing( bv2.v.tad, set = bd.term.list$set ) %>% 
  mutate(
    data = map2(data, set, ~.x %>% filter(term %in% bd.term.list$terms[[which(bd.term.list$set == .y)]])),
    n.terms = map_dbl(data, ~nrow(.x)),
    n.sig.both = map_dbl(data, ~.x %>% filter(cell_padj <= 0.05 & amp_padj <= 0.05) %>% nrow()),
    jaccard = n.sig.both/n.terms
  ) %>% 
  filter(n.terms > 2, !is.na(tg)) %>% 
  mutate(
    cor = map(data, ~ cor.test(.x[['amp_nes']], .x[['cell_nes']], method = 'kendall')),
    correlation = map_dbl(cor, ~broom::tidy(.x) %>% pull(estimate)),
    p.value = map_dbl(cor, ~broom::tidy(.x) %>% pull(p.value))
  ) %>% 
  mutate( padj = p.adjust(p.value, method = 'BH'), .by = c(tg, cell.2) ) %>% 
  mutate( significant = padj < 0.05 )  

```

#### Biodomain

```{r fig.width=5, fig.height=10}
tmp <- comp %>%
  filter(!grepl('_', set)
         # , target != 'siRNA control'
         ) %>% 
  group_by(tg) %>% 
  mutate(#any_sig = any(n.sig.both > 0),
         any_sig = any(significant), #[(set %in% c('All Terms', 'No Biodomain'))==F]
         set = factor(set, 
                      levels = unique(comp$set) %>% 
                        str_subset(., 'All Terms|No Biodomain',negate = T) %>% 
                        c('All Terms','No Biodomain',.))) %>% 
  filter( any_sig ) %>% 
  mutate(cell = if_else(tg == 'siRNA control', 'psen2 vs scramble', cell.2),
         tg = fct_relevel(tg, c('siRNA control')))

p <- ggplot(tmp, aes(x = set, y = cell.2)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, significant),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'top', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 0, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "bottom",
    legend.box = 'vertical'
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_biodom.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=10
  )
```

```{r fig.width=7, fig.height=3.5}
tmp <- comp %>%
  filter(!grepl('_', set), 
         tg == 'Pdhb'
         # , target != 'siRNA control'
         ) %>% 
  group_by(tg) %>% 
  mutate(#any_sig = any(n.sig.both > 0),
         any_sig = any(significant[(set %in% c('All Terms', 'No Biodomain'))==F]),
         set = factor(set, 
                      levels = unique(comp$set) %>% 
                        str_subset(., 'All Terms|No Biodomain',negate = T) %>% 
                        c('All Terms','No Biodomain',.))) %>% 
  filter( any_sig ) %>% 
  mutate(cell = if_else(tg == 'siRNA control', 'psen2 vs scramble', cell.2),
         tg = fct_relevel(tg, c('siRNA control')))

p <- ggplot(tmp, aes(x = set, y = cell.2)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, significant),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 4),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'top', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 0, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "bottom",
    legend.box = 'vertical'
  )

print(p)
```


```{r fig.width=9, fig.height=4.5}
t <- comp %>% filter(set %in% c('Immune Response', 'Mitochondrial Metabolism') & padj <= 0.05) %>% pull(tg)

tmp <- comp %>%
  filter(!grepl('_', set)
         , tg != 'siRNA control'
         # , tg %in% c('Ap2a2','Pdhb','Pdha1','Psmc3','Dlat',
         #                 'Dld','Yy1','Itgav')
         , tg %in% t
         ) %>% 
  group_by(tg) %>% 
  mutate(#any_sig = any(n.sig.both > 0),
         any_sig = any(significant[(set %in% c('All Terms', 'No Biodomain'))==F]),
         set = factor(set, 
                      levels = unique(comp$set) %>% 
                        str_subset(., 'All Terms|No Biodomain',negate = T) %>% 
                        c('All Terms','No Biodomain',.))) %>% 
  filter( any_sig ) %>% 
  mutate(cell.2 = if_else(tg == 'siRNA control', 'psen2 vs scramble', cell.2),
         tg = fct_relevel(tg, c('siRNA control')))

p <- ggplot(tmp, aes(y = set, x = cell.2)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, padj <= 5e-2),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(cols = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'bottom', drop = T) +
  scale_y_discrete(position = 'left', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "right",
    legend.box = 'vertical'
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_biodom_select_tg.pdf'),
  plot = p,
  device = cairo_pdf,
  width=9, height=4.5
  )
```

```{r fig.width=7, fig.height=6}
t <- comp %>% filter(set %in% c('Immune Response', 'Mitochondrial Metabolism') & padj <= 0.05) %>% pull(tg)

tmp <- comp %>%
  filter(!grepl('_', set)
         , tg != 'siRNA control'
         # , tg %in% c('Ap2a2','Pdhb','Pdha1','Psmc3','Dlat',
         #                 'Dld','Yy1','Itgav')
         , tg %in% t
         ) %>% 
  group_by(tg) %>% 
  mutate(#any_sig = any(n.sig.both > 0),
         any_sig = any(significant[(set %in% c('All Terms', 'No Biodomain'))==F]),
         set = factor(set, 
                      levels = unique(comp$set) %>% 
                        str_subset(., 'All Terms|No Biodomain',negate = T) %>% 
                        c('All Terms','No Biodomain',.))) %>% 
  filter( any_sig ) %>% 
  mutate(cell.2 = if_else(tg == 'siRNA control', 'psen2 vs scramble', cell.2),
         tg = fct_relevel(tg, c('siRNA control')))

p <- ggplot(tmp, aes(x = set, y = cell.2)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, padj <= 5e-2),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'top', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 0, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "left",
    legend.box = 'vertical'
  )

print(p)
```

#### select subdomains

```{r fig.width=12, fig.height=9}
tmp <- comp %>%
  mutate(abbr = str_split_fixed(set, '_', 2) %>% .[,1],
         set1 = str_remove_all(set, '^[A-Za-z]{2}_') %>% str_trunc(., 40)) %>% 
  filter(grepl('_', set)
         , tg != 'siRNA control'
         # , (abbr %in% c('DR','Ep','RS'))==F
         ) %>% 
  mutate(any_sig_tg = any(significant[jaccard > 0]), .by = tg) %>% 
  mutate(any_sig_set = any(significant[jaccard > 0]), .by = set) %>% 
  filter(any_sig_tg,any_sig_set) %>% 
  mutate(cell.2 = if_else(tg == 'siRNA control', 'psen2 vs scramble', cell.2),
         tg = fct_relevel(tg, c('siRNA control')))

p <- ggplot(tmp, aes(x = cell.2, y = set1)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, padj <= 5e-2 & jaccard > 0),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(abbr), cols = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'bottom', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "top",
    legend.box = 'horizontal'
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_select_subdomain.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=9
  )
```

```{r fig.width=8, fig.height=8}
tmp <- comp %>%
  mutate(abbr = str_split_fixed(set, '_', 2) %>% .[,1],
         set1 = str_remove_all(set, '^[A-Za-z]{2}_') %>% str_trunc(., 40)) %>% 
  filter(grepl('_', set)
         , tg != 'siRNA control'
         , tg %in% c('Ap2a2','Pdhb','Pdha1','Psmc3','Dlat',
                         'Dld','Yy1','Itgav')
         , (abbr %in% c('DR','Ep','RS'))==F
         ) %>% 
  mutate(any_sig_tg = any(padj[jaccard > 0] <= 0.2), .by = tg) %>% 
  mutate(any_sig_set = any(padj[jaccard > 0] <= 0.2), .by = set) %>% 
  filter(any_sig_tg,any_sig_set) %>% 
  mutate(cell.2 = if_else(tg == 'siRNA control', 'psen2 vs scramble', cell.2),
         tg = fct_relevel(tg, c('siRNA control')))

p <- ggplot(tmp, aes(x = cell.2, y = set1)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, padj <= 5e-2 & jaccard > 0),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(abbr), cols = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'bottom', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "left",
    legend.box = 'vertical'
  )

print(p)
```

```{r fig.width=9, fig.height=5}
tmp <- comp %>%
  mutate(abbr = str_split_fixed(set, '_', 2) %>% .[,1],
         set1 = str_remove_all(set, '^[A-Za-z]{2}_') %>% str_trunc(., 40)) %>% 
  filter(grepl('_', set)
         , tg != 'siRNA control'
         # , target %in% c('Ap2a2','Pdhb','Pdha1','Psmc3','Dlat',
         #                 'Dld','Yy1','Itgav')
         , abbr %in% c('MM','IR')
         ) %>% 
  mutate(any_sig_tg = any(n.sig.both > 2), .by = tg) %>%
  # mutate(any_sig_set = any(tau.padj[jaccard > 0] <= 0.2), .by = set) %>% 
  filter(any_sig_tg) %>%
  mutate(cell.2 = if_else(tg == 'siRNA control', 'psen2 vs scramble', cell.2),
         tg = fct_relevel(tg, c('siRNA control')))

p <- ggplot(tmp, aes(x = cell.2, y = set1)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, padj <= 5e-2 & jaccard > 0),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(abbr), cols = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'bottom', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "left",
    legend.box = 'vertical'
  )

print(p)
```

```{r fig.width=12, fig.height=18}
integer_breaks <- function(x) {
  seq(floor(min(x)), ceiling(max(x)))
}

tmp <- comp %>% 
  filter(tg != 'siRNA control', grepl('_',set)==T) %>% 
  mutate(abbr = str_split_fixed(set, '_',2) %>% .[,1]) %>% 
  group_by(tg, cell.2, abbr) %>% 
  summarise(
    n = length((set[which(significant)])),
    n.pos = length((set[which(significant & correlation > 0)])),
    n.neg = length((set[which(significant & correlation < 0)])),
    ) %>% 
  # filter(any(n>4)) %>% 
  ungroup() %>% 
  # mutate(target = fct_reorder(target, rev(n.neg) )) %>% 
  pivot_longer(cols = c(n.pos, n.neg)) %>% 
  left_join(., dom.cols %>% select(domain, abbr)) %>% 
  left_join(., biohyv_targets %>% mutate(tg = str_to_sentence(target))) %>% 
  mutate(hyp = if_else(grepl('\\|', hyp_area), 'Both', str_split_fixed(hyp_area, ' ', 2) %>% .[,1]))

p <- ggplot(tmp, aes(value, domain, fill = name))+
  # facet_grid(cols = vars(target))+ #, rows = vars(name)
  facet_wrap(~tg+cell.2+hyp, ncol = 10)+ #, rows = vars(name)
  geom_bar(stat= 'identity', position = 'dodge', color = 'grey20', lwd = .3, width = .6)+
  scale_fill_discrete('correlation sign', labels = c('negative','positive'))+
  scale_y_discrete(limits = rev)+
  scale_x_continuous(breaks = seq(0,12,2))+
  labs(x = '# significantly correlated Subdomains', y = '')+
  theme(legend.position = 'top')

print(p)

# ggsave(
#   here::here('manuscript','n_subdom_corr.pdf'),
#   plot = p,
#   device = cairo_pdf,
#   width=12, height=18
#   )
```

#### examples

```{r fig.width=5, fig.height=5}
t = 'Pdhb'
# c = 'psen2_kd'
d = 'Immune Response'

tmp <- comp %>% filter(grepl(d, set), tg == t) %>% unnest(data) %>% 
  left_join(., biodom %>% select(term = GO_ID, Biodomain, Subdomain, subdomain_idx, abbr:color) ) %>% 
  filter(set == Biodomain) %>% 
  distinct() 

p1 <- ggplot(tmp, aes(amp_nes, cell_nes))+
  geom_vline(xintercept = 0, lwd = .2)+
  geom_hline(yintercept = 0, lwd = .2)+
  geom_point( aes(fill = color), shape = 21, color = 'grey20',
              stroke = .5, alpha = .5, size = 2)+
  geom_smooth( method = 'lm', formula = y ~ x - 1,  lwd = .5, lty = 2)+
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(x = 'AMP-AD\nNES', y = 'BV2 cell\nNES')+
  scale_fill_identity()

# print(p1)

cont.table  <-  comp %>% 
  filter(grepl(d, set), tg == t) %>% 
  select(set, cell.2, data, correlation, p.value) %>% 
  mutate(
    c = map(data, ~ .x %>% mutate(
      amp = case_when(
        amp_padj > 0.05 ~ 'NonSig',
        amp_padj <= 0.05 & amp_nes > 0 ~ 'Up',
        amp_padj <= 0.05 & amp_nes < 0 ~ 'Dn'
      ),
      cell = case_when(
        cell_padj > 0.05 ~ 'NonSig',
        cell_padj <= 0.05 & cell_nes > 0 ~ 'Up',
        cell_padj <= 0.05 & cell_nes < 0 ~ 'Dn'
      )
    )),
    d = crossing(
      amp.s = c('Up', 'Dn', 'NonSig'),
      cell.s = c('Up', 'Dn', 'NonSig')
    ) %>% list(),
    d = map2(d, c, ~ .x %>% rowwise() %>%
               mutate(
                 n = .y %>% filter(amp == amp.s, cell == cell.s) %>% nrow()
               ))
  ) %>% 
  unnest(d)

p2 = ggplot(cont.table, aes(amp.s, cell.s, fill = n)) +
  geom_tile()+ 
  geom_text(aes(label = n))+
  scale_fill_gradient(low = "white", high = "blue", guide = 'none') +
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(y = 'BV2 cell\n# terms', 
       x = 'AMP-AD\n# terms')+
  theme_bw()

# print(p2)
p = cowplot::plot_grid(p1,p2, ncol = 1, align = 'v', axis = 'lr' )
print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_Pdhb_IR.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=5
  )
```

```{r fig.width=5, fig.height=5}
t = 'Pdhb'
# c = 'psen2_kd'
d = 'Mitochondrial Metabolism'

tmp <- comp %>% filter(grepl(d, set), tg == t) %>% unnest(data) %>% 
  left_join(., biodom %>% select(term = GO_ID, Biodomain, Subdomain, subdomain_idx, abbr:color) ) %>% 
  filter(set == Biodomain) %>% 
  distinct() 

p1 <- ggplot(tmp, aes(amp_nes, cell_nes))+
  geom_vline(xintercept = 0, lwd = .2)+
  geom_hline(yintercept = 0, lwd = .2)+
  geom_point( aes(fill = color), shape = 21, color = 'grey20',
              stroke = .5, alpha = .5, size = 2)+
  geom_smooth( method = 'lm', formula = y ~ x - 1,  lwd = .5, lty = 2)+
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(x = 'AMP-AD\nNES', y = 'BV2 cell\nNES')+
  scale_fill_identity()

# print(p1)

cont.table  <-  comp %>% 
  filter(grepl(d, set), tg == t) %>% 
  select(set, cell.2, data, correlation, p.value) %>% 
  mutate(
    c = map(data, ~ .x %>% mutate(
      amp = case_when(
        amp_padj > 0.05 ~ 'NonSig',
        amp_padj <= 0.05 & amp_nes > 0 ~ 'Up',
        amp_padj <= 0.05 & amp_nes < 0 ~ 'Dn'
      ),
      cell = case_when(
        cell_padj > 0.05 ~ 'NonSig',
        cell_padj <= 0.05 & cell_nes > 0 ~ 'Up',
        cell_padj <= 0.05 & cell_nes < 0 ~ 'Dn'
      )
    )),
    d = crossing(
      amp.s = c('Up', 'Dn', 'NonSig'),
      cell.s = c('Up', 'Dn', 'NonSig')
    ) %>% list(),
    d = map2(d, c, ~ .x %>% rowwise() %>%
               mutate(
                 n = .y %>% filter(amp == amp.s, cell == cell.s) %>% nrow()
               ))
  ) %>% 
  unnest(d)

p2 = ggplot(cont.table, aes(amp.s, cell.s, fill = n)) +
  geom_tile()+ 
  geom_text(aes(label = n))+
  scale_fill_gradient(low = "white", high = "blue", guide = 'none') +
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(y = 'BV2 cell\n# terms', 
       x = 'AMP-AD\n# terms')+
  theme_bw()

# print(p2)
p = cowplot::plot_grid(p1,p2, ncol = 1, align = 'v', axis = 'lr' )
print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_Pdhb_MM.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=5
  )
```


```{r fig.width=5, fig.height=5}
t = 'Ap2a2'
# c = 'psen2_kd'
d = 'Mitochondrial Metabolism'

tmp <- comp %>% filter(grepl(d, set), tg == t) %>% unnest(data) %>% 
  left_join(., biodom %>% select(term = GO_ID, Biodomain, Subdomain, subdomain_idx, abbr:color) ) %>% 
  filter(set == Biodomain) %>% 
  distinct() 

p1 <- ggplot(tmp, aes(amp_nes, cell_nes))+
  geom_vline(xintercept = 0, lwd = .2)+
  geom_hline(yintercept = 0, lwd = .2)+
  geom_point( aes(fill = color), shape = 21, color = 'grey20',
              stroke = .5, alpha = .5, size = 2)+
  geom_smooth( method = 'lm', formula = y ~ x - 1,  lwd = .5, lty = 2)+
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(x = 'AMP-AD\nNES', y = 'BV2 cell\nNES')+
  scale_fill_identity()

# print(p1)

cont.table  <-  comp %>% 
  filter(grepl(d, set), tg == t) %>% 
  select(set, cell.2, data, correlation, p.value) %>% 
  mutate(
    c = map(data, ~ .x %>% mutate(
      amp = case_when(
        amp_padj > 0.05 ~ 'NonSig',
        amp_padj <= 0.05 & amp_nes > 0 ~ 'Up',
        amp_padj <= 0.05 & amp_nes < 0 ~ 'Dn'
      ),
      cell = case_when(
        cell_padj > 0.05 ~ 'NonSig',
        cell_padj <= 0.05 & cell_nes > 0 ~ 'Up',
        cell_padj <= 0.05 & cell_nes < 0 ~ 'Dn'
      )
    )),
    d = crossing(
      amp.s = c('Up', 'Dn', 'NonSig'),
      cell.s = c('Up', 'Dn', 'NonSig')
    ) %>% list(),
    d = map2(d, c, ~ .x %>% rowwise() %>%
               mutate(
                 n = .y %>% filter(amp == amp.s, cell == cell.s) %>% nrow()
               ))
  ) %>% 
  unnest(d)

p2 = ggplot(cont.table, aes(amp.s, cell.s, fill = n)) +
  geom_tile()+ 
  geom_text(aes(label = n))+
  scale_fill_gradient(low = "white", high = "blue", guide = 'none') +
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(y = 'BV2 cell\n# terms', 
       x = 'AMP-AD\n# terms')+
  theme_bw()

# print(p2)
p = cowplot::plot_grid(p1,p2, ncol = 1, align = 'v', axis = 'lr' )
print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_Ap2a2_MM.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=5
  )
```

```{r fig.width=5, fig.height=5}
t = 'Dlat'
# c = 'psen2_kd'
d = 'Immune Response'

tmp <- comp %>% filter(grepl(d, set), tg == t) %>% unnest(data) %>% 
  left_join(., biodom %>% select(term = GO_ID, Biodomain, Subdomain, subdomain_idx, abbr:color) ) %>% 
  filter(set == Biodomain) %>% 
  distinct() 

p1 <- ggplot(tmp, aes(amp_nes, cell_nes))+
  geom_vline(xintercept = 0, lwd = .2)+
  geom_hline(yintercept = 0, lwd = .2)+
  geom_point( aes(fill = color), shape = 21, color = 'grey20',
              stroke = .5, alpha = .5, size = 2)+
  geom_smooth( method = 'lm', formula = y ~ x - 1,  lwd = .5, lty = 2)+
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(x = 'AMP-AD\nNES', y = 'BV2 cell\nNES')+
  scale_fill_identity()

# print(p1)

cont.table  <-  comp %>% 
  filter(grepl(d, set), tg == t) %>% 
  select(set, cell.2, data, correlation, p.value) %>% 
  mutate(
    c = map(data, ~ .x %>% mutate(
      amp = case_when(
        amp_padj > 0.05 ~ 'NonSig',
        amp_padj <= 0.05 & amp_nes > 0 ~ 'Up',
        amp_padj <= 0.05 & amp_nes < 0 ~ 'Dn'
      ),
      cell = case_when(
        cell_padj > 0.05 ~ 'NonSig',
        cell_padj <= 0.05 & cell_nes > 0 ~ 'Up',
        cell_padj <= 0.05 & cell_nes < 0 ~ 'Dn'
      )
    )),
    d = crossing(
      amp.s = c('Up', 'Dn', 'NonSig'),
      cell.s = c('Up', 'Dn', 'NonSig')
    ) %>% list(),
    d = map2(d, c, ~ .x %>% rowwise() %>%
               mutate(
                 n = .y %>% filter(amp == amp.s, cell == cell.s) %>% nrow()
               ))
  ) %>% 
  unnest(d)

p2 = ggplot(cont.table, aes(amp.s, cell.s, fill = n)) +
  geom_tile()+ 
  geom_text(aes(label = n))+
  scale_fill_gradient(low = "white", high = "blue", guide = 'none') +
  facet_grid(rows = vars(set), cols = vars(cell.2))+
  labs(y = 'BV2 cell\n# terms', 
       x = 'AMP-AD\n# terms')+
  theme_bw()

# print(p2)
p = cowplot::plot_grid(p1,p2, ncol = 1, align = 'v', axis = 'lr' )
print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_Dlat_IR.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=5
  )
```


### vs Psen2 KD

```{r warning=FALSE, message=FALSE}
# plan('multisession', workers = parallel::detectCores() - 2)
sig = 0.05

oe <- gsea %>% filter(tg == 'siRNA control', name == 'all') %>% unnest(anno) %>% 
  ungroup() %>% select(term = ID, amp_nes = NES, amp_padj = p.adjust) %>% distinct()

# oe <- omics.gsea %>% filter(score == 'Pro_TE') %>% select(tg = score, anno) %>% unnest(anno) %>% 
#   select(term = GO_ID, Description, amp_nes = NES, amp_padj = p.adjust) %>% distinct()
#   # prot.enr@result %>% select(term = ID, amp.nes = NES, amp.padj = p.adjust) %>% distinct()

bv2.v.tad <- gsea %>% 
  ungroup() %>% 
  filter(name == 'all') %>%
  select(tg, cell.2, anno) %>% 
  unnest(anno) %>% 
  select(tg, cell.2, term = ID, Description, cell_nes = NES, cell_padj = p.adjust) %>% 
  distinct() %>% 
  full_join(., oe, by = c('term')) %>% 
  mutate(
    across(contains('padj'), ~ if_else(is.na(.x), 1, .x)),
    across(contains('nes'), ~ if_else(is.na(.x), 0, .x)),
    amp_nes = if_else(amp_padj > 0.05, 0, amp_nes),
    cell_nes = if_else(cell_padj > 0.05, 0, cell_nes)
  ) %>% 
  distinct() %>% 
  nest(data = c(term, Description, cell_nes, cell_padj, amp_nes, amp_padj), .by = c(tg, cell.2))
  
comp = crossing( bv2.v.tad, set = bd.term.list$set ) %>% 
  mutate(
    data = map2(data, set, ~.x %>% filter(term %in% bd.term.list$terms[[which(bd.term.list$set == .y)]])),
    n.terms = map_dbl(data, ~nrow(.x)),
    n.sig.both = map_dbl(data, ~.x %>% filter(cell_padj <= 0.05 & amp_padj <= 0.05) %>% nrow()),
    jaccard = n.sig.both/n.terms
  ) %>% 
  filter(n.terms > 2, !is.na(tg)) %>% 
  mutate(
    cor = map(data, ~ cor.test(.x[['amp_nes']], .x[['cell_nes']], method = 'kendall')),
    correlation = map_dbl(cor, ~broom::tidy(.x) %>% pull(estimate)),
    p.value = map_dbl(cor, ~broom::tidy(.x) %>% pull(p.value))
  ) %>% 
  mutate( padj = p.adjust(p.value, method = 'BH'), .by = c(tg, cell.2) ) %>% 
  mutate( significant = padj < 0.05 )  

```

#### Biodomain

```{r fig.width=5, fig.height=12}
tmp <- comp %>%
  filter(!grepl('_', set), tg != 'siRNA control') %>% 
  group_by(tg) %>% 
  mutate(#any_sig = any(n.sig.both > 0),
         any_sig = any(significant[(set %in% c('All Terms', 'No Biodomain'))==F]),
         set = factor(set, 
                      levels = unique(comp$set) %>% 
                        str_subset(., 'All Terms|No Biodomain',negate = T) %>% 
                        c('All Terms','No Biodomain',.))) %>% 
  filter( any_sig ) 

p <- ggplot(tmp, aes(x = set, y = cell.2)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, padj <= 5e-2),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'top', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 0, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "bottom",
    legend.box = 'vertical'
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_biodom_vs_psen_kd.pdf'),
  plot = p,
  device = cairo_pdf,
  width=5, height=12
  )
```

#### select subdomains

```{r fig.width=12, fig.height=5}
tmp <- comp %>%
  filter(grepl('_', set), tg != 'siRNA control') %>% 
  mutate(abbr = str_split_fixed(set, '_', 2) %>% .[,1],
         set1 = str_remove_all(set, '^[A-Za-z]{2}_') %>% str_trunc(., 40)) %>% 
  mutate(any_sig_tg = any(significant[jaccard > 0]), .by = tg) %>% 
  mutate(any_sig_set = any(significant[jaccard > 0]), .by = set) %>% 
  filter(any_sig_tg,any_sig_set)

p <- ggplot(tmp, aes(x = cell.2, y = set1)) +
  geom_tile(colour = "black", fill = "white") +
  geom_point(aes(fill = correlation, size = jaccard ) #.data$n.sig.both /.data$n.terms
             ,color = 'grey50', shape = 21) +
  geom_point(data = subset(tmp, padj <= 5e-2 & jaccard > 0),
             ,color="black", shape=0, size= 5, stroke = .75) +
  facet_grid(rows = vars(abbr), cols = vars(tg), scales = "free", space = "free", switch = 'y') +
  scale_size( "Jaccard Index",  range = c(1, 5),  limits = c(1e-9,1) ) +
  scale_fill_gradient2( name = "Kendall's \u03c4", 
                        low = "#85070C", high = "#164B6E", mid = 'grey95'
                        , guide = guide_colorbar(ticks = T) , limits = c(-1,1) ) +
  scale_x_discrete(position = 'bottom', drop = T) +
  scale_y_discrete(position = 'right', limits = rev)+
  labs(x = NULL, y = NULL) +
  theme(
    # plot.margin = margin(5,50,5,5),
    strip.text.x = element_text(size = 8, colour = c("black")),
    strip.text.y.left = element_text(angle = 0),
    axis.ticks = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    axis.text.y = element_text(angle = 0),
    panel.background = element_blank(),
    plot.title = element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = element_blank(),
    legend.position = "top",
    legend.box = 'horizontal'
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','gsea_comp_select_subdomain_vs_psen_kd.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=5
  )
```

# Fig 5 - Integrated phenotypes and omics


```{r fig.width=5, fig.height=9}
gh = 'AP2A2'
gm = 'Ap2a2'

# plot raw data
x = phagocytosis_lmm %>% 
  # select(-value) %>% 
  filter(siRNA %in% c(gh, 'Control')) %>% 
  pivot_longer(., starts_with('rep')) %>% 
  mutate( siRNA = factor( siRNA, levels = unique(phagocytosis_lmm$siRNA) ) )

p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
  # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .5, show.legend = F)+
  geom_point( aes(color = BV2.cell.line) , position = position_jitterdodge(), show.legend = F )+
  scale_y_discrete(limits = rev) +
  scale_x_continuous(trans = 'log10')+
  # facet_wrap(~hyp_area, scale = 'free_y')+
  theme(legend.position = 'top') +
  labs(y = 'siRNA target', x = '% positive cells\n ')

x = phagocytosis_lmm %>% 
  filter(siRNA %in% c(gh)) %>% 
  select(siRNA, BV2.cell.line, estimate, conf.low, conf.high, p.value, p.adj.BH) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(phagocytosis_lmm$siRNA) ) )

p2 <- ggplot(x, aes( estimate, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = 0, lwd = .2)+
  geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'right') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = conf.low, xmax = conf.high), 
                position = position_dodge(width =.9),
                lwd = .2, width = .4)+
  labs(y = '', 
       x = '% positive cells\nlog2 fold change, siRNA vs control'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow = 2 , rel_heights = c(1,.9), align = 'v', axis = 'lr', labels = 'AUTO')

x <- gsea %>% filter(tg == gm, name == 'all')
# x %>% unnest(anno) %>% ungroup() %>%
#     select(tg, cell.2, Biodomain, Subdomain, ID, Description, setSize, NES, p.adjust) %>%
#     pivot_wider(names_from = cell.2, values_from = c(p.adjust,NES,setSize))

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[1]]
z = left_join(y@result, biodom %>% rename('ID' = 'GO_ID'))
i = which(y$Description == 'phagocytosis')

p3 <- gseaplot(
    y, 
    geneSetID = i[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[1], '\n', 
                  y$ID[i[1]],': ',y$Description[i[1]], '\n',
                  'NES = ', signif(y$NES[i[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[i[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[2]]
j = which(y$Description == 'phagocytosis')

p4 <- gseaplot(
    y, 
    geneSetID = j[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[2], '\n', 
                  y$ID[j[1]],': ',y$Description[j[1]], '\n',
                  'NES = ', signif(y$NES[j[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[j[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# p <- cowplot::plot_grid(p,cowplot::plot_grid(p4, p3, nrow = 2, labels = c('C')), nrow = 1, rel_widths = c(1, .8))

p <- cowplot::plot_grid(
  p2,
  cowplot::plot_grid(p4, p3, nrow = 2),
  nrow = 2,
  rel_heights = c(.5,1),
  labels = 'AUTO'
)

print(p)

ggsave(
  here::here('manuscript','ap2a2_phagocytosis_details.pdf'),
  plot = p,
  width = 5, height = 9
)
```

```{r fig.width=5, fig.height=9}
gh = 'PDHB'
gm = 'Pdhb'

# plot raw data
x = mitotracker_lmm %>% 
  select(-value) %>%
  filter(siRNA %in% c(gh, 'Control')) %>% 
  pivot_longer(., starts_with('rep')) %>% 
  mutate( siRNA = factor( siRNA, levels = unique(mitotracker_lmm$siRNA) ) )

p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
  # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .5, show.legend = F)+
  geom_point( aes(color = BV2.cell.line) , position = position_jitterdodge(), show.legend = F )+
  scale_y_discrete(limits = rev) +
  scale_x_continuous(trans = 'log10')+
  # facet_wrap(~hyp_area, scale = 'free_y')+
  theme(legend.position = 'top') +
  labs(y = 'siRNA target', x = 'MitoTracker avg intensity')

x = mitotracker_lmm %>% 
  filter(siRNA %in% c(gh)) %>% 
  select(siRNA, BV2.cell.line, estimate, conf.low, conf.high, p.value, p.adj.BH) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(mitotracker_lmm$siRNA) ) )

p2 <- ggplot(x, aes( estimate, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = 0, lwd = .2)+
  geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'right') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = conf.low, xmax = conf.high), 
                position = position_dodge(width =.9),
                lwd = .2, width = .4)+
  labs(y = '', 
       x = 'normalized intensity fold change, log2, siRNA vs control'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow = 2 , rel_heights = c(1,.9), align = 'v', axis = 'lr', labels = 'AUTO')

x <- gsea %>% filter(tg == gm, name == 'all')
# x %>% unnest(anno) %>% ungroup() %>% 
#     select(tg, cell.2, Biodomain, Subdomain, ID, Description, setSize, NES, p.adjust) %>% 
#     pivot_wider(names_from = cell.2, values_from = c(p.adjust,NES,setSize)) 


# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[1]]
# i = which(y$Description == 'electron transport chain')
# i = which(y$Description == 'mitochondrial inner membrane')
i = which(y$Description == 'pyruvate metabolic process')

p3 <- gseaplot(
    y, 
    geneSetID = i[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[1], '\n', 
                  y$ID[i[1]],': ',y$Description[i[1]], '\n',
                  'NES = ', signif(y$NES[i[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[i[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[2]]
# j = which(y$Description == 'electron transport chain')
# j = which(y$Description == 'mitochondrial inner membrane')
j = which(y$Description == 'pyruvate metabolic process')

p4 <- gseaplot(
    y, 
    geneSetID = j[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[2], '\n', 
                  y$ID[j[1]],': ',y$Description[j[1]], '\n',
                  'NES = ', signif(y$NES[j[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[j[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# p <- cowplot::plot_grid(p,cowplot::plot_grid(p4, p3, nrow = 2, labels = c('C')), nrow = 1, rel_widths = c(1, .8))

p <- cowplot::plot_grid(
  p2,
  cowplot::plot_grid(p4, p3, nrow = 2),
  nrow = 2,
  rel_heights = c(.5,1),
  labels = 'AUTO'
)

print(p)

ggsave(
  here::here('manuscript','pdhb_etc_details.pdf'),
  plot = p,
  width = 5, height = 9
)
```

```{r fig.width=5, fig.height=9}
gh = 'CAPN1'
gm = 'Capn1'

# plot raw data
x = nfkb_lmm %>% 
  # select(-value) %>%
  filter(siRNA %in% c(gh, 'Control'), tx == 'none') %>% 
  pivot_longer(., starts_with('rep')) %>% 
  mutate( siRNA = factor( siRNA, levels = unique(nfkb_lmm$siRNA) ) )

p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
  # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .5, show.legend = F)+
  geom_point( aes(color = BV2.cell.line) , position = position_jitterdodge(), show.legend = F )+
  scale_y_discrete(limits = rev) +
  scale_x_continuous(trans = 'log10')+
  # facet_wrap(~hyp_area, scale = 'free_y')+
  theme(legend.position = 'top') +
  labs(y = 'siRNA target', x = 'MitoTracker avg intensity')

x = nfkb_lmm %>% 
  filter(siRNA %in% c(gh), tx == 'none') %>% 
  select(siRNA, BV2.cell.line, estimate, conf.low, conf.high, p.value, p.adj.BH) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(nfkb_lmm$siRNA) ) )

p2 <- ggplot(x, aes( estimate, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'right') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = conf.low, xmax = conf.high), 
                position = position_dodge(width =.9),
                lwd = .2, width = .4)+
  labs(y = '', 
       x = 'normalized intensity fold change, log2, siRNA vs control'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow = 2 , rel_heights = c(1,.9), align = 'v', axis = 'lr', labels = 'AUTO')

x <- gsea %>% filter(tg == gm, name == 'all')
# x %>% unnest(anno) %>% ungroup() %>%
#     select(tg, cell.2, Biodomain, Subdomain, ID, Description, setSize, NES, p.adjust) %>%
#     pivot_wider(names_from = cell.2, values_from = c(p.adjust,NES,setSize))


# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[1]]
# i = which(y$Description == 'electron transport chain')
i = which(y$Description == 'oxidative phosphorylation')

p3 <- gseaplot(
    y, 
    geneSetID = i[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[1], '\n', 
                  y$ID[i[1]],': ',y$Description[i[1]], '\n',
                  'NES = ', signif(y$NES[i[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[i[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[2]]
# j = which(y$Description == 'electron transport chain')
j = which(y$Description == 'oxidative phosphorylation')

p4 <- gseaplot(
    y, 
    geneSetID = j[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[2], '\n', 
                  y$ID[j[1]],': ',y$Description[j[1]], '\n',
                  'NES = ', signif(y$NES[j[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[j[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# p <- cowplot::plot_grid(p,cowplot::plot_grid(p4, p3, nrow = 2, labels = c('C')), nrow = 1, rel_widths = c(1, .8))

p <- cowplot::plot_grid(
  p2,
  cowplot::plot_grid(p4, p3, nrow = 2),
  nrow = 2,
  rel_heights = c(.5,1),
  labels = 'AUTO'
)

print(p)

ggsave(
  here::here('manuscript','pdhb_etc_details.pdf'),
  plot = p,
  width = 5, height = 9
)
```

## focused Subdomain enrichment


```{r fig.width=12, fig.height=6, warning=F}
sig = 0.1
t = 'Ap2a2'

tmp <- gsea %>% 
  select(-gl, -res) %>%
  filter(tg == t, name == 'all') %>% 
  unnest(anno) %>% 
  ungroup() %>% 
  filter(Biodomain %in% c('Mitochondrial Metabolism', 
                          'Immune Response',
                          'Autophagy', 
                          'Cell Cycle',
                          'Apoptosis')) %>% 
  mutate( n_sig = length(unique(ID[ p.adjust <= sig ])),
          med_nes = median(NES[ p.adjust <= sig ]),
          .by = sd) %>% 
  mutate( med_nes = if_else(is.na(med_nes),0, med_nes),
          sd = fct_reorder(sd, med_nes) ) %>% 
  arrange(sd)

p1 <- ggplot(tmp, aes(NES, sd, fill = color))+
  geom_vline(xintercept = 0, lwd =.1)+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_jitter(data = subset(tmp, p.adjust <= sig), shape = 21,
              alpha = .5, stroke = .5, aes(size = -log10(p.adjust)))+
  facet_grid(rows = vars(abbr), cols = vars(cell.2), scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(
    labels = function(x) str_remove_all(x, '^[A-Za-z]{2} \\| '), 
    position = 'right')+
  scale_fill_identity()+scale_color_identity()+
  scale_size_continuous(range = c(.5,4))+
  labs(y = '', title = t, subtitle = str_c('p.adjust \u2264 ', sig))+
  theme(legend.position = 'bottom',
        strip.text.y.left = element_text(angle = 0))

tmp <- biohyv.lmm %>% 
  filter(pheno != 'caspase',
         pheno != 'nfkb_LPS',
         target == str_to_upper(t)) %>%
  mutate(line = factor(line, c('scramble','psen2_kd')),
         pheno = case_when(pheno == 'nfkb_LPS' ~ 'NFkB +LPS',
                           pheno == 'nfkb_none' ~ 'NFkB -LPS',
                           pheno == 'alamar_blue' ~ 'alamarBlue',
                           T ~ pheno),
         pheno = factor(pheno, c('alamarBlue','mitotracker',
                                 'NFkB -LPS', 'NFkB +LPS', 'caspase', 'phagocytosis') )
         ) 

p2 <- ggplot(tmp, aes( target, line )) + 
  geom_point( aes(fill = lfc, size = -log10(p.value), shape = hit_class
                  , color = hit_class))+ #, alpha = hit_class
  facet_grid(rows = vars(pheno), scales = 'free', space = 'free', switch = 'y')+
  guides( alpha = 'none',  color = 'none', 
          size = guide_legend( override.aes=list(pch = 22)) ) +
  scale_y_discrete(limits = rev, position = 'right')+ 
  scale_size_continuous(guide = 'none')+ 
  scale_color_manual(values = c('grey20','grey50','grey20'))+
  scale_alpha_manual(values = c(1,1,1))+
  scale_shape_manual('Hit Class', values = c(24,22,25), 
                     breaks = c('Positive Hit','Non-Hit','Negative Hit') )+
  scale_fill_gradient2('Log2FC\nsiRNA v Cont', 
                       low = 'purple', mid = 'white', high = 'green', midpoint = 0)+
  labs(x = '', y = '' )+
  theme( axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
         legend.position = 'left')

p <- cowplot::plot_grid(p2,p1, nrow = 1, rel_widths = c(.33,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','Ap2a2_pheno_subdom_alignment.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=6
  )
```

```{r fig.width=12, fig.height=6, warning=F}
sig = 0.1
t = 'Pdhb'

tmp <- gsea %>% 
  select(-gl, -res) %>%
  filter(tg == t, name == 'all') %>% 
  unnest(anno) %>% 
  ungroup() %>% 
  filter(Biodomain %in% c('Mitochondrial Metabolism', 
                          'Immune Response',
                          'Autophagy', 
                          'Cell Cycle',
                          'Apoptosis')) %>% 
  mutate( n_sig = length(unique(ID[ p.adjust <= sig ])),
          med_nes = median(NES[ p.adjust <= sig ]),
          .by = sd) %>% 
  mutate( med_nes = if_else(is.na(med_nes),0, med_nes),
          sd = fct_reorder(sd, med_nes) ) %>% 
  arrange(sd)

p1 <- ggplot(tmp, aes(NES, sd, fill = color))+
  geom_vline(xintercept = 0, lwd =.1)+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_jitter(data = subset(tmp, p.adjust <= sig), shape = 21,
              alpha = .5, stroke = .5, aes(size = -log10(p.adjust)))+
  facet_grid(rows = vars(abbr), cols = vars(cell.2), scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(
    labels = function(x) str_remove_all(x, '^[A-Za-z]{2} \\| '), 
    position = 'right')+
  scale_fill_identity()+scale_color_identity()+
  scale_size_continuous(range = c(.5,4))+
  labs(y = '', title = t, subtitle = str_c('p.adjust \u2264 ', sig))+
  theme(legend.position = 'bottom',
        strip.text.y.left = element_text(angle = 0))

tmp <- biohyv.lmm %>% 
  filter(pheno != 'caspase',
         pheno != 'nfkb_LPS',
         target == str_to_upper(t)) %>%
  mutate(line = factor(line, c('scramble','psen2_kd')),
         pheno = case_when(pheno == 'nfkb_LPS' ~ 'NFkB +LPS',
                           pheno == 'nfkb_none' ~ 'NFkB -LPS',
                           pheno == 'alamar_blue' ~ 'alamarBlue',
                           T ~ pheno),
         pheno = factor(pheno, c('alamarBlue','mitotracker',
                                 'NFkB -LPS', 'NFkB +LPS', 'caspase', 'phagocytosis') )
         ) 

p2 <- ggplot(tmp, aes( target, line )) + 
  geom_point( aes(fill = lfc, size = -log10(p.value), shape = hit_class
                  , color = hit_class))+ #, alpha = hit_class
  facet_grid(rows = vars(pheno), scales = 'free', space = 'free', switch = 'y')+
  guides( alpha = 'none',  color = 'none', 
          size = guide_legend( override.aes=list(pch = 22)) ) +
  scale_y_discrete(limits = rev, position = 'right')+ 
  scale_size_continuous(guide = 'none')+ 
  scale_color_manual(values = c('grey20','grey50','grey20'))+
  scale_alpha_manual(values = c(1,1,1))+
  scale_shape_manual('Hit Class', values = c(24,22,25), 
                     breaks = c('Positive Hit','Non-Hit','Negative Hit') )+
  scale_fill_gradient2('Log2FC\nsiRNA v Cont', 
                       low = 'purple', mid = 'white', high = 'green', midpoint = 0)+
  labs(x = '', y = '' )+
  theme( axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
         legend.position = 'left')

p <- cowplot::plot_grid(p2,p1, nrow = 1, rel_widths = c(.33,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','Pdhb_pheno_subdom_alignment.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=6
  )
```

```{r fig.width=12, fig.height=6, warning=F}
sig = 0.1
t = 'Pdha1'

tmp <- gsea %>% 
  select(-gl, -res) %>%
  filter(tg == t, name == 'all') %>% 
  unnest(anno) %>% 
  ungroup() %>% 
  filter(Biodomain %in% c('Mitochondrial Metabolism', 
                          'Immune Response',
                          'Autophagy', 
                          'Cell Cycle',
                          'Apoptosis')) %>% 
  mutate( n_sig = length(unique(ID[ p.adjust <= sig ])),
          med_nes = median(NES[ p.adjust <= sig ]),
          .by = sd) %>% 
  mutate( med_nes = if_else(is.na(med_nes),0, med_nes),
          sd = fct_reorder(sd, med_nes) ) %>% 
  arrange(sd)

p1 <- ggplot(tmp, aes(NES, sd, fill = color))+
  geom_vline(xintercept = 0, lwd =.1)+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_jitter(data = subset(tmp, p.adjust <= sig), shape = 21,
              alpha = .5, stroke = .5, aes(size = -log10(p.adjust)))+
  facet_grid(rows = vars(abbr), cols = vars(cell.2), scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(
    labels = function(x) str_remove_all(x, '^[A-Za-z]{2} \\| '), 
    position = 'right')+
  scale_fill_identity()+scale_color_identity()+
  scale_size_continuous(range = c(.5,4))+
  labs(y = '', title = t, subtitle = str_c('p.adjust \u2264 ', sig))+
  theme(legend.position = 'bottom',
        strip.text.y.left = element_text(angle = 0))

tmp <- biohyv.lmm %>% 
  filter(pheno != 'caspase',
         pheno != 'nfkb_LPS',
         target == str_to_upper(t)) %>%
  mutate(line = factor(line, c('scramble','psen2_kd')),
         pheno = case_when(pheno == 'nfkb_LPS' ~ 'NFkB +LPS',
                           pheno == 'nfkb_none' ~ 'NFkB -LPS',
                           pheno == 'alamar_blue' ~ 'alamarBlue',
                           T ~ pheno),
         pheno = factor(pheno, c('alamarBlue','mitotracker',
                                 'NFkB -LPS', 'NFkB +LPS', 'caspase', 'phagocytosis') )
         ) 

p2 <- ggplot(tmp, aes( target, line )) + 
  geom_point( aes(fill = lfc, size = -log10(p.value), shape = hit_class
                  , color = hit_class))+ #, alpha = hit_class
  facet_grid(rows = vars(pheno), scales = 'free', space = 'free', switch = 'y')+
  guides( alpha = 'none',  color = 'none', 
          size = guide_legend( override.aes=list(pch = 22)) ) +
  scale_y_discrete(limits = rev, position = 'right')+ 
  scale_size_continuous(guide = 'none')+ 
  scale_color_manual(values = c('grey20','grey50','grey20'))+
  scale_alpha_manual(values = c(1,1,1))+
  scale_shape_manual('Hit Class', values = c(24,22,25), 
                     breaks = c('Positive Hit','Non-Hit','Negative Hit') )+
  scale_fill_gradient2('Log2FC\nsiRNA v Cont', 
                       low = 'purple', mid = 'white', high = 'green', midpoint = 0)+
  labs(x = '', y = '' )+
  theme( axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
         legend.position = 'left')

p <- cowplot::plot_grid(p2,p1, nrow = 1, rel_widths = c(.33,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','Pdha1_pheno_subdom_alignment.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=6
  )
```

```{r fig.width=12, fig.height=6, warning=F}
sig = 0.1
t = 'Dlat'

tmp <- gsea %>% 
  select(-gl, -res) %>%
  filter(tg == t, name == 'all') %>% 
  unnest(anno) %>% 
  ungroup() %>% 
  filter(Biodomain %in% c('Mitochondrial Metabolism', 
                          'Immune Response',
                          'Autophagy', 
                          'Cell Cycle',
                          'Apoptosis')) %>% 
  mutate( n_sig = length(unique(ID[ p.adjust <= sig ])),
          med_nes = median(NES[ p.adjust <= sig ]),
          .by = sd) %>% 
  mutate( med_nes = if_else(is.na(med_nes),0, med_nes),
          sd = fct_reorder(sd, med_nes) ) %>% 
  arrange(sd)

p1 <- ggplot(tmp, aes(NES, sd, fill = color))+
  geom_vline(xintercept = 0, lwd =.1)+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_jitter(data = subset(tmp, p.adjust <= sig), shape = 21,
              alpha = .5, stroke = .5, aes(size = -log10(p.adjust)))+
  facet_grid(rows = vars(abbr), cols = vars(cell.2), scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(
    labels = function(x) str_remove_all(x, '^[A-Za-z]{2} \\| '), 
    position = 'right')+
  scale_fill_identity()+scale_color_identity()+
  scale_size_continuous(range = c(.5,4))+
  labs(y = '', title = t, subtitle = str_c('p.adjust \u2264 ', sig))+
  theme(legend.position = 'bottom',
        strip.text.y.left = element_text(angle = 0))

tmp <- biohyv.lmm %>% 
  filter(pheno != 'caspase',
         pheno != 'nfkb_LPS',
         target == str_to_upper(t)) %>%
  mutate(line = factor(line, c('scramble','psen2_kd')),
         pheno = case_when(pheno == 'nfkb_LPS' ~ 'NFkB +LPS',
                           pheno == 'nfkb_none' ~ 'NFkB -LPS',
                           pheno == 'alamar_blue' ~ 'alamarBlue',
                           T ~ pheno),
         pheno = factor(pheno, c('alamarBlue','mitotracker',
                                 'NFkB -LPS', 'NFkB +LPS', 'caspase', 'phagocytosis') )
         ) 

p2 <- ggplot(tmp, aes( target, line )) + 
  geom_point( aes(fill = lfc, size = -log10(p.value), shape = hit_class
                  , color = hit_class))+ #, alpha = hit_class
  facet_grid(rows = vars(pheno), scales = 'free', space = 'free', switch = 'y')+
  guides( alpha = 'none',  color = 'none', 
          size = guide_legend( override.aes=list(pch = 22)) ) +
  scale_y_discrete(limits = rev, position = 'right')+ 
  scale_size_continuous(guide = 'none')+ 
  scale_color_manual(values = c('grey20','grey50','grey20'))+
  scale_alpha_manual(values = c(1,1,1))+
  scale_shape_manual('Hit Class', values = c(24,22,25), 
                     breaks = c('Positive Hit','Non-Hit','Negative Hit') )+
  scale_fill_gradient2('Log2FC\nsiRNA v Cont', 
                       low = 'purple', mid = 'white', high = 'green', midpoint = 0)+
  labs(x = '', y = '' )+
  theme( axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
         legend.position = 'left')

p <- cowplot::plot_grid(p2,p1, nrow = 1, rel_widths = c(.33,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','Dlat_pheno_subdom_alignment.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=6
  )
```

```{r fig.width=12, fig.height=6, warning=F}
sig = 0.1
t = 'Psmc3'

tmp <- gsea %>% 
  select(-gl, -res) %>%
  filter(tg == t, name == 'all') %>% 
  unnest(anno) %>% 
  ungroup() %>% 
  filter(Biodomain %in% c('Mitochondrial Metabolism', 
                          'Immune Response',
                          'Autophagy', 
                          'Cell Cycle',
                          'Apoptosis')) %>% 
  mutate( n_sig = length(unique(ID[ p.adjust <= sig ])),
          med_nes = median(NES[ p.adjust <= sig ]),
          .by = sd) %>% 
  mutate( med_nes = if_else(is.na(med_nes),0, med_nes),
          sd = fct_reorder(sd, med_nes) ) %>% 
  arrange(sd)

p1 <- ggplot(tmp, aes(NES, sd, fill = color))+
  geom_vline(xintercept = 0, lwd =.1)+
  geom_violin(data = subset(tmp, NES>0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_violin(data = subset(tmp, NES<0 & p.adjust <= sig), lwd = .2,
              alpha = .1, scale= 'width', draw_quantiles = .5)+
  geom_jitter(data = subset(tmp, p.adjust <= sig), shape = 21,
              alpha = .5, stroke = .5, aes(size = -log10(p.adjust)))+
  facet_grid(rows = vars(abbr), cols = vars(cell.2), scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(
    labels = function(x) str_remove_all(x, '^[A-Za-z]{2} \\| '), 
    position = 'right')+
  scale_fill_identity()+scale_color_identity()+
  scale_size_continuous(range = c(.5,4))+
  labs(y = '', title = t, subtitle = str_c('p.adjust \u2264 ', sig))+
  theme(legend.position = 'bottom',
        strip.text.y.left = element_text(angle = 0))

tmp <- biohyv.lmm %>% 
  filter(pheno != 'caspase',
         pheno != 'nfkb_LPS',
         target == str_to_upper(t)) %>%
  mutate(line = factor(line, c('scramble','psen2_kd')),
         pheno = case_when(pheno == 'nfkb_LPS' ~ 'NFkB +LPS',
                           pheno == 'nfkb_none' ~ 'NFkB -LPS',
                           pheno == 'alamar_blue' ~ 'alamarBlue',
                           T ~ pheno),
         pheno = factor(pheno, c('alamarBlue','mitotracker',
                                 'NFkB -LPS', 'NFkB +LPS', 'caspase', 'phagocytosis') )
         ) 

p2 <- ggplot(tmp, aes( target, line )) + 
  geom_point( aes(fill = lfc, size = -log10(p.value), shape = hit_class
                  , color = hit_class))+ #, alpha = hit_class
  facet_grid(rows = vars(pheno), scales = 'free', space = 'free', switch = 'y')+
  guides( alpha = 'none',  color = 'none', 
          size = guide_legend( override.aes=list(pch = 22)) ) +
  scale_y_discrete(limits = rev, position = 'right')+ 
  scale_size_continuous(guide = 'none')+ 
  scale_color_manual(values = c('grey20','grey50','grey20'))+
  scale_alpha_manual(values = c(1,1,1))+
  scale_shape_manual('Hit Class', values = c(24,22,25), 
                     breaks = c('Positive Hit','Non-Hit','Negative Hit') )+
  scale_fill_gradient2('Log2FC\nsiRNA v Cont', 
                       low = 'purple', mid = 'white', high = 'green', midpoint = 0)+
  labs(x = '', y = '' )+
  theme( axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
         legend.position = 'left')

p <- cowplot::plot_grid(p2,p1, nrow = 1, rel_widths = c(.33,1), align = 'h', axis = 'tb', labels = 'AUTO')
print(p)
```

```{r}
ggsave(
  here::here('manuscript','Psmc3_pheno_subdom_alignment.pdf'),
  plot = p,
  device = cairo_pdf,
  width=12, height=6
  )
```

## correlate NES and assay logFC

### gseGO

```{r fig.width=12, fig.height=4}
tmp3 <- full_join(
    biohyv.lmm %>% ungroup() %>% 
      select(pheno, target, line, lfc, p.value, p.adj.BH) %>% distinct(),
    
    gsea %>% ungroup() %>% unnest(anno) %>%
      mutate(tg = str_to_upper(tg)) %>%
      select(target = tg, line = cell.2, ID, 
             Description, Biodomain, Subdomain, NES, p.adjust, pvalue, abbr, color, sd) %>% 
      distinct()
  ) %>% 
  # filter(pvalue <= 0.05) %>%
  nest(data = c(target, line, lfc, p.value, p.adj.BH, NES, pvalue, p.adjust)) %>% 
  mutate(nr = map_dbl(data, ~nrow(.x))) %>% 
  filter(nr > 2) %>% 
  mutate(
    rho = map(data, ~ cor.test(.x$NES, .x$lfc, 
                               method = 'spearman') %>%
                broom::tidy()),
  ) %>% 
  unnest(rho) %>% 
  mutate(padj = p.adjust(p.value, method = 'BH'), .by = pheno)

p <- tmp3 %>% 
  filter(padj <= 0.05, pheno != 'nfkb_LPS') %>% 
  ggplot(aes(estimate, Biodomain))+
  geom_vline(xintercept = 0, lwd = .1)+
  facet_grid(cols = vars(pheno))+
  geom_jitter(
    aes(fill = color, size = -log10(p.value)),
    alpha = .6, shape = 21
    )+
  scale_fill_identity()+
  labs(x = "Spearman's \u03C1", y= '')

print(p)
```

```{r fig.width=16, fig.height=4.5}
x <- tmp3 %>% 
  select(-Subdomain, -sd) %>% 
  filter(
    pheno == 'phagocytosis' & Description == 'phagocytosis' & abbr == 'IR' | 
      pheno == 'mitotracker' & Description == 'mitochondrial inner membrane' & abbr == 'MM' | 
      pheno == 'nfkb_none' & Description == 'proteasome accessory complex' & abbr == 'Pr' |
      pheno == 'alamar_blue' & Description == 'regulation of mitotic cell cycle phase transition' & abbr == 'CC' 
  ) %>% 
  distinct() %>% 
  rename(pv = p.value) %>% 
  mutate(
    pheno = case_when(
      pheno == 'mitotracker' ~ 'MitoTracker',
      pheno == 'nfkb_none' ~ 'NFkB -LPS',
      pheno == 'alamar_blue' ~ 'alamarBlue',
      T~pheno
    )
  ) %>% 
  unnest(data) %>% 
  mutate(target = str_to_sentence(target))

p <- ggplot(x, aes(lfc, NES))+
  geom_vline(xintercept = 0, lwd = 0.2)+
  geom_hline(yintercept = 0, lwd = 0.2)+
  geom_point( data = subset(x, p.adjust > 0.05), aes(shape = line), fill = 'grey', size = 2, show.legend = F )+
  geom_point( data = subset(x, p.adjust <= 0.05), aes(shape = line, fill = -log10(p.adjust)), size = 3)+
  geom_smooth(method = 'lm', lwd = .2, alpha = .3)+
  ggrepel::geom_text_repel(
    data = subset(x, p.adjust <= 0.05 | p.adj.BH <= 0.05),
    aes(label = target), size = 3)+
  facet_wrap(~pheno+str_c(ID,': ',Description)+
               str_c("Spearman's \u03C1 = ",signif(estimate, digits = 3),
                     '; p = ',signif(padj, digits = 3)),
             scales = 'free', nrow = 1)+
  viridis::scale_fill_viridis('GSEA -log10(p.adjust)')+
  scale_shape_manual('BV2 cell',values=c(21,23), breaks = c('scramble','psen2_kd'))+
  labs(x = 'linear mixed model effect size', y = 'GSEA NES')+
  theme(legend.position = 'top')

print(p)

p.cor <- p

# ggsave(
#   here::here('manuscript','term_pheno_corr.pdf'),
#   p, width = 12, height = 4.5
# )
```

```{r fig.width=12, fig.height=3}
x = biohyv.lmm %>% 
  filter(
    pheno == 'alamar_blue' & target == 'PSMC3' |
      pheno == 'mitotracker' & target == 'PDHB' |
      pheno == 'nfkb_none' & target == 'PDHA1' |
      pheno == 'phagocytosis' & target == 'AP2A2'
  ) %>% 
  mutate(
    pheno = case_when(
      pheno == 'mitotracker' ~ 'MitoTracker',
      pheno == 'nfkb_none' ~ 'NFkB -LPS',
      pheno == 'alamar_blue' ~ 'alamarBlue',
      T~pheno),
    target = str_to_sentence(target)
  )

p <- ggplot(x, aes( lfc, target, fill = line )) + 
  facet_wrap(~pheno, scales = 'free', strip.position = 'top', nrow = 1)+
  geom_vline(xintercept = 0, lwd = .2)+
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  geom_errorbar(
    aes(xmin = conf.low, xmax = conf.high), 
    position = position_dodge(width =.9), lwd = .2, width = .3)+
  scale_fill_discrete('BV2 cell')+
  labs(y = '', x = 'linear mixed model effect size')+
  theme(legend.position = 'top')

print(p)

p.lmm <- p

# ggsave(
#   here::here('manuscript','pheno_panel.pdf'),
#   p, width = 9, height = 3
# )
```



```{r fig.width=16, fig.height=5}
tg = c('Psmc3','Pdhb','Pdha1','Ap2a2')
tm = c('regulation of mitotic cell cycle phase transition',
       'mitochondrial inner membrane',
       'proteasome accessory complex',
       'phagocytosis')

p <- map2(
  tg,tm,
  ~ {
    x <- gsea %>% filter(tg == .x, cell.2 == 'scramble') %>% pull(res) %>% .[[1]]
    i <- which(x$Description == .y)
    p1 <- gseaplot( 
      x, geneSetID = i[1], by = 'runningScore', color.line = '#00BA38',
      title = str_c(.x,' scramble', '\n', x$ID[i[1]],': ',x$Description[i[1]], '\n',
                    'NES = ', signif(x$NES[i[1]], digits = 3),
                    '; p.adj = ', signif(x$p.adjust[i[1]], digits = 3)))+
      theme(
        text = element_text(size = 8),
        axis.title = element_text(size = 10), 
        axis.text.x = element_text(size = 8),
        axis.text.y = element_text(size = 8),
        plot.margin = margin(5.5,20,5.5,20)
      )
    
    x <- gsea %>% filter(tg == .x, cell.2 == 'psen2_kd') %>% pull(res) %>% .[[1]]
    i <- which(x$Description == .y)
    p2 <- gseaplot( 
      x, geneSetID = i[1], by = 'runningScore', color.line = '#00BA38',
      title = str_c(.x,' psen2_kd', '\n', x$ID[i[1]],': ',x$Description[i[1]], '\n',
                    'NES = ', signif(x$NES[i[1]], digits = 3),
                    '; p.adj = ', signif(x$p.adjust[i[1]], digits = 3)))+
      theme(
        text = element_text(size = 8),
        axis.title = element_text(size = 10), 
        axis.text.x = element_text(size = 8),
        axis.text.y = element_text(size = 8),
        plot.margin = margin(5.5,20,5.5,20)
      )
    cowplot::plot_grid(p1, p2, nrow = 2)
          })

p.gse <- cowplot::plot_grid(plotlist = p, nrow = 1)

print(p.gse)

# ggsave(
#   here::here('manuscript','term_panel.pdf'),
#   p.gse, width = 12, height = 5
# )
```

```{r fig.width = 16, fig.height=12}
p <- cowplot::plot_grid( p.cor, p.lmm, p.gse, nrow = 3, rel_heights = c(.8,.5,1), labels = 'AUTO')

print(p)

ggsave(
  here::here('manuscript','term_pheno_corr_figure.pdf'),
  p, width = 16, height = 12,
  device = cairo_pdf
)
```



```{r fig.width=5, fig.height=9}
gh = 'AP2A2'
gm = 'Ap2a2'

# # plot raw data
# x = phagocytosis_lmm %>% 
#   # select(-value) %>% 
#   filter(siRNA %in% c(gh, 'Control')) %>% 
#   pivot_longer(., starts_with('rep')) %>% 
#   mutate( siRNA = factor( siRNA, levels = unique(phagocytosis_lmm$siRNA) ) )
# 
# p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
#   # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
#   geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .5, show.legend = F)+
#   geom_point( aes(color = BV2.cell.line) , position = position_jitterdodge(), show.legend = F )+
#   scale_y_discrete(limits = rev) +
#   scale_x_continuous(trans = 'log10')+
#   # facet_wrap(~hyp_area, scale = 'free_y')+
#   theme(legend.position = 'top') +
#   labs(y = 'siRNA target', x = '% positive cells\n ')

x = phagocytosis_lmm %>% 
  filter(siRNA %in% c(gh)) %>% 
  select(siRNA, BV2.cell.line, estimate, conf.low, conf.high, p.value, p.adj.BH) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(phagocytosis_lmm$siRNA) ) )

p2 <- ggplot(x, aes( estimate, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = 0, lwd = .2)+
  # geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  # geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'top') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = conf.low, xmax = conf.high), 
                position = position_dodge(width =.9),
                lwd = .2, width = .4)+
  labs(y = '', 
       # x = '% positive cells\nlog2 fold change, siRNA vs control'
       x = 'linear mixed model\neffect size'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow = 2 , rel_heights = c(1,.9), align = 'v', axis = 'lr', labels = 'AUTO')

x <- gsea %>% filter(tg == gm, name == 'all')
# x %>% unnest(anno) %>% ungroup() %>%
#     select(tg, cell.2, Biodomain, Subdomain, ID, Description, setSize, NES, p.adjust) %>%
#     pivot_wider(names_from = cell.2, values_from = c(p.adjust,NES,setSize))

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[1]]
z = left_join(y@result, biodom %>% rename('ID' = 'GO_ID'))
i = which(y$Description == 'phagocytosis')

p3 <- gseaplot(
    y, 
    geneSetID = i[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[1], '\n', 
                  y$ID[i[1]],': ',y$Description[i[1]], '\n',
                  'NES = ', signif(y$NES[i[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[i[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[2]]
j = which(y$Description == 'phagocytosis')

p4 <- gseaplot(
    y, 
    geneSetID = j[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[2], '\n', 
                  y$ID[j[1]],': ',y$Description[j[1]], '\n',
                  'NES = ', signif(y$NES[j[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[j[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# p <- cowplot::plot_grid(p,cowplot::plot_grid(p4, p3, nrow = 2, labels = c('C')), nrow = 1, rel_widths = c(1, .8))

p <- cowplot::plot_grid(
  p2,
  cowplot::plot_grid(p4, p3, nrow = 2),
  nrow = 2,
  rel_heights = c(.5,1),
  labels = 'AUTO'
)

print(p)

p.ph <- p

ggsave(
  here::here('manuscript','ap2a2_phagocytosis.pdf'),
  plot = p,
  width = 5, height = 9
)
```

```{r}
View( 
    tmp3 %>% 
    filter(p.value <= 0.05, !is.na(Biodomain), pheno == 'mitotracker') %>% 
    mutate(f = map_lgl(data, ~ .x %>% filter(target %in% c('PKM','PDHB','PDHA1','DLD','PDHX') & pvalue <= 0.05) %>% 
                           nrow() > 1)) %>% 
    filter(f, abbr == 'MM') %>% arrange(p.value) )
```


```{r fig.width=5, fig.height=9}
gh = 'PDHB'
gm = 'Pdhb'

# # plot raw data
# x = mitotracker_lmm %>% 
#   select(-value) %>%
#   filter(siRNA %in% c(gh, 'Control')) %>% 
#   pivot_longer(., starts_with('rep')) %>% 
#   mutate( siRNA = factor( siRNA, levels = unique(mitotracker_lmm$siRNA) ) )
# 
# p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
#   # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
#   geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .5, show.legend = F)+
#   geom_point( aes(color = BV2.cell.line) , position = position_jitterdodge(), show.legend = F )+
#   scale_y_discrete(limits = rev) +
#   scale_x_continuous(trans = 'log10')+
#   # facet_wrap(~hyp_area, scale = 'free_y')+
#   theme(legend.position = 'top') +
#   labs(y = 'siRNA target', x = 'MitoTracker avg intensity')

x = mitotracker_lmm %>% 
  filter(siRNA %in% c(gh)) %>% 
  select(siRNA, BV2.cell.line, estimate, conf.low, conf.high, p.value, p.adj.BH) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(mitotracker_lmm$siRNA) ) )

p2 <- ggplot(x, aes( estimate, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = 0, lwd = .2)+
  # geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  # geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'top') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = conf.low, xmax = conf.high), 
                position = position_dodge(width =.9),
                lwd = .2, width = .4)+
  labs(y = '', 
       # x = 'normalized intensity fold change, log2, siRNA vs control'
       x = 'linear mixed model\neffect size'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow = 2 , rel_heights = c(1,.9), align = 'v', axis = 'lr', labels = 'AUTO')

x <- gsea %>% filter(tg == gm, name == 'all')
# x %>% unnest(anno) %>% ungroup() %>% 
#     select(tg, cell.2, Biodomain, Subdomain, ID, Description, setSize, NES, p.adjust) %>% 
#     pivot_wider(names_from = cell.2, values_from = c(p.adjust,NES,setSize)) 


# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[1]]
# i = which(y$Description == 'electron transport chain')
i = which(y$Description == 'mitochondrial inner membrane')
# i = which(y$Description == 'pyruvate metabolic process')

p3 <- gseaplot(
    y, 
    geneSetID = i[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[1], '\n', 
                  y$ID[i[1]],': ',y$Description[i[1]], '\n',
                  'NES = ', signif(y$NES[i[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[i[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[2]]
# j = which(y$Description == 'electron transport chain')
j = which(y$Description == 'mitochondrial inner membrane')
# j = which(y$Description == 'pyruvate metabolic process')

p4 <- gseaplot(
    y, 
    geneSetID = j[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[2], '\n', 
                  y$ID[j[1]],': ',y$Description[j[1]], '\n',
                  'NES = ', signif(y$NES[j[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[j[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# p <- cowplot::plot_grid(p,cowplot::plot_grid(p4, p3, nrow = 2, labels = c('C')), nrow = 1, rel_widths = c(1, .8))

p <- cowplot::plot_grid(
  p2,
  cowplot::plot_grid(p4, p3, nrow = 2),
  nrow = 2,
  rel_heights = c(.5,1),
  labels = 'AUTO'
)

print(p)

p.mt <- p

ggsave(
  here::here('manuscript','pdhb_etc_details.pdf'),
  plot = p,
  width = 5, height = 9
)
```

```{r}
View( 
    tmp3 %>% 
    filter(p.value <= 0.05, !is.na(Biodomain), pheno == 'nfkb_none') %>% 
    mutate(f = map_lgl(data, ~ .x %>% filter(target %in% c('DLAT') & p.adjust <= 0.05) %>% 
                           nrow() > 1)) %>% 
    filter(f, abbr == 'IR') %>% arrange(p.value) )
```


```{r fig.width=5, fig.height=9}
gh = 'DLAT'
gm = 'Dlat'

# # plot raw data
# x = nfkb_lmm %>% 
#   # select(-value) %>%
#   filter(siRNA %in% c(gh, 'Control'), tx == 'none') %>% 
#   pivot_longer(., starts_with('rep')) %>% 
#   mutate( siRNA = factor( siRNA, levels = unique(nfkb_lmm$siRNA) ) )
# 
# p1 <- ggplot(x, aes( value, siRNA, fill = BV2.cell.line )) + 
#   # geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
#   geom_violin(scale = 'width', draw_quantiles = 0.5, color = 'grey30', alpha = .5, show.legend = F)+
#   geom_point( aes(color = BV2.cell.line) , position = position_jitterdodge(), show.legend = F )+
#   scale_y_discrete(limits = rev) +
#   scale_x_continuous(trans = 'log10')+
#   # facet_wrap(~hyp_area, scale = 'free_y')+
#   theme(legend.position = 'top') +
#   labs(y = 'siRNA target', x = 'MitoTracker avg intensity')

x = nfkb_lmm %>% 
  filter(siRNA %in% c(gh), tx == 'none') %>% 
  select(siRNA, BV2.cell.line, estimate, conf.low, conf.high, p.value, p.adj.BH) %>% 
  distinct() %>% 
  mutate( siRNA = factor( siRNA, levels = unique(nfkb_lmm$siRNA) ) )

p2 <- ggplot(x, aes( estimate, siRNA, fill = BV2.cell.line )) + 
  geom_bar(stat = 'identity', position = 'dodge', color = 'grey30', alpha= .6) +
  scale_y_discrete(limits = rev) +
  geom_vline(xintercept = 0, lwd = .2)+
  # geom_vline(xintercept = log2(1.5/1), lty = 2, lwd = .5)+
  # geom_vline(xintercept = log2(1/1.5), lty = 2, lwd = .5)+
  theme(legend.position = 'top') +
  # facet_wrap(~hyp_area, scales = 'free_y')+
  geom_errorbar(aes(xmin = conf.low, xmax = conf.high), 
                position = position_dodge(width =.9),
                lwd = .2, width = .4)+
  labs(y = '', 
       # x = 'normalized intensity fold change, log2, siRNA vs control'
       x = 'linear mixed model\neffect size'
       # , subtitle = 'normalized against cell line specific control siRNA'
       )

p <- cowplot::plot_grid(p1,p2, nrow = 2 , rel_heights = c(1,.9), align = 'v', axis = 'lr', labels = 'AUTO')

x <- gsea %>% filter(tg == gm, name == 'all')
# x %>% unnest(anno) %>% ungroup() %>%
#     select(tg, cell.2, Biodomain, Subdomain, ID, Description, setSize, NES, p.adjust) %>%
#     pivot_wider(names_from = cell.2, values_from = c(p.adjust,NES,setSize))


# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[1]]
# i = which(y$Description == 'electron transport chain')
i = which(y$Description == 'innate immune response')

p3 <- gseaplot(
    y, 
    geneSetID = i[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[1], '\n', 
                  y$ID[i[1]],': ',y$Description[i[1]], '\n',
                  'NES = ', signif(y$NES[i[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[i[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# i = which(grepl('hagocytosis',x$res[[1]]$Description))
y = x$res[[2]]
# j = which(y$Description == 'electron transport chain')
j = which(y$Description == 'innate immune response')

p4 <- gseaplot(
    y, 
    geneSetID = j[1], 
    by = 'runningScore', 
    title = str_c(x$cell.2[2], '\n', 
                  y$ID[j[1]],': ',y$Description[j[1]], '\n',
                  'NES = ', signif(y$NES[j[1]], digits = 3),
                  '; p.adj = ', signif(y$p.adjust[j[1]], digits = 3)),
    color.line = '#00BA38')+
    theme(
      text = element_text(size = 8),
      axis.title = element_text(size = 10), 
      axis.text.x = element_text(size = 8),
      axis.text.y = element_text(size = 8)
    )

# p <- cowplot::plot_grid(p,cowplot::plot_grid(p4, p3, nrow = 2, labels = c('C')), nrow = 1, rel_widths = c(1, .8))

p <- cowplot::plot_grid(
  p2,
  cowplot::plot_grid(p4, p3, nrow = 2),
  nrow = 2,
  rel_heights = c(.5,1),
  labels = 'AUTO'
)

print(p)

p.nfkb <- p

ggsave(
  here::here('manuscript','dlat_innate_immune_nfkb.pdf'),
  plot = p,
  width = 5, height = 9
)
```


```{r fig.width = 12, fig.height=10}
p <- cowplot::plot_grid(
  p.cor,
  cowplot::plot_grid(
    p.mt+theme(plot.margin = margin(5.5, 20, 5.5, 20)), 
    p.nfkb+theme(plot.margin = margin(5.5, 20, 5.5, 20)), 
    p.ph+theme(plot.margin = margin(5.5, 20, 5.5, 20)), 
    nrow = 1 ),
  nrow = 2, rel_heights = c(.65,1)
)

print(p)

ggsave(
  here::here('manuscript','term_pheno_corr_figure.pdf'),
  p, width = 12, height = 10
)
```



### domain fgsea

```{r echo=FALSE}
pathways <- bd.genes.mm %>% 
        summarise(genes = list(mm.symbol), .by = set) %>% 
  pull(genes, name = set)

bd.fgsea <- gsea %>% select(tg:gl) %>% 
  mutate(
    res = map(gl, 
              ~ fgsea::fgseaMultilevel(
                pathways = pathways, 
                stats = unlist(.x), 
                eps = 0, nproc = 4)
    )) %>% 
  unnest(res) %>%
  ungroup() %>% 
  mutate(tg = str_to_upper(tg)) %>%
  left_join(., dom.cols %>% select(abbr, pathway = domain)) %>% 
  mutate(abbr = if_else(grepl('_',pathway),
                        str_split_fixed(pathway, '_',2) %>% .[,1],
                        abbr)) %>% 
  left_join(., dom.cols %>% select(abbr, domain, color))

```

```{r}
ggplot(bd.fgsea, aes(NES, domain))+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_jitter(aes(fill = color, size = -log10(padj), shape = domain == pathway),
              alpha = .7)+
  facet_wrap(~tg+cell.2)+
  scale_fill_identity()+
  scale_shape_manual('',labels = c('Subdomain', 'Biodomain'), values = c(21,23))

```


```{r fig.width=12, fig.height=4}
tmp3 <- full_join(
    biohyv.lmm %>% ungroup() %>% 
      select(pheno, target, line, lfc, p.value,p.adj.BH) %>% distinct(),
      
    bd.fgsea %>% unnest(res) %>%
      ungroup() %>% 
      mutate(tg = str_to_upper(tg)) %>%
      select(target = tg, line = cell.2, pathway, NES, pval, padj) 
  ) %>% 
  filter(pval <= 0.1) %>%
  nest(data = c(target, line, lfc, p.value, p.adj.BH, NES, pval, padj)) %>% 
  mutate(nr = map_dbl(data, ~nrow(.x))) %>% 
  filter(nr > 2) %>% 
  mutate(
    rho = map(data, ~ cor.test(.x$NES, .x$lfc,method = 'spearman') %>%
                broom::tidy())
  ) %>% 
  unnest(rho) %>% 
  # mutate(padj = p.adjust(p.value, method = 'BH'), .by = pheno) %>% 
  left_join(., dom.cols %>% select(abbr, pathway = domain)) %>% 
  mutate(abbr = if_else(grepl('_',pathway),
                        str_split_fixed(pathway, '_',2) %>% .[,1],
                        abbr)) %>% 
  left_join(., dom.cols %>% select(abbr, domain, color))

p <- tmp3 %>% 
  filter(p.value <= 0.05, pheno != 'nfkb_LPS') %>% 
  ggplot(aes(estimate, domain))+
  geom_vline(xintercept = 0, lwd = .1)+
  facet_grid(cols = vars(pheno))+
  geom_jitter(
    aes(fill = color, size = -log10(p.value), shape = pathway == domain),
    alpha = .6
    )+
  scale_shape_manual(
    '', labels = c('Subdomain', 'Biodomain'),
    values = c(21,23))+
  scale_fill_identity()+
  labs(x = "Spearman's \u03C1", y= '')

print(p)
```


```{r}
tmp1 <- full_join(
    biohyv.lmm %>% ungroup() %>% 
      select(pheno, target, line, lfc, p.value, p.adj.BH) %>% distinct(),
    
    gsea %>% ungroup() %>% unnest(anno) %>%
      mutate(tg = str_to_upper(tg)) %>%
      select(target = tg, line = cell.2, ID, 
             Description, Biodomain, Subdomain, NES, p.adjust, pvalue, abbr, color, sd) %>% 
      distinct()
  ) %>% 
  filter(pvalue <= 0.3, p.value <= 0.3) %>%
  nest(data = c(target, line, lfc, p.value, p.adj.BH, NES, pvalue, p.adjust)) %>% 
  mutate(nr = map_dbl(data, ~nrow(.x))) %>% 
  filter(nr > 2) %>% 
  mutate(
    rho = map(data, ~ cor.test(.x$NES, .x$lfc, 
                               method = 'spearman') %>%
                broom::tidy()),
  ) %>% 
  unnest(rho) %>% 
  mutate(padj = p.adjust(p.value, method = 'BH'), .by = pheno)

tmp3 <- full_join(
    biohyv.lmm %>% ungroup() %>% 
      select(pheno, target, line, lfc, p.adj.BH) %>% distinct(),
      
    bd.fgsea %>% unnest(res) %>%
      ungroup() %>% 
      mutate(tg = str_to_upper(tg)) %>%
      select(target = tg, line = cell.2, pathway, NES, pval, padj) 
  ) %>% 
  filter(padj <= 0.05) %>%
  nest(data = c(target, line, lfc, p.adj.BH, NES, pval, padj)) %>% 
  mutate(nr = map_dbl(data, ~nrow(.x))) %>% 
  filter(nr > 2) %>% 
  mutate(
    rho = map(data, ~ cor.test(.x$NES, .x$lfc, 
                               method = 'spearman') %>%
                broom::tidy()),
  ) %>% 
  unnest(rho) %>% 
  mutate(padj = p.adjust(p.value, method = 'BH'), .by = pheno)

tmp2 <- gsea %>% ungroup() %>% unnest(anno) %>% select(tg, cell.2, Biodomain, Subdomain, Description, setSize, p.adjust, NES) %>% distinct() 
```
```{r}
tmp3 <- full_join(
    biohyv.lmm %>% ungroup() %>% 
      select(pheno, target, line, lfc, p.adj.BH) %>% distinct(),
      
    bd.fgsea %>% unnest(res) %>%
      ungroup() %>% 
      mutate(tg = str_to_upper(tg)) %>%
      select(target = tg, line = cell.2, pathway, NES, pval, padj) 
  ) %>% 
  filter(padj <= 0.05) %>%
  nest(data = c(target, line, lfc, p.adj.BH, NES, pval, padj)) %>% 
  mutate(nr = map_dbl(data, ~nrow(.x))) %>% 
  filter(nr > 2) %>% 
  mutate(
    rho = map(data, ~ cor.test(.x$NES, .x$lfc, 
                               method = 'spearman') %>%
                broom::tidy()),
  ) %>% 
  unnest(rho) %>% 
  mutate(padj = p.adjust(p.value, method = 'BH'), .by = pheno) %>% 
  mutate(abbr = if_else(grepl('_',pathway),
                        str_split_fixed(pathway, '_',2) %>% .[,1],
                        NA_character_)) %>% 
  left_join(.,dom.cols %>% select(abbr, domain, color))


```



```{r}
tmp <- full_join(
    biohyv.lmm %>% ungroup() %>% 
        select(pheno, target, line, lfc) %>% 
        pivot_wider(names_from = line, values_from = lfc) %>%
        mutate(d.lfc = psen2_kd - scramble) %>% 
        select(-psen2_kd, -scramble),
    bd.fgsea %>% unnest(res) %>%
        ungroup() %>% 
        mutate(tg = str_to_upper(tg)) %>%
        select(target = tg, line = cell.2, pathway, NES) %>% 
        pivot_wider(names_from = line, values_from = NES) %>% 
        mutate(d.nes = psen2_kd - scramble) %>% 
        select(-psen2_kd, -scramble)
  ) %>% 
  nest(data = c(target,d.lfc, d.nes)) %>% 
  mutate(nr = map_dbl(data, ~nrow(.x))) %>% 
  filter(nr > 2) %>% 
  mutate(
    rho = map(data, ~ cor.test(.x$d.nes, .x$d.lfc, 
                               method = 'spearman') %>%
                broom::tidy()),
  ) %>% 
  unnest(rho) %>% 
  mutate(padj = p.adjust(p.value, method = 'BH'), .by = pheno)


```

```{r}
bv.pheno.sd <- 
  left_join(biohyv.lmm,
            
            # bd.fgsea %>% unnest(res) %>%
            #   mutate(tg = str_to_upper(tg)) %>%
            #   select(target = tg, line = cell.2, pathway, pval, padj, NES, size),
            
            gsea %>%
              mutate(tg = str_to_upper(tg)) %>%
              filter(name == 'all') %>%
              select(-gl, -res, -name) %>%
              unnest(anno) %>%
              # mutate(Subdomain = if_else(is.na(Subdomain), 'none',Subdomain),
              #         sd = str_c(abbr, ' | ', Subdomain)) %>%
              nest(anno = ONTOLOGY:sd) %>%
              rename(target = tg, line = cell.2),
            
            by = c('target','line')
            ) %>% 
  left_join(., biohyv_targets)
```

```{r fig.width=7, fig.height=5}
tmp <- bv.pheno.sd %>%
  unnest(anno) %>%
  filter(
    # p.value <= 0.05,
    # p.adj.BH <= .05,
    # abs(lfc) > log2(1.5/1), # hit_class != 'Non-Hit', # cell assay signif
    # p.adjust <= 0.1 # GSEA signif
    # pvalue <= .05,
    , pheno != 'caspase'
    # , !is.na(Biodomain), !is.na(sd)
    ) %>% 

  group_by(pheno, line, target, lfc, sd) %>%  # hyp_area,
  summarise( mn_NES = median(NES) ) %>%
  group_by(pheno, sd) %>% #, hyp_area 

  # group_by(pheno, pathway) %>% #, hyp_area 
  summarise(
    n = length(unique(target)),
    # tau = try(Kendall::Kendall(mn_NES,lfc) %>% broom::tidy(),silent = T) %>% list(),
    tau = try(cor.test(mn_NES, lfc, method = 'kendall') %>% broom::tidy(), silent = T) %>% list(),
    rho = try(cor.test(mn_NES, lfc, method = 'spearman') %>% broom::tidy(), silent = T) %>% list()
  ) %>%
  mutate( 
    # abbr = str_extract(sd, '^[A-Za-z]{2}'),
    tau.no.err = map_lgl(tau, ~ length(.x) > 1),
    rho.no.err = map_lgl(rho, ~ length(.x) > 1)) %>% 
  filter(rho.no.err) %>% unnest(rho) %>% 
  # filter(tau.no.err) %>% unnest(tau) %>% 
  mutate(padj = p.adjust(p.value, method = 'BH'))

p <- tmp %>% 
  filter(
    # grepl('_',pathway) == T
    , padj <= 0.05
    ) %>%
  mutate(
    # pathway = fct_reorder(pathway, estimate),
    # abbr = str_split_fixed(pathway, '_', 2) %>% .[,1]
      abbr = str_split_fixed(sd, ' \\| ', 2) %>% .[,1]
    ) %>%
  # ggplot(aes(estimate, pathway))+
  ggplot(aes(estimate, sd))+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_point(alpha = .7, aes(color = -log10(p.value), shape = padj <= 0.05) )+ #, shape = p.value <= 0.05
  facet_grid(cols = vars(pheno), rows = vars(abbr), 
             scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(position = 'right', label = ~ str_remove(.x, '^[A-Za-z]{2} \\| '))+
  scale_size_continuous('# targets')+
  # scale_shape_manual('significant', values = c(22,18))+
  viridis::scale_color_viridis()+
  # labs(x = "Kendall's \u03c4", y = '')+
  labs(x = "Spearman's \u03c1", y = '')+
  theme(
    legend.position = 'bottom',
    legend.box = 'vertical',
    strip.text.y.left = element_text(angle = 0)
  )

print(p)
```

```{r fig.width=7, fig.height=5}
tmp <- bv.pheno.sd %>% 
  unnest(hyp_area) %>% 
  unnest(anno) %>% 
  filter(
    p.value <= 0.05,
    # p.adj.BH <= .05,
    # abs(lfc) > log2(1.5/1), # hit_class != 'Non-Hit', # cell assay signif
    # p.adjust <= 0.1 # GSEA signif
    pvalue <= .05,
    , pheno != 'caspase'
    , !is.na(Biodomain), !is.na(sd)
    ) %>% 
  # group_by(pheno, line, target, lfc, sd) %>%  # hyp_area, 
  # summarise( mn_NES = median(NES) ) %>% 
  group_by(pheno, Description, Biodomain, Subdomain) %>% #, hyp_area 
  summarise(
    n = length(unique(target)),
    # tau = try(Kendall::Kendall(mn_NES,lfc) %>% broom::tidy(),silent = T) %>% list(),
    tau = try(cor.test(NES, lfc, method = 'kendall') %>% broom::tidy(), silent = T) %>% list(),
    rho = try(cor.test(NES, lfc, method = 'spearman') %>% broom::tidy(), silent = T) %>% list()
  ) %>%
  mutate( 
    # abbr = str_extract(sd, '^[A-Za-z]{2}'),
    tau.no.err = map_lgl(tau, ~ length(.x) > 1),
    rho.no.err = map_lgl(rho, ~ length(.x) > 1)) %>% 
  filter(rho.no.err) %>% unnest(rho) 

p <- tmp %>% 
  # filter(tau.no.err) %>% unnest(tau) %>% 
  filter(rho.no.err) %>% unnest(rho) %>%
  mutate(padj = p.adjust(p.value, method = 'BH')) %>%
  filter(p.value <= 0.05) %>%
  mutate(sd = fct_reorder(sd, estimate)) %>% 
  ggplot(aes(estimate, sd))+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_point(alpha = .7, aes(size = n, color = -log10(p.value)) )+ #, shape = p.value <= 0.05
  facet_grid(cols = vars(pheno), rows = vars(abbr), 
             scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(position = 'right', label = ~ str_remove(.x, '^[A-Za-z]{2} \\| '))+
  scale_size_continuous('# targets')+
  # scale_shape_manual('significant', values = c(22,18))+
  viridis::scale_color_viridis()+
  # labs(x = "Kendall's \u03c4", y = '')+
  labs(x = "Spearman's \u03c1", y = '')+
  theme(
    legend.position = 'bottom',
    legend.box = 'vertical',
    strip.text.y.left = element_text(angle = 0)
  )

print(p)
```

```{r}
ggsave(
  here::here('manuscript','comp_pheno_gsea.pdf'),
  plot = p,
  device = cairo_pdf,
  width=7, height=5
  )
```

```{r fig.width=6, fig.height=4}
tmp <- bv.pheno.sd %>% 
  unnest(hyp_area) %>% 
  unnest(anno) %>% 
  filter(
    p.value <= 0.05,
    # p.adj.BH < .5, #abs(lfc) > log2(1.5/1), # hit_class != 'Non-Hit', # cell assay signif
    pvalue <= 0.05,
    # p.adjust < .5 # GSEA signif
    , pheno == 'phagocytosis', pheno != 'caspase'
    , Biodomain == 'Immune Response', Subdomain == 'phagocytosis'
         ) %>% 
  group_by(line, target, hyp_area, lfc, p.value, sd, color) %>% 
  summarise(mn_NES = median(NES), mn_padj = median(-log10(p.adjust))) %>% 
  mutate(sd = str_replace_all(sd,' \\| ', '\n')) 

p <- ggplot(tmp, aes(lfc, mn_NES))+ 
  geom_hline(yintercept = 0, lwd = .1)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_smooth(method = 'lm', lwd = .2, lty = 2, se = F)+
  geom_point(alpha = .5, color = 'darkblue', aes(size = mn_padj))+ # color = line, 
  ggrepel::geom_text_repel(
    aes(label = target), 
    min.segment.length = 0, size = 5, alpha = .9
  )+
  facet_grid( rows = vars(str_trunc(sd, width = 25)) )+
  scale_color_discrete('cell line')+
  scale_size_continuous('median GSEA\np.adjusted (-log10)', range = c(2,5))+
  labs(x = 'assay log fold change', y = 'median GSEA NES'
       ,title = 'phagocytosis assay'
       )+
  theme(legend.position = 'right')

print(p)

```

```{r}
ggsave(
  here::here('manuscript','phagocytosis_assay_vs_NES.pdf'),
  plot = p,
  device = cairo_pdf,
  width=6, height=4
  )
```

```{r fig.width=6, fig.height=4}
tmp <- bv.pheno.sd %>% 
  unnest(hyp_area) %>% 
  unnest(anno) %>% 
  filter(
    p.value <= 0.05,
    # p.adj.BH < .5, #abs(lfc) > log2(1.5/1), # hit_class != 'Non-Hit', # cell assay signif
    pvalue <= 0.05,
    # p.adjust < .5 # GSEA signif
    , pheno == 'mitotracker', pheno != 'caspase'
    , Biodomain == 'Mitochondrial Metabolism', Subdomain == 'electron transport chain'
         ) %>% 
  group_by(line, target, hyp_area, lfc, p.value, sd, color) %>% 
  summarise(mn_NES = median(NES), mn_padj = median(-log10(p.adjust))) %>% 
  mutate(sd = str_replace_all(sd,' \\| ', '\n')) 

p <- ggplot(tmp, aes(lfc, mn_NES))+ 
  geom_hline(yintercept = 0, lwd = .1)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_smooth(method = 'lm', lwd = .2, lty = 2, se = F)+
  geom_point(alpha = .5
             # , color = 'darkblue'
             , aes(size = mn_padj
                   ,color = line
                   ))+ # , 
  ggrepel::geom_text_repel(
    aes(label = target), 
    min.segment.length = 0, size = 5, alpha = .9
  )+
  facet_grid( rows = vars(str_trunc(sd, width = 25)) )+
  scale_color_discrete('cell line')+
  scale_size_continuous('median GSEA\np.adjusted (-log10)', range = c(2,5))+
  labs(x = 'assay log fold change', y = 'median GSEA NES'
       ,title = 'phagocytosis assay'
       )+
  theme(legend.position = 'right')

print(p)

```

```{r fig.width=6, fig.height=4}
tmp <- bv.pheno.sd %>% 
  unnest(hyp_area) %>% 
  unnest(anno) %>% 
  filter(
    p.value <= 0.9,
    # p.adj.BH < .5, #abs(lfc) > log2(1.5/1), # hit_class != 'Non-Hit', # cell assay signif
    pvalue <= 0.9,
    # p.adjust < .5 # GSEA signif
    , pheno == 'NFkB_reporter_LPS_stim', pheno != 'caspase'
    , Biodomain == 'Immune Response', Subdomain == 'adaptive immune response'
         ) %>% 
  group_by(line, target, hyp_area, lfc, p.value, sd, color) %>% 
  summarise(mn_NES = median(NES), mn_padj = median(-log10(p.adjust))) %>% 
  mutate(sd = str_replace_all(sd,' \\| ', '\n')) 

p <- ggplot(tmp, aes(lfc, mn_NES))+ 
  geom_hline(yintercept = 0, lwd = .1)+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_smooth(method = 'lm', lwd = .2, lty = 2, se = F)+
  geom_point(alpha = .5,  aes(color = line, size = mn_padj))+ # color = 'darkblue',
  ggrepel::geom_text_repel(
    aes(label = target), 
    min.segment.length = 0, size = 5, alpha = .9
  )+
  facet_grid( rows = vars(str_trunc(sd, width = 100)) )+
  scale_color_discrete('cell line')+
  scale_size_continuous('median GSEA\np.adjusted (-log10)', range = c(2,5))+
  labs(x = 'assay log fold change', y = 'median GSEA NES'
       ,title = 'NF\u03baB reporter assay (+LPS)'
       )+
  theme(legend.position = 'right')

print(p)
```

```{r}
ggsave(
  here::here('manuscript','nfkb_assay_vs_NES.pdf'),
  plot = p,
  device = cairo_pdf,
  width=6, height=4
  )
```

## correlate prot logFC with pheno logFC

```{r}
bv.pheno.sd.prot <- 
  left_join(biohyv.lmm,
            de.results %>%
              filter(
                !is.na(diff), 
                (cell.1 == cell.2 & (grepl('siRNA', tg.1)|grepl('siRNA',tg.2))) 
                | (grepl('siRNA', tg.1) & grepl('siRNA',tg.2)) 
                ) %>% 
              mutate(tg = str_to_upper(tg)) %>% 
              select(target = tg, line = cell.2, prot = gene, log2.fc = diff, p.adj, n.pep) %>%
              left_join(., bd.genes.mm %>% select(prot = mm.symbol, set)) %>%
              distinct() %>% 
              nest(anno = c(set, prot, log2.fc, p.adj, n.pep)) 
            , by = c('target','line')
            ) %>% 
  left_join(., biohyv_targets)
```

```{r fig.width=10, fig.height=10}
tmp <- bv.pheno.sd.prot %>% 
  # unnest(hyp_area) %>% 
  unnest(anno) %>% 
  filter(
    # p.value <= 0.1,
    p.adj.BH <= .05, # assay signif
    # abs(lfc) > log2(1.5/1), # hit_class != 'Non-Hit', # cell assay signif
    p.adj <= 0.1 # prot signif
    # pvalue <= .05
    , pheno != 'caspase'
    , grepl('_',set)) %>% 
  group_by(pheno, line, target, log2.fc, set) %>%  # hyp_area, 
  summarise( mn_lfc = median(log2.fc, na.rm = T) ) %>% 
  group_by(pheno, set) %>% #, hyp_area 
  summarise(
    n = length(unique(target)),
    # tau = try(Kendall::Kendall(mn_NES,lfc) %>% broom::tidy(),silent = T) %>% list(),
    tau = try(cor.test(mn_lfc, lfc, method = 'kendall') %>% broom::tidy(), silent = T) %>% list(),
    rho = try(cor.test(mn_lfc, lfc, method = 'spearman') %>% broom::tidy(), silent = T) %>% list()
  ) %>%
  mutate( 
    abbr = str_extract(set, '^[A-Za-z]{2}'),
    tau.no.err = map_lgl(tau, ~ length(.x) > 1),
    rho.no.err = map_lgl(rho, ~ length(.x) > 1))

p <- tmp %>% 
  # filter(tau.no.err) %>% unnest(tau) %>% 
  filter(rho.no.err) %>% unnest(rho) %>%
  mutate(padj = p.adjust(p.value, method = 'BH')) %>%
  filter(p.value <= 0.05) %>%
  mutate(set = fct_reorder(set, estimate)) %>% 
  ggplot(aes(estimate, set))+
  geom_vline(xintercept = 0, lwd = .1)+
  geom_point(alpha = .7, aes(size = n, color = -log10(p.value)) )+
  facet_grid(cols = vars(pheno), rows = vars(abbr), 
             scales = 'free_y', space = 'free', switch = 'y')+
  scale_y_discrete(position = 'right', label = ~ str_remove(.x, '^[A-Za-z]{2}_'))+
  scale_size_continuous('# targets')+
  viridis::scale_color_viridis()+
  # labs(x = "Kendall's \u03c4", y = '')+
  labs(x = "Spearman's \u03c1", y = '')+
  theme(
    legend.position = 'bottom',
    legend.box = 'vertical',
    strip.text.y.left = element_text(angle = 0)
  )

print(p)
```

# Session
```{r}
save.image(here::here('results','biohyv_bv2_manuscript.rdata'))
sessionInfo()
```

